commit dc2ec87ba105935d8446cf6e0df1f005f1bce5c8
Author: Montel Laurent <montel@kde.org>
Date:   Wed Feb 22 09:50:05 2012 +0100

    Fix crash for user which still use 4.4

diff --git a/kmailcvt/filter_pmail.cxx b/kmailcvt/filter_pmail.cxx
index 56064d6..714d410 100644
--- a/kmailcvt/filter_pmail.cxx
+++ b/kmailcvt/filter_pmail.cxx
@@ -75,7 +75,6 @@ void FilterPMail::import(FilterInfo *info)
     info->addLog( i18n("Finished importing emails from %1", chosenDir ));
     info->setCurrent(100);
     info->setOverall(100);
-    delete kfd;
 }
 
 /** this looks for all files with the filemask 'mask' and calls the 'workFunc' on each of them */

commit fb1874f8e2805f0afb3b9e848e8fd4d9ee11a97f
Author: Montel Laurent <montel@kde.org>
Date:   Mon Jun 20 13:27:05 2011 +0200

    Fix bug #276086 crash on exit

diff --git a/kmail/accountmanager.cpp b/kmail/accountmanager.cpp
index 1745aa4..ca2a7c8 100644
--- a/kmail/accountmanager.cpp
+++ b/kmail/accountmanager.cpp
@@ -227,7 +227,8 @@ void AccountManager::processNextCheck( bool _newMail )
 
   curAccount->setCheckingMail( true );
   mAcctChecking.append( curAccount );
-  kmkernel->filterMgr()->ref();
+  if( kmkernel->filterMgr() )
+      kmkernel->filterMgr()->ref();
   curAccount->processNewMail( mInteractive );
 }
 

commit 6f62ea1bff318d8efea42f59bdc965ee2ab187a0
Author: Thomas McGuire <mcguire@kde.org>
Date:   Wed May 11 18:21:32 2011 +0100

    Make spell checking work again.
    The text edit used the wrong config file for loading spell settings.
    After using the config file, the correct settigns get loaded again,
    since the Loader from Sonnet is a singleton and therefore shares the
    settings, so even the textedit with the wrong settings file name gets
    updated.
    
    cherry-picked from d1708effbb68d6eae36ee5177a599c965973725f.
    
    BUG: 247486

diff --git a/kmail/kmcomposereditor.cpp b/kmail/kmcomposereditor.cpp
index f1accab..74de65b 100644
--- a/kmail/kmcomposereditor.cpp
+++ b/kmail/kmcomposereditor.cpp
@@ -44,7 +44,7 @@
 #include <QFileInfo>
 
 KMComposerEditor::KMComposerEditor( KMComposeWin *win,QWidget *parent)
- :KMeditor(parent),m_composerWin(win)
+ :KMeditor(parent, "kmailrc"),m_composerWin(win)
 {
 }
 
diff --git a/libkdepim/kmeditor.cpp b/libkdepim/kmeditor.cpp
index e8f0d36..067c781 100644
--- a/libkdepim/kmeditor.cpp
+++ b/libkdepim/kmeditor.cpp
@@ -28,6 +28,7 @@
 #include <KProcess>
 #include <KPushButton>
 #include <KTemporaryFile>
+#include <kdeversion.h>
 
 #include <QApplication>
 #include <QClipboard>
@@ -215,6 +216,17 @@ KMeditor::KMeditor( QWidget *parent )
   d->init();
 }
 
+KMeditor::KMeditor( QWidget *parent, const QString & configFile )
+#if KDE_IS_VERSION(4,6,0)
+ : TextEdit( parent, configFile ), d( new KMeditorPrivate( this ) )
+#else
+ : TextEdit( parent ), d( new KMeditorPrivate( this ) )
+#endif
+{
+  Q_UNUSED( configFile )
+  d->init();
+}
+
 KMeditor::~KMeditor()
 {
   delete d;
diff --git a/libkdepim/kmeditor.h b/libkdepim/kmeditor.h
index 4aab8aa..66ebb4a 100644
--- a/libkdepim/kmeditor.h
+++ b/libkdepim/kmeditor.h
@@ -61,6 +61,12 @@ class KDEPIM_EXPORT KMeditor : public KPIMTextEdit::TextEdit
      */
     explicit KMeditor( QWidget *parent = 0 );
 
+    /**
+     * Constructs a KMeditor object.
+     */
+    explicit KMeditor( QWidget *parent, const QString& configFile );
+
+
     virtual ~KMeditor();
 
     //Redefine it for each apps

commit f1b87b19592bcc99e4aacdbf6a68d21a9669615e
Merge: 4b504e7 0b44b1a
Author: David Jarvie <djarvie@kde.org>
Date:   Tue Apr 26 23:30:28 2011 +0100

    Merge branch '4.4' of git.kde.org:kdepim into 4.4

commit 4b504e760e0c994e676b55bfaca731d74841f1f4
Author: David Jarvie <djarvie@kde.org>
Date:   Tue Apr 26 23:29:19 2011 +0100

    Bug 271580: Fix crash displaying New Alarm dialog from system tray
    
    Fixes a crash on some systems when the New Alarm dialog is displayed
    from the system tray icon menu.

diff --git a/kalarm/lib/spinbox2.cpp b/kalarm/lib/spinbox2.cpp
index cc10882..f1e8ab9 100644
--- a/kalarm/lib/spinbox2.cpp
+++ b/kalarm/lib/spinbox2.cpp
@@ -1,7 +1,7 @@
 /*
  *  spinbox2.cpp  -  spin box with extra pair of spin buttons (for Qt 3)
  *  Program:  kalarm
- *  Copyright © 2001-2009 by David Jarvie <djarvie@kde.org>
+ *  Copyright © 2001-2009,2011 by David Jarvie <djarvie@kde.org>
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -28,7 +28,7 @@
 #include <QMouseEvent>
 #include <QStyleOptionSpinBox>
 #include <QGraphicsPixmapItem>
-#include <QPaintEngine>
+#include <QPainter>
 #include <QTimer>
 #include <QFrame>
 #include <QBrush>
@@ -588,8 +588,9 @@ void SpinMirror::setFrame()
 		p = grabWidget(mMainSpinbox, QRect(x, 0, 1, height()));
 		// Blot out edit field stuff from the middle of the slice
 		QPixmap dot = grabWidget(mMainSpinbox, QRect(x, editOffsetY, 1, 1));
-		QPaintEngine* pe = p.paintEngine();
-		pe->drawTiledPixmap(QRectF(0, editOffsetY, 1, height() - 2*editOffsetY), dot, QPointF(0, 0));
+		QPainter painter(&p);
+		painter.drawTiledPixmap(0, editOffsetY, 1, height() - 2*editOffsetY, dot, 0, 0);
+		painter.end();
 		// Horizontally fill the mirror widget with the vertical slice
 		p = p.scaled(size());
 		// Grab the left hand border of the main spinbox, and draw it into the mirror widget.

commit 0b44b1aa3d66f974e3d255c6e027947f1375b685
Author: Stephen Kelly <steveire@gmail.com>
Date:   Mon Apr 25 22:09:08 2011 +0200

    Only start akonadi asynchronously on KDE 4.6.2.
    
    Revises cfa404b7188e4c26bddbc9579728f6d25f8cd214 to hopefully fix
    the bug seen on fedora.
    
    Please re-test this on 4.6.2 on fedora too. Unfortunately that's the
    best I can do because I can't reproduce the issue.
    
    BUG: 268120

diff --git a/kaddressbook/main.cpp b/kaddressbook/main.cpp
index 200f8ec..e4cb3f7 100644
--- a/kaddressbook/main.cpp
+++ b/kaddressbook/main.cpp
@@ -49,7 +49,7 @@ int main( int argc, char **argv )
   MainWindow *window = new MainWindow;
   window->show();
 
-#if !KDE_IS_VERSION(4,6,0)
+#if !KDE_IS_VERSION(4,6,2)
   if ( !Akonadi::Control::start( window ) ) {
     //TODO: add message box after string freeze
     kWarning() << "Unable to start Akonadi server, exit application";
diff --git a/kmail/kmmainwidget.cpp b/kmail/kmmainwidget.cpp
index 10b14e4..f6461d9 100644
--- a/kmail/kmmainwidget.cpp
+++ b/kmail/kmmainwidget.cpp
@@ -189,7 +189,7 @@ KMMainWidget::KMMainWidget( QWidget *parent, KXMLGUIClient *aGUIClient,
     mVacationIndicatorActive( false ),
     mGoToFirstUnreadMessageInSelectedFolder( false )
 {
-#if KDE_IS_VERSION(4,6,0)
+#if KDE_IS_VERSION(4,6,2)
   Akonadi::Control::widgetNeedsAkonadi(this);
   Akonadi::ServerManager::start();
 #endif
diff --git a/kmail/main.cpp b/kmail/main.cpp
index c2dd1c3..f59d5e6 100644
--- a/kmail/main.cpp
+++ b/kmail/main.cpp
@@ -145,7 +145,7 @@ int main(int argc, char *argv[])
   app.setEventLoopReached();
   app.delayedInstanceCreation();
 
-#if !KDE_IS_VERSION(4,6,0)
+#if !KDE_IS_VERSION(4,6,2)
   // Start Akonadi
   if ( !Akonadi::Control::start( kmkernel->getKMMainWidget() ) ) {
     //TODO: add message box after string freeze
diff --git a/kontact/src/main.cpp b/kontact/src/main.cpp
index cb4a235..ae460be 100644
--- a/kontact/src/main.cpp
+++ b/kontact/src/main.cpp
@@ -199,7 +199,7 @@ int main( int argc, char **argv )
 
   KontactApp app;
 
-#if !KDE_IS_VERSION(4,6,0)
+#if !KDE_IS_VERSION(4,6,2)
   // KDE 4.4: do akonadi startup before creating any window, since creating
   // the window loads kmail. In 4.5 we'll do this startup async instead.
   Akonadi::Control::start( 0 );

commit 712e45630ccd4ec128a087cb3de396b2df9e4ffc
Author: David Jarvie <djarvie@kde.org>
Date:   Fri Apr 22 00:36:20 2011 +0100

    Bug 266082: highlight alarm when message window KAlarm button clicked
    
    Fix the KAlarm button in the alarm window not always showing the main
    window, and not highlighting the alarm in the main window.

diff --git a/kalarm/eventlistview.cpp b/kalarm/eventlistview.cpp
index 75e8be8..0880c7f 100644
--- a/kalarm/eventlistview.cpp
+++ b/kalarm/eventlistview.cpp
@@ -1,7 +1,7 @@
 /*
  *  eventlistview.cpp  -  base class for widget showing list of alarms
  *  Program:  kalarm
- *  Copyright © 2007-2009 by David Jarvie <djarvie@kde.org>
+ *  Copyright © 2007-2009,2011 by David Jarvie <djarvie@kde.org>
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -62,15 +62,18 @@ KAEvent* EventListView::event(int row) const
 
 /******************************************************************************
 * Select one event and make it the current item.
+* Optionally scroll to ensure that the  event is visible.
 */
-void EventListView::select(const QString& eventId)
+void EventListView::select(const QString& eventId, bool scrollToEvent)
 {
-	select(eventModel()->eventIndex(eventId));
+	select(eventModel()->eventIndex(eventId), scrollToEvent);
 }
 
-void EventListView::select(const QModelIndex& index)
+void EventListView::select(const QModelIndex& index, bool scrollToIndex)
 {
 	selectionModel()->select(index, QItemSelectionModel::SelectCurrent | QItemSelectionModel::Rows);
+	if (scrollToIndex)
+		scrollTo(index);
 }
 
 /******************************************************************************
diff --git a/kalarm/eventlistview.h b/kalarm/eventlistview.h
index 23b97b1..f0dc7ca 100644
--- a/kalarm/eventlistview.h
+++ b/kalarm/eventlistview.h
@@ -1,7 +1,7 @@
 /*
  *  eventlistview.h  -  base class for widget showing list of alarms
  *  Program:  kalarm
- *  Copyright © 2007,2008 by David Jarvie <djarvie@kde.org>
+ *  Copyright © 2007,2008,2011 by David Jarvie <djarvie@kde.org>
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -42,8 +42,8 @@ class EventListView : public QTreeView
 		EventListModel*   eventModel() const   { return static_cast<EventListModel*>(static_cast<QAbstractProxyModel*>(model())->sourceModel()); }
 		KAEvent*          event(int row) const;
 		KAEvent*          event(const QModelIndex&) const;
-		void              select(const QString& eventId);
-		void              select(const QModelIndex&);
+		void              select(const QString& eventId, bool scrollToEvent = false);
+		void              select(const QModelIndex&, bool scrollToIndex = false);
 		QModelIndex       selectedIndex() const;
 		KAEvent*          selectedEvent() const;
 		KAEvent::List     selectedEvents() const;
diff --git a/kalarm/functions.cpp b/kalarm/functions.cpp
index ab1b6f1..307539a 100644
--- a/kalarm/functions.cpp
+++ b/kalarm/functions.cpp
@@ -132,14 +132,9 @@ MainWindow* displayMainWindowSelected(const QString& eventID)
 	else
 	{
 		// There is already a main window, so make it the active window
-		bool visible = win->isVisible();
-		if (visible)
-			win->hide();        // in case it's on a different desktop
-		if (!visible  ||  win->isMinimized())
-		{
-			win->setWindowState(win->windowState() & ~Qt::WindowMinimized);
-			win->show();
-		}
+		win->hide();        // in case it's on a different desktop
+		win->setWindowState(win->windowState() & ~Qt::WindowMinimized);
+		win->show();
 		win->raise();
 		win->activateWindow();
 	}
diff --git a/kalarm/kalarm.h b/kalarm/kalarm.h
index 63333b2..e887eb3 100644
--- a/kalarm/kalarm.h
+++ b/kalarm/kalarm.h
@@ -23,7 +23,7 @@
 
 #undef QT3_SUPPORT
 
-#define KALARM_VERSION "2.4.11"
+#define KALARM_VERSION "2.4.12"
 #define KALARM_NAME "KAlarm"
 #define KALARM_DBUS_SERVICE  "org.kde.kalarm"  // D-Bus service name of KAlarm application
 
diff --git a/kalarm/mainwindow.cpp b/kalarm/mainwindow.cpp
index 95ce321..dd3b915 100644
--- a/kalarm/mainwindow.cpp
+++ b/kalarm/mainwindow.cpp
@@ -657,12 +657,7 @@ void MainWindow::updateKeepArchived(int days)
 void MainWindow::selectEvent(const QString& eventID)
 {
 	mListView->clearSelection();
-	QModelIndex index = EventListModel::alarms()->eventIndex(eventID);
-	if (index.isValid())
-	{
-		mListView->select(index);
-		mListView->scrollTo(index);
-	}
+	mListView->select(eventID, true);
 }
 
 /******************************************************************************
