--- trunk/KDE/kdepim/kmail/kmfilterdlg.cpp	2009/02/23 09:44:29	930380
+++ trunk/KDE/kdepim/kmail/kmfilterdlg.cpp	2009/02/23 09:45:34	930381
@@ -19,6 +19,7 @@
 */
 
 #include "kmfilterdlg.h"
+#include <klistwidgetsearchline.h>
 
 // other KMail headers:
 #include "kmsearchpatternedit.h"
@@ -610,6 +611,9 @@
   mListWidget->setMinimumWidth(150);
   mListWidget->setWhatsThis( i18n(_wt_filterlist) );
 
+  KListWidgetSearchLine* mSearchListWidget= new KListWidgetSearchLine( this, mListWidget );
+
+  layout->addWidget( mSearchListWidget );
   layout->addWidget( mListWidget );
 
   //----------- the first row of buttons
