Index: akregator/Messages.sh
===================================================================
--- akregator/Messages.sh	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ akregator/Messages.sh	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1,3 +1,3 @@
 #! /bin/sh
 $EXTRACTRC `find . -name "*.rc" -o -name "*.ui" -o -name "*.kcfg"` >> rc.cpp || exit 11
-$XGETTEXT $(find -name "*.cpp" ) -o $podir/akregator.pot
+$XGETTEXT $(find . -name "*.cpp" -o -name "*.h") -o $podir/akregator.pot
Index: akregator/src/subscriptionlistmodel.cpp
===================================================================
--- akregator/src/subscriptionlistmodel.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ akregator/src/subscriptionlistmodel.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -447,6 +447,9 @@
     {
         const TreeNode* const droppedOnNode = qobject_cast<const TreeNode*>( nodeForIndex( parent, m_feedList ) );
 
+        if ( !droppedOnNode )
+            return false;
+
         const Folder* const destFolder = droppedOnNode->isGroup() ? qobject_cast<const Folder*>( droppedOnNode ) : droppedOnNode->parent();
         if ( !destFolder )
             return false;
Index: libkleo/libkleopatrarc.desktop
===================================================================
--- libkleo/libkleopatrarc.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ libkleo/libkleopatrarc.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -288,6 +288,7 @@
 Name[it]=Chiavi per le firme qualificate
 Name[ja]=適格電子署名用の鍵
 Name[km]=សោ​សម្រាប់​ហត្ថលេខា​ដែល​មាន​​គុណភាព
+Name[lt]=Raktai kvalifikuotiems parašams
 Name[lv]=Derīgo parakstu atslēgas
 Name[nb]=Nøkler for kvalifiserte signaturer
 Name[nds]=Slötels för pröövt Ünnerschriften
Index: kode/kwsdl/kwsdl_compiler.desktop
===================================================================
--- kode/kwsdl/kwsdl_compiler.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kode/kwsdl/kwsdl_compiler.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -14,6 +14,7 @@
 Name[kk]=KWSDL компиляторы
 Name[km]=កម្មវិធី​ចងក្រង KWSDL
 Name[ko]=KWSDL 컴파일러
+Name[lt]=KWSDL kompiliatorius
 Name[lv]=KWSDL kompilators
 Name[nb]=KWSDL-kompilator
 Name[nds]=KWSDL-Kompilerer
Index: kode/kxforms/kxforms.desktop
===================================================================
--- kode/kxforms/kxforms.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kode/kxforms/kxforms.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -23,6 +23,7 @@
 GenericName[kk]=XML пішін өңдегіші
 GenericName[km]=កម្មវិធី​និពន្ធ​សំណុំ​បែបបទ​របស់ XML
 GenericName[ko]=XML 폼 편집기
+GenericName[lt]=XML formų rengyklė
 GenericName[lv]=XML formu redaktors
 GenericName[nb]=XML skjemaredigerer
 GenericName[nds]=XML-Kiekwarkeditor
Index: kresources/scalix/kabc/scalix.desktop
===================================================================
--- kresources/scalix/kabc/scalix.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kresources/scalix/kabc/scalix.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -44,6 +44,7 @@
 Comment[it]=Consente l'accesso ai contatti memorizzati su un server Scalix che usa IMAP tramite KMail o Kontact
 Comment[ja]=KMail または Kontact 経由で IMAP を使って、Scalix サーバに保存されている連絡先へのアクセスを提供します
 Comment[km]=ផ្ដល់​សិទ្ធិ​ចូល​ដំណើរការ​ទៅកាន់​ទំនាក់ទំនង​ដែល​បាន​រក្សាទុក​នៅ​លើ​ម៉ាស៊ីន​បម្រើ Scalix ដោយប្រើ IMAP តាមរយៈ KMail ឬ Kontact
+Comment[lt]=Suteikia prieigą prie kontaktų, saugomų Scalix serveryje, naudojant KMail (Kontact) programos IMAP protokolą
 Comment[lv]=Nodrošina piekļuvi kontaktiem, kas glabājas Scalix severī, izmantojot IMAP caur KMail vai Kontact
 Comment[nb]=Gir tilgang til en kalender lagret på en Scalix-tjener som bruker IMAP, via KMail eller Kontact
 Comment[nds]=Stellt mit KMail oder Kontact över IMAP Topgriep op Kontakten op en Scalix-Server praat
Index: kresources/slox/kabc_ox.desktop
===================================================================
--- kresources/slox/kabc_ox.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kresources/slox/kabc_ox.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -57,6 +57,7 @@
 Comment[it]=Consente l'accesso ai contatti memorizzati su un server Open-Xchange
 Comment[ja]=Open-Xchange サーバに保存されている連絡先へのアクセスを提供します
 Comment[km]=ផ្ដល់​សិទ្ធិ​ចូល​ដំណើរការ​ទៅកាន់​ទំនាក់ទំនង​ដែល​បាន​រក្សាទុក​នៅ​លើ​ម៉ាស៊ីន​បម្រើ Open-Xchange
+Comment[lt]=Suteikia prieigą prie kontaktų, saugomų Open-Xchange serveryje
 Comment[lv]=Nodrošina piekļuvi kontaktiem, kas glabājas Open-Xchange serverī.
 Comment[nb]=Gir tilgang til kontakter lagret på en Open-Xchange-tjener
 Comment[nds]=Stellt Togriep op de op en Open-Xchange-Server wohrten Kontakten praat
Index: kresources/akonadi/kabc/akonadi.desktop
===================================================================
--- kresources/akonadi/kabc/akonadi.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kresources/akonadi/kabc/akonadi.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -13,6 +13,7 @@
 Name[it]=Rubrica indirizzi di Akonadi
 Name[ja]=Akonadi アドレス帳
 Name[km]=សៀវភៅ​អាសយដ្ឋាន Akonadi
+Name[lt]=Akonadi adresų knygelės
 Name[lv]=Akonadi adrešu grāmatas
 Name[nb]=Akonadi adressebøker
 Name[nds]=Akonadi-Adressbook instellen
Index: kmail/dbusmail.desktop
===================================================================
--- kmail/dbusmail.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kmail/dbusmail.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -20,6 +20,7 @@
 Comment[kk]=DBUS интерфейсті пошта бағдарламасы
 Comment[km]=កម្មវិធី​សំបុត្រ​ដែល​មាន​ចំណុច​ប្រទាក់ DBUS
 Comment[ko]=DBUS 인터페이스를 사용하는 전자 우편 프로그램
+Comment[lt]=Pašto programa su D-Bus sąsaja
 Comment[lv]=Pasta programma ar DBUS saskarni
 Comment[nb]=E-postprogram med DBUS-grensesnitt
 Comment[nds]=Nettpostprogramm mit en D-Bus-Koppelsteed
Index: kmail/messagelistview/core/model.cpp
===================================================================
--- kmail/messagelistview/core/model.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kmail/messagelistview/core/model.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -2926,9 +2926,9 @@
       // Handle saving the current selection: if this item was the current before the step
       // then zero it out. We have killed it and it's OK for the current item to change.
 
-      if ( dyingMessage->isViewable() &&
-           dyingMessage == mCurrentItemToRestoreAfterViewItemJobStep )
+      if ( dyingMessage == mCurrentItemToRestoreAfterViewItemJobStep )
       {
+        Q_ASSERT( dyingMessage->isViewable() );
         // Try to select the item below the removed one as it helps in doing a "readon" of emails:
         // you read a message, decide to delete it and then go to the next.
         // Qt tends to select the message above the removed one instead (this is a hardcoded logic in
@@ -3028,12 +3028,21 @@
       // Parent is gone
       childMessage->setThreadingStatus( MessageItem::ParentMissing );
 
-      // If the child is going to be selected, we must immediately
-      // reattach it to a temporary group in order for the selection
-      // to be preserved across multiple steps. Otherwise we could end
+      // If the child (or any message in its subtree) is going to be selected,
+      // then we must immediately reattach it to a temporary group in order for the
+      // selection to be preserved across multiple steps. Otherwise we could end
       // with the child-to-be-selected being non viewable at the end
       // of the view job step. Attach to a temporary group.
-      if ( childMessage == mCurrentItemToRestoreAfterViewItemJobStep )
+      if (
+           // child is going to be re-selected
+           ( childMessage == mCurrentItemToRestoreAfterViewItemJobStep ) ||
+           (
+             // there is a message that is going to be re-selected
+             mCurrentItemToRestoreAfterViewItemJobStep && 
+             // that message is in the childMessage subtree
+             mCurrentItemToRestoreAfterViewItemJobStep->hasAncestor( childMessage )
+           )
+         )
       {
         attachMessageToGroupHeader( childMessage );
 
Index: kmail/kmail_config_composer.desktop
===================================================================
--- kmail/kmail_config_composer.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kmail/kmail_config_composer.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -80,6 +80,7 @@
 Comment[ja]=メール作成の設定
 Comment[km]=ការ​កំណត់​កម្មវិធី​តែង​សារ
 Comment[ko]=메시지 작성기 설정
+Comment[lt]=Laiškų rengyklės nustatymai
 Comment[lv]=Vēstuļu rastītāja iestatījumi
 Comment[nb]=Innstillinger for meldingskomposer
 Comment[nds]=Narichteneditor instellen
Index: kmail/dbusimap.desktop
===================================================================
--- kmail/dbusimap.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kmail/dbusimap.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -20,6 +20,7 @@
 Comment[kk]=DBUS интерфейсті пошта бағдарламасы
 Comment[km]=កម្មវិធី​សំបុត្រ​ដែល​មាន​ចំណុច​ប្រទាក់ DBUS
 Comment[ko]=DBUS 인터페이스를 사용하는 전자 우편 프로그램
+Comment[lt]=Pašto programa su D-Bus sąsaja
 Comment[lv]=Pasta programma ar DBUS saskarni
 Comment[nb]=E-postprogram med DBUS-grensesnitt
 Comment[nds]=Nettpostprogramm mit en D-Bus-Koppelsteed
Index: kmail/kmfoldercachedimap.cpp
===================================================================
--- kmail/kmfoldercachedimap.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kmail/kmfoldercachedimap.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -202,7 +202,8 @@
     /*mHoldSyncs( false ),*/
     mFolderRemoved( false ),
     mRecurse( true ),
-    mStatusChangedLocally( false ), mAnnotationFolderTypeChanged( false ),
+    mStatusChangedLocally( false ),
+    mAnnotationFolderTypeChanged( false ),
     mIncidencesForChanged( false ), mPersonalNamespacesCheckDone( true ),
     mQuotaInfo(), mAlarmsBlocked( false ),
     mRescueCommandCount( 0 ),
@@ -303,18 +304,22 @@
   KMFolderMaildir::readConfig();
 
   mStatusChangedLocally = group.readEntry( "StatusChangedLocally", false );
+  QStringList uidsChanged = group.readEntry( "UIDStatusChangedLocally", QStringList() );
+  foreach( const QString &uid, uidsChanged ) {
+    mUIDsOfLocallyChangedStatuses.append( uid.toUInt() );
+  }
   mAnnotationFolderTypeChanged = group.readEntry( "AnnotationFolderTypeChanged", false );
   mIncidencesForChanged = group.readEntry( "IncidencesForChanged", false );
   if ( mImapPath.isEmpty() ) {
     mImapPathCreation = group.readEntry("ImapPathCreation");
   }
 
-  QStringList uids = group.readEntry( "UIDSDeletedSinceLastSync", QStringList() );
+  QStringList delUids = group.readEntry( "UIDSDeletedSinceLastSync" , QStringList() );
 #if MAIL_LOSS_DEBUGGING
   kDebug( 5006 ) << "READING IN UIDSDeletedSinceLastSync: " << folder()->prettyUrl() << endl << uids;
 #endif
-  for ( QStringList::iterator it = uids.begin(); it != uids.end(); ++it ) {
-      mDeletedUIDsSinceLastSync.insert( (*it).toULong(), 0);
+  for ( QStringList::iterator it = delUids.begin(); it != delUids.end(); it++ ) {
+    mDeletedUIDsSinceLastSync.insert( (*it).toULong(), 0);
   }
 }
 
@@ -331,7 +336,14 @@
   configGroup.writeEntry( "NoContent", mNoContent );
   configGroup.writeEntry( "ReadOnly", mReadOnly );
   configGroup.writeEntry( "FolderAttributes", mFolderAttributes );
-  configGroup.writeEntry( "StatusChangedLocally", mStatusChangedLocally );
+
+  // StatusChangedLocally is always false, as we use UIDStatusChangedLocally now
+  configGroup.writeEntry( "StatusChangedLocally", false );
+  QStringList uidsToWrite;
+  foreach( const ulong uid, mUIDsOfLocallyChangedStatuses ) {
+    uidsToWrite.append( QString::number( uid ) );
+  }
+  configGroup.writeEntry( "UIDStatusChangedLocally", uidsToWrite );
   if ( !mImapPathCreation.isEmpty() ) {
     if ( mImapPath.isEmpty() ) {
       configGroup.writeEntry( "ImapPathCreation", mImapPathCreation );
@@ -953,12 +965,13 @@
       // Upload flags, unless we know from the ACL that we're not allowed
       // to do that or they did not change locally
       if ( mUserRights <= 0 || ( mUserRights & KMail::ACLJobs::WriteFlags ) ) {
-        if ( mStatusChangedLocally ) {
+        if ( !mUIDsOfLocallyChangedStatuses.isEmpty() || mStatusChangedLocally ) {
           uploadFlags();
           break;
         }
-      } else if ( mUserRights & KMail::ACLJobs::WriteSeenFlag ) {
-        if ( mStatusChangedLocally ) {
+      }
+      else if ( mUserRights & KMail::ACLJobs::WriteSeenFlag ) {
+        if ( !mUIDsOfLocallyChangedStatuses.isEmpty() || mStatusChangedLocally ) {
           uploadSeenFlags();
           break;
         }
@@ -1425,6 +1438,10 @@
         // Either not a valid message or not one that is on the server yet
         continue;
       }
+      if ( mUIDsOfLocallyChangedStatuses.indexOf( msg->UID() ) < 0 && !mStatusChangedLocally ) {
+        // This message has not had its status changed locally
+        continue;
+      }
 
       QString flags = KMFolderImap::statusToFlags( msg->status(), mPermanentFlags );
       // Collect uids for each typem of flags.
@@ -1468,6 +1485,11 @@
         // Either not a valid message or not one that is on the server yet
         continue;
 
+      if ( mUIDsOfLocallyChangedStatuses.indexOf( msg->UID() ) < 0 && !mStatusChangedLocally ) {
+        // This message has not had its status changed locally
+        continue;
+      }
+
       if ( msg->status().isOld() || msg->status().isRead() )
         seenUids.append( msg->UID() );
       else
@@ -1525,13 +1547,21 @@
 void KMFolderCachedImap::setStatus( int idx, const MessageStatus &status, bool toggle )
 {
   KMFolderMaildir::setStatus( idx, status, toggle );
-  mStatusChangedLocally = true;
+  const KMMsgBase *msg = getMsgBase( idx );
+  Q_ASSERT( msg );
+  if ( msg )
+    mUIDsOfLocallyChangedStatuses.append( msg->UID() );
 }
 
 void KMFolderCachedImap::setStatus( QList<int> &ids, const MessageStatus &status, bool toggle )
 {
   KMFolderMaildir::setStatus( ids, status, toggle );
-  mStatusChangedLocally = true;
+  foreach( const int id, ids ) {
+    const KMMsgBase *msg = getMsgBase( id );
+    Q_ASSERT( msg );
+    if ( msg )
+      mUIDsOfLocallyChangedStatuses.append( msg->UID() );
+  }
 }
 
 /* Upload new folders to server */
@@ -1888,7 +1918,8 @@
   } else {
     if ( lastSet ) { // always true here (this comes from online-imap...)
       mContentState = imapFinished;
-      mStatusChangedLocally = false; // we are up to date again
+      mUIDsOfLocallyChangedStatuses.clear(); // we are up to date again
+      mStatusChangedLocally = false;
     }
   }
   serverSyncInternal();
Index: kmail/kmfoldercachedimap.h
===================================================================
--- kmail/kmfoldercachedimap.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kmail/kmfoldercachedimap.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -627,12 +627,18 @@
     bool mFolderRemoved;
     bool mRecurse;
 
+   /**
+    * UIDs added by setStatus. Indicates that the client has changed
+    * the status of those mails. The mail flags for changed mails will be
+    * uploaded to the server, overwriting the server's notion of the status
+    * of the mails in this folder.
+    */
+    QList<ulong> mUIDsOfLocallyChangedStatuses;
+
     /**
-      Set to true by setStatus. Indicates that the client has changed
-      the status of at least one mail. The mail flags will therefore be
-      uploaded to the server, overwriting the server's notion of the status
-      of the mails in this folder.
-    */
+     * Same as above, but uploads the flags of all mails, even if not all changed.
+     * Only still here for config compatibility.
+     */
     bool mStatusChangedLocally;
 
     /**
Index: kmail/kmail_view.desktop
===================================================================
--- kmail/kmail_view.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kmail/kmail_view.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -13,6 +13,7 @@
 Name[ja]=KMail ビュー
 Name[km]=ទិដ្ឋភាព KMail
 Name[ko]=KMail 보기
+Name[lt]=KMail peržiūra
 Name[lv]=KMail skats
 Name[nb]=KMail-visning
 Name[nds]=KMail-Ansicht
Index: knode/kncomposerview.cpp
===================================================================
--- knode/kncomposerview.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ knode/kncomposerview.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -93,6 +93,7 @@
   e_dit=new KNComposerEditor(this);
   e_dit->switchToPlainText();
   e_dit->setMinimumHeight(50);
+  e_dit->setFont(KGlobalSettings::fixedFont());
 
   QVBoxLayout *notL=new QVBoxLayout(e_dit);
   notL->addStretch(1);
Index: dependency-graph.dot
===================================================================
--- dependency-graph.dot	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ dependency-graph.dot	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -48,7 +48,6 @@
         "konsolekalendar"[style=filled, fillcolor=lightcyan, URL="konsolekalendar/html/index.html"];
         "kontact"[style=filled, fillcolor=lightcyan, URL="kontact/html/index.html"];
         "korganizer"[style=filled, fillcolor=lightcyan, URL="korganizer/html/index.html"];
-        "korn"[style=filled, fillcolor=lightcyan];
         "kresources"[style=filled, fillcolor=palegreen];
         "libakonadi"[style=filled, fillcolor=lightyellow, URL="akonadi/libakonadi/html/index.html"];
         "libkdepim"[style=filled, fillcolor=lightyellow, URL="libkdepim/html/index.html"];
Index: kalarm/kalarm.h
===================================================================
--- kalarm/kalarm.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kalarm/kalarm.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -23,7 +23,7 @@
 
 #undef QT3_SUPPORT
 
-#define KALARM_VERSION "2.1.5"
+#define KALARM_VERSION "2.1.6"
 #define KALARM_NAME "KAlarm"
 #define KALARM_DBUS_SERVICE  "org.kde.kalarm"  // D-Bus service name of KAlarm application
 
Index: kalarm/mainwindow.cpp
===================================================================
--- kalarm/mainwindow.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kalarm/mainwindow.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -51,6 +51,7 @@
 #include <QDropEvent>
 #include <QResizeEvent>
 #include <QCloseEvent>
+#include <QTimer>
 
 #include <kmenubar.h>
 #include <ktoolbar.h>
@@ -518,7 +519,10 @@
 	mActionFindPrev = KStandardAction::findPrev(mListView, SLOT(slotFindPrev()), actions);
 	KStandardAction::selectAll(mListView, SLOT(selectAll()), actions);
 	KStandardAction::deselect(mListView, SLOT(clearSelection()), actions);
-	KStandardAction::quit(this, SLOT(slotQuit()), actions);
+	// Quit only once the event loop is called; otherwise, the parent window will
+	// be deleted while still processing the action, resulting in a crash.
+	KAction* act = KStandardAction::quit(0, 0, actions);
+	connect(act, SIGNAL(triggered(bool)), SLOT(slotQuit()), Qt::QueuedConnection);
 	KStandardAction::keyBindings(this, SLOT(slotConfigureKeys()), actions);
 	KStandardAction::configureToolbars(this, SLOT(slotConfigureToolbar()), actions);
 	KStandardAction::preferences(this, SLOT(slotPreferences()), actions);
@@ -1069,7 +1073,10 @@
 }
 
 /******************************************************************************
-*  Called when the Quit menu item is selected.
+* Called when the Quit menu item is selected.
+* Note that this must be called by the event loop, not directly from the menu
+* item, since otherwise the window will be deleted while still processing the
+* menu, resulting in a crash.
 */
 void MainWindow::slotQuit()
 {
@@ -1097,7 +1104,9 @@
 		{
 			// There is no system tray icon, and this is the last main
 			// window, so closing it will quit the application.
-			theApp()->doQuit(this);
+			// Quit once the close event has been processed. Deleting the
+			// window while still processing the event handler can cause a crash.
+			QTimer::singleShot(0, this, SLOT(slotDoQuit()));
 			ce->ignore();
 			return;
 		}
Index: kalarm/Changelog
===================================================================
--- kalarm/Changelog	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kalarm/Changelog	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1,6 +1,9 @@
 KAlarm Change Log
 
-=== Version 2.1.5 --- 7 February 2009 ===
+=== Version 2.1.6 --- 4 March 2009 ===
+Fix crash when KAlarm quits.
+
+=== Version 2.1.5 (KDE 4.2.1) --- 7 February 2009 ===
 - Disable inapplicable alarm types in alarm edit dialogue Load Template list.
 - Prevent multiple identical error messages being displayed for the same alarm.
 - Fix possible crash on alarm refresh, or removal or disabling of a resource.
Index: kalarm/kalarmapp.cpp
===================================================================
--- kalarm/kalarmapp.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kalarm/kalarmapp.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1021,7 +1021,6 @@
 void KAlarmApp::removeWindow(TrayWindow*)
 {
 	mTrayWindow = 0;
-	quitIf();
 }
 
 /******************************************************************************
Index: kalarm/resources/kalarm_manager.desktop
===================================================================
--- kalarm/resources/kalarm_manager.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kalarm/resources/kalarm_manager.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -18,6 +18,7 @@
 Name[kk]=Ескертулер
 Name[km]=ជូន​ដំណឹង
 Name[ko]=알람
+Name[lt]=Priminimai
 Name[lv]=Atgādinājumi
 Name[nb]=Varslinger
 Name[ne]=संसूचक
Index: kalarm/traywindow.h
===================================================================
--- kalarm/traywindow.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kalarm/traywindow.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1,7 +1,7 @@
 /*
  *  traywindow.h  -  the KDE system tray applet
  *  Program:  kalarm
- *  Copyright © 2002-2008 by David Jarvie <djarvie@kde.org>
+ *  Copyright © 2002-2009 by David Jarvie <djarvie@kde.org>
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -45,9 +45,6 @@
 		MainWindow*  assocMainWindow() const               { return mAssocMainWindow; }
 		void         setAssocMainWindow(MainWindow* win)   { mAssocMainWindow = win; }
 
-	public slots:
-		void         slotQuit();
-
 	signals:
 		void         deleted();
 
@@ -63,6 +60,7 @@
 		void         slotPreferences();
 		void         setEnabledStatus(bool status);
 		void         slotResourceStatusChanged();
+		void         slotQuit();
 
 	private:
 		QString      tooltipAlarmText() const;
Index: kalarm/traywindow.cpp
===================================================================
--- kalarm/traywindow.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kalarm/traywindow.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -39,6 +39,7 @@
 #include <QToolTip>
 #include <QMouseEvent>
 #include <QList>
+#include <QTimer>
 
 #include <kactioncollection.h>
 #include <ktoggleaction.h>
@@ -108,7 +109,10 @@
 	// Replace the default handler for the Quit context menu item
 	const char* quitName = KStandardAction::name(KStandardAction::Quit);
 	delete actions->action(quitName);
-	KStandardAction::quit(this, SLOT(slotQuit()), actions);
+	// Quit only once the event loop is called; otherwise, the parent tray icon
+	// will be deleted while still processing the action, resulting in a crash.
+	a = KStandardAction::quit(0, 0, actions);
+	connect(a, SIGNAL(triggered(bool)), SLOT(slotQuit()), Qt::QueuedConnection);
 
 	// Set icon to correspond with the alarms enabled menu status
 	setEnabledStatus(theApp()->alarmsEnabled());
@@ -163,6 +167,9 @@
 
 /******************************************************************************
 * Called when the Quit context menu item is selected.
+* Note that this must be called by the event loop, not directly from the menu
+* item, since otherwise the tray icon will be deleted while still processing
+* the menu, resulting in a crash.
 */
 void TrayWindow::slotQuit()
 {
Index: Mainpage.dox
===================================================================
--- Mainpage.dox	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ Mainpage.dox	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -30,8 +30,6 @@
         * - <a href="kmail/html/index.html"><b>kmail</b></a>
         *   (<a href="kmail/html/classes.html">classes</a>)\n
         *   <i>A fully-featured email client.</i>
-        * - <a href="kmobiletools/html/index.html"><b>kmobiletools</b></a>
-        *   (<a href="kmobiletools/html/classes.html">classes</a>)\n
         *   <i>An application specially designed to manage mobile phones.</i>
         * - <a href="knode/html/index.html"><b>knode</b></a>
         *   (<a href="knode/html/classes.html">classes</a>)\n
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ CMakeLists.txt	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -89,15 +89,6 @@
     macro_log_feature(X11_Xscreensaver_FOUND "X11 screensaver extension" "X Window Screen Saver Development Library" "http://xorg.freedesktop.org" FALSE "" "Required for building the KTimeTracker Program")
   endif(X11_FOUND)
 
-# uncomment the following block for 4.3 in order to enable kitchensync
-#  if (NOT WIN32)
-#     set(OPENSYNC_MIN_VERSION 0.38)
-#     macro_optional_find_package(OpenSync)
-#     macro_log_feature(OPENSYNC_FOUND "opensync" "OpenSync Development Libraries" "http://www.opensync.org" FALSE "0.38 or greater" "Needed to provide syncing functionality in KDE PIM applications. Necessary to compile kitchensync.")
-#     macro_optional_find_package(GLIB2)
-#     macro_log_feature(GLIB2_FOUND "glib2" "Low-level core library for data structure handling, portability wrappers, etc." "http://www.gtk.org" FALSE "2.x" "Needed to provide synching applications from KDE PIM applications. Necessary to compile kitchensync and other PIM applications.")
-#  endif (NOT WIN32)
-
   macro_optional_find_package(GNOKII)
   macro_bool_to_01(GNOKII_FOUND HAVE_GNOKII_H)
   macro_log_feature(GNOKII_FOUND "gnokii" "Gnokii Development Libraries" "http://www.gnokii.org" FALSE "" "Needed to provide the ability to import and export KDE addressbook entries to/from mobile phones that are accessible via gnokii")
@@ -219,43 +210,15 @@
   endif(KMAIL_SUPPORTED)
 
   if(NOT WIN32)
-#enable kmobiletools 4.3 if there is interest
-#     set(KMOBILETOOLS_SUPPORTED TRUE)
-#     add_definitions(-DKMOBILETOOLS_SUPPORTED)
-#     message(STATUS "KMobileTools support enabled.")
-#     macro_optional_add_subdirectory(kmobiletools)
-      set(KMOBILETOOLS_SUPPORTED FALSE)
-
      set(KPILOT_SUPPORTED TRUE)
      add_definitions(-DKPILOT_SUPPORTED)
      message(STATUS "KPilot support enabled.")
      macro_optional_add_subdirectory(kpilot)
-
-     if(OPENSYNC_FOUND AND GLIB2_FOUND)
-       set(KITCHENSYNC_SUPPORTED TRUE)
-       add_definitions(-DKITCHENSYNC_SUPPORTED)
-       message(STATUS "KitchenSync support enabled.")
-       macro_optional_add_subdirectory(kitchensync)
-     else(OPENSYNC_FOUND AND GLIB2_FOUND)
-       set(KITCHENSYNC_SUPPORTED FALSE)
-       if(NOT OPENSYNC_FOUND)
-         message(STATUS "KitchenSync support disabled (OpenSync not found).")
-       endif(NOT OPENSYNC_FOUND)
-       if(NOT GLIB2_FOUND)
-         message(STATUS "KitchenSync support disabled (GLIB not found).")
-       endif(NOT GLIB2_FOUND)
-     endif(OPENSYNC_FOUND AND GLIB2_FOUND)
   else(NOT WIN32)
-     set(KMOBILETOOLS_SUPPORTED FALSE)
-     message(STATUS "KMobileTools support DISABLED (Windows target).")
      set(KPILOT_SUPPORTED FALSE)
      message(STATUS "KPilot support DISABLED (Windows target).")
-     set(KITCHENSYNC_SUPPORTED FALSE)
-     message(STATUS "KitchenSync support DISABLED (Windows target).")
   endif(NOT WIN32)
 
-#enable korn in 4.3 if there is interest
-# macro_optional_add_subdirectory(korn)
   add_subdirectory(kresources)
   macro_optional_add_subdirectory(kontact)
 
Index: kjots/Kjots.desktop
===================================================================
--- kjots/Kjots.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kjots/Kjots.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -20,7 +20,7 @@
 GenericName[ja]=メモ帳
 GenericName[km]=កម្មវិធី​ទទួល​ចំណាំ
 GenericName[ko]=메모장
-GenericName[lt]=Pastabų gavėjas
+GenericName[lt]=Konspektavimo įrankis
 GenericName[lv]=Piezīmju blociņš
 GenericName[nb]=Notisblokk
 GenericName[nds]=Notizenprogramm
Index: kjots/kjotspart.desktop
===================================================================
--- kjots/kjotspart.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kjots/kjotspart.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Name=KJotsPart
+Name[lt]=KJots dalis
 Name[nn]=KJots-del
 Name[pa]=ਕੇਜੋਟਸਪਾਰਟ
 Name[pl]=Moduł Notatek
Index: README
===================================================================
--- README	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ README	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -22,7 +22,6 @@
 * kaddressbook: The KDE addressbook application.
 * kmail: the KDE mail client
 * kmailcvt: tool for importing mail related data from other programs
-* kmobiletools: an application specially designed to manage mobile phones
 * knode: UseNet client
 * korganizer: a calendar-of-events, todo-list manager, journal 
 * kalarm: gui for setting up personal alarm messages, emails and commands
@@ -33,7 +32,6 @@
 * kfile-plugins: vCard KFIleItem plugin.
 * knotes: yellow notes application
 * konsolekalendar: Command line tool for accessing calendar files.
-* korn: mail checker
 * akregator: Feed reader
 * kjots: manager for several "books" with a subject and notes
 
Index: kpilot/Messages.sh
===================================================================
--- kpilot/Messages.sh	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kpilot/Messages.sh	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1,2 +1,3 @@
 #! /usr/bin/env bash
+$EXTRACTRC `find . -name \*.rc -o -name \*.ui -o -name \*.kcfg` >> rc.cpp
 $XGETTEXT `find . -name \*.cc -o -name \*.cpp -o -name \*.h` -o $podir/kpilot.pot
Index: kpilot/conduits/todo/kpilot-conduit-todo.desktop
===================================================================
--- kpilot/conduits/todo/kpilot-conduit-todo.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kpilot/conduits/todo/kpilot-conduit-todo.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -42,6 +42,7 @@
 Name[ja]=To-Do
 Name[km]=​ការងារ​ត្រូវ​ធ្វើ
 Name[ko]=할 일
+Name[lt]=Darbas
 Name[lv]=Darbi
 Name[nb]=Gjøremål
 Name[nds]=Opgaav
Index: kpilot/conduits/calendar/calendarconduit.cc
===================================================================
--- kpilot/conduits/calendar/calendarconduit.cc	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kpilot/conduits/calendar/calendarconduit.cc	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -205,6 +205,11 @@
 			}
 			else
 			{
+				// Events that end at midnight cause problems
+				// (http://bugs.kde.org/show_bug.cgi?id=183631); however, events with
+				// no time associated with them are stored as beginning and ending
+				// at midnight on the same day, and these are OK.  Other events that end at
+				// midnight will need to be adjusted.
 				QDateTime hhDtEnd = hhEntry.dtEnd();
 				QDateTime hhDtStart = hhEntry.dtStart();
 				QDateTime pcDtEnd = pcEvent->dtEnd().dateTime().toLocalTime();
Index: kontact/plugins/akregator/akregator.setdlg
===================================================================
--- kontact/plugins/akregator/akregator.setdlg	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/akregator/akregator.setdlg	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -66,6 +66,7 @@
 Comment[ja]=フィードリーダー
 Comment[km]=កម្មវិធី​អាន​មតិព័ត៌មាន
 Comment[ko]=피드 리더
+Comment[lt]=Kanalų skaitytuvė
 Comment[lv]=Ziņu lasītājs
 Comment[nb]=Kildeleser
 Comment[nds]=Mellenstroom-Kieker
Index: kontact/plugins/planner/planner.setdlg
===================================================================
--- kontact/plugins/planner/planner.setdlg	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/planner/planner.setdlg	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -12,6 +12,7 @@
 Name[ja]=スケジュール帳
 Name[km]=កម្មវិធីធ្វើ​គម្រោង
 Name[ko]=플래너
+Name[lt]=Planuoklis
 Name[lv]=Plānotājs
 Name[nb]=Planlegger
 Name[nn]=Planleggjar
@@ -41,6 +42,7 @@
 Comment[ja]=スケジュール帳の要約
 Comment[km]=សេចក្ដី​សង្ខេប​របស់​កម្មវិធីធ្វើ​គម្រោង
 Comment[ko]=플래너 요약
+Comment[lt]=Planuoklio santrauka
 Comment[lv]=Plānotāja kopsavilkums
 Comment[nb]=Planlegger-aammendrag
 Comment[nds]=Planer-Översicht
Index: kontact/plugins/planner/kcmplanner.desktop
===================================================================
--- kontact/plugins/planner/kcmplanner.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/planner/kcmplanner.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -24,6 +24,7 @@
 Name[ja]=スケジュール帳
 Name[km]=កម្មវិធីធ្វើ​គម្រោង
 Name[ko]=플래너
+Name[lt]=Planuoklis
 Name[lv]=Plānotājs
 Name[nb]=Planlegger
 Name[nn]=Planleggjar
@@ -53,6 +54,7 @@
 Comment[ja]=スケジュール帳の設定
 Comment[km]=កា​ររៀបចំ​កម្មវិធី​ធ្វើ​គម្រោង
 Comment[ko]=플래너 설정
+Comment[lt]=Planuoklio nustatymai
 Comment[lv]=Plānotāja iestatījumi
 Comment[nb]=Planlegger-oppsett
 Comment[nds]=Planner inrichten
Index: kontact/plugins/planner/plannerplugin.desktop
===================================================================
--- kontact/plugins/planner/plannerplugin.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/planner/plannerplugin.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -26,6 +26,7 @@
 Name[ja]=スケジュール帳
 Name[km]=កម្មវិធីធ្វើ​គម្រោង
 Name[ko]=플래너
+Name[lt]=Planuoklis
 Name[lv]=Plānotājs
 Name[nb]=Planlegger
 Name[nn]=Planleggjar
@@ -55,6 +56,7 @@
 Comment[ja]=スケジュール帳プラグイン
 Comment[km]=កម្មវិធី​ជំនួយ​​របស់​កម្មវិធី​ធ្វើ​គម្រោង
 Comment[ko]=플래너 플러그인
+Comment[lt]=Planuoklio priedas
 Comment[lv]=Plānotāja spraudnis
 Comment[nb]=Planlegger-programtillegg
 Comment[nds]=Planner-Moduul
Index: kontact/plugins/knode/knode.setdlg
===================================================================
--- kontact/plugins/knode/knode.setdlg	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/knode/knode.setdlg	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -24,6 +24,7 @@
 Comment[kk]=Пошта компоненті
 Comment[km]=មាសភាគ​របស់ Usenet
 Comment[ko]=유즈넷 구성 요소
+Comment[lt]=Usenet komponentas
 Comment[lv]=Usenet komponente
 Comment[nb]=Usenet-komponent
 Comment[nds]=Usenet-Komponent
Index: kontact/plugins/specialdates/kcmsdsummary.desktop
===================================================================
--- kontact/plugins/specialdates/kcmsdsummary.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/specialdates/kcmsdsummary.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -30,6 +30,7 @@
 Name[kk]=Болашақ ерекше күндер
 Name[km]=កាលបរិច្ឆេទ​ពិសេស​ដែលនឹង​កើត​ឡើង
 Name[ko]=다가오는 기념일
+Name[lt]=Artėjančios ypatingos datos
 Name[lv]=Gaidāmie īpašie datumi
 Name[nb]=Kommende spesielle datoer
 Name[nds]=Anstahn besünner Daag
@@ -68,6 +69,7 @@
 Comment[kk]=Болашақ ерекше күндер тұжырымының баптау
 Comment[km]=ការ​រៀបចំ​សេចក្ដី​សង្ខេប​កាលបរិច្ឆេទ​ពិសេស​ដែល​នឹង​កើតឡើង
 Comment[ko]=다가오는 기념일 요약 설정
+Comment[lt]=Artėjančių ypatingų datų santraukos sąranka
 Comment[lv]=Gaidāmo īpašo datumu kopsavilkuma iestatījumi
 Comment[nb]=Sammendragsoppsett for kommende spesielle datoer
 Comment[nds]=De Översicht för anstahn besünner Daten instellen
Index: kontact/plugins/kmobiletools/kmobiletools.setdlg
===================================================================
--- kontact/plugins/kmobiletools/kmobiletools.setdlg	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/kmobiletools/kmobiletools.setdlg	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -16,6 +16,7 @@
 Name[ja]=携帯電話
 Name[km]=ទូរស័ព្ទ​ចល័ត
 Name[ko]=휴대폰
+Name[lt]=Mobilieji telefonai
 Name[lv]=Mobilie telefoni
 Name[nb]=Mobiltelefoner
 Name[nds]=Mobiltelefonen
@@ -50,6 +51,7 @@
 Comment[ja]=携帯電話コンポーネント
 Comment[km]=សមាសភាគ​ទូរស័ព្ទ​ចល័ត
 Comment[ko]=휴대폰 구성 요소
+Comment[lt]=Mobilaus telefono komponentas
 Comment[lv]=Mobilo telefonu komponente
 Comment[nb]=Mobiltelefonkomponent
 Comment[nds]=Mobiltelefoon-Komponent
Index: kontact/plugins/kmobiletools/kmobiletools.desktop
===================================================================
--- kontact/plugins/kmobiletools/kmobiletools.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/kmobiletools/kmobiletools.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -27,6 +27,7 @@
 Comment[ja]=KMobileTools プラグイン
 Comment[km]=កម្មវិធី​ជំនួយ​របស់ KMmobileTools
 Comment[ko]=KMobileTools 플러그인
+Comment[lt]=KMobileTools priedas
 Comment[lv]=KMobileTools spraudnis
 Comment[nb]=KMobileTools programtillegg
 Comment[nds]=KMobileTools-Moduul
@@ -60,6 +61,7 @@
 Name[ja]=携帯電話
 Name[km]=ទូរស័ព្ទ​ចល័ត
 Name[ko]=휴대폰
+Name[lt]=Mobilieji telefonai
 Name[lv]=Mobilie telefoni
 Name[nb]=Mobiltelefoner
 Name[nds]=Mobiltelefonen
Index: kontact/plugins/knotes/knotes.setdlg
===================================================================
--- kontact/plugins/knotes/knotes.setdlg	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/knotes/knotes.setdlg	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -14,6 +14,7 @@
 Name[ja]=ポップアップメモ
 Name[km]=ចំណាំ​លេច​ឡើង
 Name[ko]=팝업 노트
+Name[lt]=Lipnūs lapeliai
 Name[lv]=Piezīmju lapiņas
 Name[nb]=Sprettopp-notater
 Name[nds]=Opduk-Notizen
@@ -48,6 +49,7 @@
 Comment[ja]=ポップアップメモコンポーネント
 Comment[km]=សមាសភាគ​ចំណាំលេចឡើង​
 Comment[ko]=팝업 노트 구성 요소
+Comment[lt]=Lipnių lapelių komponentas
 Comment[lv]=Piezīmju lapiņu komponente
 Comment[nb]=Komponent for sprettoppnotater
 Comment[nds]=Opduknotiz-Komponent
Index: kontact/plugins/knotes/knotesplugin.desktop
===================================================================
--- kontact/plugins/knotes/knotesplugin.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/knotes/knotesplugin.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -84,6 +84,7 @@
 Name[ja]=ポップアップメモ
 Name[km]=ចំណាំ​លេច​ឡើង
 Name[ko]=팝업 노트
+Name[lt]=Lipnūs lapeliai
 Name[lv]=Piezīmju lapiņas
 Name[nb]=Sprettopp-notater
 Name[nds]=Opduk-Notizen
Index: kontact/plugins/knotes/knotes_part_p.h
===================================================================
--- kontact/plugins/knotes/knotes_part_p.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/knotes/knotes_part_p.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -155,7 +155,7 @@
       hbl->addWidget( mTitleEdit, 1, Qt::AlignVCenter );
 
       mNoteEdit = new KNoteEdit( actionCollection(), page );
-      ( (KTextEdit*)mNoteEdit )->setTextFormat( Qt::RichText );
+      ( (KTextEdit*)mNoteEdit )->setAcceptRichText( true );
       mNoteEdit->setFocus();
 
       KXMLGUIBuilder builder( page );
@@ -163,7 +163,8 @@
       factory.addClient( this );
 
       mTool = static_cast<KToolBar *>( factory.container( "note_tool", this ) );
-
+      mNoteEdit->setContextMenu( dynamic_cast<KMenu *>(
+                                   factory.container( "note_edit", this ) ) );
       layout->addWidget( mTool );
       layout->addWidget( mNoteEdit );
 
Index: kontact/plugins/knotes/knotetip.cpp
===================================================================
--- kontact/plugins/knotes/knotetip.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/knotes/knotetip.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -31,6 +31,7 @@
 */
 
 #include "knotetip.h"
+#include <kdebug.h>
 #include "knotes_part_p.h"
 
 #include <kglobalsettings.h>
@@ -89,11 +90,7 @@
     }
   } else {
     KCal::Journal *journal = item->journal();
-    if ( journal->customProperty( "KNotes", "RichText" ) == "true" ) {
-      mPreview->setTextFormat( Qt::RichText );
-    } else {
-      mPreview->setTextFormat( Qt::PlainText );
-    }
+    mPreview->setAcceptRichText( journal->customProperty( "KNotes", "RichText" ) == "true" );
 
     QColor fg( journal->customProperty( "KNotes", "FgColor" ) );
     QColor bg( journal->customProperty( "KNotes", "BgColor" ) );
@@ -103,12 +100,12 @@
     //mPreview->zoomTo( 8 );
     mPreview->sync();
 
-    int w = 400;
-    int h = mPreview->heightForWidth( w );
+    mPreview->document()->adjustSize ();
+    int w = mPreview->document () ->size().width();
+    int h = mPreview->document () ->size().height();
     while ( w > 60 && h == mPreview->heightForWidth( w - 20 ) ) {
       w -= 20;
     }
-
     QRect desk = KGlobalSettings::desktopGeometry( mNoteIVI->rect().center() );
     resize( w, qMin( h, desk.height() / 2 - 20 ) );
 
Index: kontact/plugins/ktimetracker/ktimetracker.setdlg
===================================================================
--- kontact/plugins/ktimetracker/ktimetracker.setdlg	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/ktimetracker/ktimetracker.setdlg	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -14,6 +14,7 @@
 Name[ja]=タイムトラッカー
 Name[km]=កម្មវិធី​តាមដាន​ពេលវេលា
 Name[ko]=시간 추적기
+Name[lt]=Laiko sekiklis
 Name[lv]=Patērētā laika uzskatītājs
 Name[nb]=TimeTracker
 Name[nds]=Tietlogbook
@@ -45,6 +46,7 @@
 Comment[ja]=タイムトラッカーコンポーネント
 Comment[km]=សមាសភាគ​កម្មវិធី​តាមដាន​ពេលវេលា
 Comment[ko]=시간 추적기 구성 요소
+Comment[lt]=Laiko sekiklio komponentas
 Comment[lv]=Laika uzskaitītāja komponente
 Comment[nb]=Time Traacker-komponent
 Comment[nds]=Tietlogbook-Komponent
Index: kontact/plugins/ktimetracker/ktimetracker_plugin.desktop
===================================================================
--- kontact/plugins/ktimetracker/ktimetracker_plugin.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/ktimetracker/ktimetracker_plugin.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -29,6 +29,7 @@
 Comment[ja]=タイムトラッカープラグイン
 Comment[km]=កម្មវិធី​ជំនួយ TimeTracker
 Comment[ko]=시간 추적기 플러그인
+Comment[lt]=Laiko sekiklio priedas
 Comment[lv]=KTimeTracker spraudnis
 Comment[nb]=TimeTracker programtillegg
 Comment[nds]=Tietlogbook-Moduul
@@ -61,6 +62,7 @@
 Name[ja]=タイムトラッカー
 Name[km]=កម្មវិធី​តាមដាន​ពេលវេលា
 Name[ko]=시간 추적기
+Name[lt]=Laiko sekiklis
 Name[lv]=Patērētā laika uzskatītājs
 Name[nb]=TimeTracker
 Name[nds]=Tietlogbook
Index: kontact/plugins/kjots/kjots_plugin.desktop
===================================================================
--- kontact/plugins/kjots/kjots_plugin.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/kjots/kjots_plugin.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -62,6 +62,7 @@
 Name[ja]=メモ帳
 Name[km]=សៀវភៅ​ចំណាំ
 Name[ko]=공책
+Name[lt]=Užrašinės
 Name[lv]=Piezīmju grāmatiņas
 Name[nb]=Notisbøker
 Name[nds]=Notizböker
Index: kontact/plugins/kjots/kjots.setdlg
===================================================================
--- kontact/plugins/kjots/kjots.setdlg	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/kjots/kjots.setdlg	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -13,6 +13,7 @@
 Name[ja]=メモ帳
 Name[km]=សៀវភៅ​ចំណាំ
 Name[ko]=공책
+Name[lt]=Užrašinės
 Name[lv]=Piezīmju grāmatiņas
 Name[nb]=Notisbøker
 Name[nds]=Notizböker
@@ -46,6 +47,7 @@
 Comment[ja]=メモ帳コンポーネント
 Comment[km]=សមាសភាគ​សៀវភៅ​
 Comment[ko]=공책 구성 요소
+Comment[lt]=Užrašinių komponentas
 Comment[lv]=Piezīmju grāmatiņu komponente
 Comment[nb]=Notisbok-komponent
 Comment[nds]=Notizbook-Komponent
Index: kontact/plugins/korganizer/kcmapptsummary.desktop
===================================================================
--- kontact/plugins/korganizer/kcmapptsummary.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/korganizer/kcmapptsummary.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -28,6 +28,7 @@
 Name[kk]=Болашақ оқиғалар
 Name[km]=ព្រឹត្តិការណ៍​ដែល​នឹង​កើតឡើង
 Name[ko]=다가오는 약속
+Name[lt]=Artėjantys įvykiai
 Name[lv]=Gaidāmie notikumi
 Name[nb]=Kommende hendelser
 Name[nds]=Anstahn Begeefnissen
@@ -65,6 +66,7 @@
 Comment[kk]=Болашақ оқигалар тұжырымының баптау
 Comment[km]=ការ​រៀបចំ​សេចក្ដី​សង្ខេប​ព្រឹត្តិការណ៍​ដែល​នឹង​កើតឡើង
 Comment[ko]=다가오는 약속 요약 설정
+Comment[lt]=Artėjančių įvykių santraukos nustatymai
 Comment[lv]=Gaidāmo notikumu kopsavilkuma iestatījumi
 Comment[nb]=Sammendragsoppsett for kommende hendelser
 Comment[nds]=De Översicht för anstahn Begeefnissen instellen
Index: kontact/plugins/korganizer/kcmtodosummary.desktop
===================================================================
--- kontact/plugins/korganizer/kcmtodosummary.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/korganizer/kcmtodosummary.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -28,6 +28,7 @@
 Name[kk]=Кейінге қалдырылған жоспарлар
 Name[km]=ការងារ​ត្រូវធ្វើ​ដែល​មិនទាន់​សម្រេច
 Name[ko]=남아 있는 할 일
+Name[lt]=Neatlikti darbai
 Name[lv]=Darāmie darbi
 Name[nb]=Ventende gjøremål
 Name[nds]=Anstahn Opgaven
@@ -65,6 +66,7 @@
 Comment[kk]=Кейінге қалдырылған жоспарлар тұжырымының баптау
 Comment[km]=ការ​រៀបចំ​សេចក្ដីសង្ខេបការងារ​​ធ្វើ​ដែល​មិន​ទាន់សម្រេច
 Comment[ko]=남아 있는 할 일 요약 설정
+Comment[lt]=Neatliktų darbų santraukos nustatymai
 Comment[lv]=Darbu kopsavilkuma iestatījumi
 Comment[nb]=Oppsett av sammendraget av ventende gjøremål
 Comment[nds]=De Översicht för anstahn Opgaven instellen
Index: kontact/plugins/kmail/summarywidget.cpp
===================================================================
--- kontact/plugins/kmail/summarywidget.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/kmail/summarywidget.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -88,9 +88,13 @@
 {
   // check whether we need to update the message counts
   org::kde::kmail::kmail kmail( DBUS_KMAIL, "/KMail", QDBusConnection::sessionBus() );
-  const int timeOfLastMessageCountChange = kmail.timeOfLastMessageCountChange();
-  if ( timeOfLastMessageCountChange > mTimeOfLastMessageCountUpdate ) {
-    slotUnreadCountChanged();
+  if ( kmail.isValid() ) {
+    QDBusReply<int> timeOfLastMessageCountChange = kmail.timeOfLastMessageCountChange();
+    if ( timeOfLastMessageCountChange.isValid() ) {
+      if ( timeOfLastMessageCountChange > mTimeOfLastMessageCountUpdate ) {
+        slotUnreadCountChanged();
+      }
+    }
   }
 }
 
@@ -127,70 +131,87 @@
   int counter = 0;
   QStringList::ConstIterator it;
   org::kde::kmail::kmail kmail( DBUS_KMAIL, "/KMail", QDBusConnection::sessionBus() );
-  for ( it = folders.begin(); it != folders.end(); ++it ) {
-    if ( activeFolders.contains( *it ) ) {
-      QDBusReply<QString> ref = kmail.getFolder( *it );
-      if ( ref.isValid() && !ref.value().isEmpty() ) {
-        OrgKdeKmailFolderInterface folderInterface(
-          DBUS_KMAIL, "/Folder", QDBusConnection::sessionBus() );
-        QDBusReply<int> replyNumMsg = folderInterface.messages();
-        const int numMsg = replyNumMsg;
-        QDBusReply<int> replyUnreadNumMsg = folderInterface.unreadMessages();
-        const int numUnreadMsg=replyUnreadNumMsg;
+  if ( kmail.isValid() ) {
+    for ( it = folders.begin(); it != folders.end(); ++it ) {
+      if ( activeFolders.contains( *it ) ) {
+        QDBusReply<QString> ref = kmail.getFolder( *it );
+        if ( ref.isValid() && !ref.value().isEmpty() ) {
+          OrgKdeKmailFolderInterface folderInterface(
+            DBUS_KMAIL, "/Folder", QDBusConnection::sessionBus() );
+          if ( !folderInterface.isValid() )
+            continue;
 
-        if ( numUnreadMsg == 0 ) {
-          continue;
-        }
+          QDBusReply<int> replyNumMsg = folderInterface.messages();
+          if ( !replyNumMsg.isValid() )
+            continue;
 
-        // folder icon
-        QString name = folderInterface.unreadIconPath();
-        if ( name.isEmpty() ) {
-          name = defName;
-        }
-        label = new QLabel( this );
-        label->setPixmap( KIconLoader::global()->loadIcon( name, KIconLoader::Small ) );
-        label->setMaximumWidth( label->minimumSizeHint().width() );
-        label->setAlignment( Qt::AlignVCenter );
-        mLayout->addWidget( label, counter, 0 );
-        mLabels.append( label );
+          const int numMsg = replyNumMsg;
+          QDBusReply<int> replyUnreadNumMsg = folderInterface.unreadMessages();
+          if ( !replyUnreadNumMsg.isValid() )
+            continue;
 
-        // folder path
-        QString folderPath;
-        QDBusReply<QString> replyFolderPath;
-        if ( config.readEntry( "ShowFullPath", true ) ) {
-          replyFolderPath= folderInterface.displayPath();
-        } else {
-          replyFolderPath= folderInterface.displayName();
-        }
-        folderPath = replyFolderPath;
-        KUrlLabel *urlLabel = new KUrlLabel( *it, folderPath, this );
-        urlLabel->installEventFilter( this );
-        urlLabel->setAlignment( Qt::AlignLeft );
-        urlLabel->setWordWrap( true );
-        mLayout->addWidget( urlLabel, counter, 1 );
-        mLabels.append( urlLabel );
+          const int numUnreadMsg = replyUnreadNumMsg;
+          if ( numUnreadMsg == 0 ) {
+            continue;
+          }
 
-        connect( urlLabel, SIGNAL(leftClickedUrl(const QString&)),
-                 SLOT(selectFolder(const QString&)) );
+          // folder icon
+          QDBusReply<QString> dbusName = folderInterface.unreadIconPath();
+          QString name;
+          if ( dbusName.isValid() ) {
+            name = dbusName;
+            if ( name.isEmpty() )
+              name = defName;
+          }
+          else
+            name = defName;
+          label = new QLabel( this );
+          label->setPixmap( KIconLoader::global()->loadIcon( name, KIconLoader::Small ) );
+          label->setMaximumWidth( label->minimumSizeHint().width() );
+          label->setAlignment( Qt::AlignVCenter );
+          mLayout->addWidget( label, counter, 0 );
+          mLabels.append( label );
 
-        // unread of total
-        label = new QLabel( i18nc( "%1: number of unread messages "
-                                   "%2: total number of messages",
-                                   "%1 / %2", numUnreadMsg, numMsg ), this );
-        label->setAlignment( Qt::AlignLeft );
-        mLayout->addWidget( label, counter, 2 );
-        mLabels.append( label );
+          // folder path
+          QString folderPath;
+          QDBusReply<QString> replyFolderPath;
+          if ( config.readEntry( "ShowFullPath", true ) ) {
+            replyFolderPath= folderInterface.displayPath();
+          } else {
+            replyFolderPath= folderInterface.displayName();
+          }
+          if ( replyFolderPath.isValid() )
+            folderPath = replyFolderPath;
 
-        // tooltip
-        urlLabel->setToolTip( i18n( "<qt><b>%1</b>"
-                                    "<br/>Total: %2<br/>"
-                                    "Unread: %3</qt>",
-                                    folderPath,
-                                    numMsg,
-                                    numUnreadMsg ) );
-        //TODO: put the folder size in the tooltip
-        //so we need to add a folderSize() to the interface
-        counter++;
+          KUrlLabel *urlLabel = new KUrlLabel( *it, folderPath, this );
+          urlLabel->installEventFilter( this );
+          urlLabel->setAlignment( Qt::AlignLeft );
+          urlLabel->setWordWrap( true );
+          mLayout->addWidget( urlLabel, counter, 1 );
+          mLabels.append( urlLabel );
+
+          connect( urlLabel, SIGNAL(leftClickedUrl(const QString&)),
+                  SLOT(selectFolder(const QString&)) );
+
+          // unread of total
+          label = new QLabel( i18nc( "%1: number of unread messages "
+                                    "%2: total number of messages",
+                                    "%1 / %2", numUnreadMsg, numMsg ), this );
+          label->setAlignment( Qt::AlignLeft );
+          mLayout->addWidget( label, counter, 2 );
+          mLabels.append( label );
+
+          // tooltip
+          urlLabel->setToolTip( i18n( "<qt><b>%1</b>"
+                                      "<br/>Total: %2<br/>"
+                                      "Unread: %3</qt>",
+                                      folderPath,
+                                      numMsg,
+                                      numUnreadMsg ) );
+          //TODO: put the folder size in the tooltip
+          //so we need to add a folderSize() to the interface
+          counter++;
+        }
       }
     }
   }
Index: kontact/plugins/kmail/kcmkmailsummary.desktop
===================================================================
--- kontact/plugins/kmail/kcmkmailsummary.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/kmail/kcmkmailsummary.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -31,6 +31,7 @@
 Name[kk]=Жаңа хабарламалар
 Name[km]=សារ​ថ្មី
 Name[ko]=새 메시지
+Name[lt]=Nauji laiškai
 Name[lv]=Jaunas vēstules
 Name[nb]=Nye meldinger
 Name[nds]=Nieg Narichten
Index: kontact/plugins/summary/kcmkontactsummary.desktop
===================================================================
--- kontact/plugins/summary/kcmkontactsummary.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/summary/kcmkontactsummary.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -26,6 +26,7 @@
 Name[ja]=要約
 Name[km]=សេចក្ដីសង្ខេប
 Name[ko]=요약
+Name[lt]=Santraukos
 Name[lv]=Kopsavilkumi
 Name[nb]=Sammendrag
 Name[nds]=Tosamenfaten
@@ -60,6 +61,7 @@
 Comment[ja]=表示する要約を選択
 Comment[km]=ជម្រើស​សេចក្ដី​សង្ខេប
 Comment[ko]=요약 선택
+Comment[lt]=Santraukos pasirinkimas
 Comment[lv]=Kopsavilkuma izvēle
 Comment[nb]=Sammendragsutvalg
 Comment[nds]=Tosamenfaten utsöken
Index: kontact/plugins/newsticker/kcmkontactknt.desktop
===================================================================
--- kontact/plugins/newsticker/kcmkontactknt.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ kontact/plugins/newsticker/kcmkontactknt.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -30,6 +30,7 @@
 Name[kk]=Ақпарлар
 Name[km]=មតិ​ព័ត៌មាន
 Name[ko]=뉴스 피드
+Name[lt]=Naujienų kanalai
 Name[lv]=Ziņu barotnes
 Name[nb]=Nyhetskilder
 Name[nds]=Mellenströöm
Index: MAINTAINERS
===================================================================
--- MAINTAINERS	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ MAINTAINERS	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -7,13 +7,11 @@
                   Lutz Rogowski <lutz@klaralvdalens-datakonsult.se>
 kfile-plugins     ?
 kioslave          ?
-kitchensync       Tobias Koenig <tokoe@kde.org>
 kjots             Stephen Kelly <steveire@gmail.com>
 kleopatra         Marc Mutz <mutz@kde.org>
                   Steffen Hansen <hansen@kde.org>
 kmail             Thomas McGuire <mcguire@kde.org>
 kmailcvt          Danny Kukawka <danny.kukawka@web.de>
-kmobiletools      Marco Gulino <marco.gulino@gmail.com>
 knode             Volker Krause <vkrause@kde.org>
 knotes            Guillermo Antonio Amaral Bastidas <me@guillermoamaral.com>
 kode              Cornelius Schumacher <schumacher@kde.org>
@@ -22,7 +20,6 @@
 kontactinterfaces Allen Winter <winter@kde.org>
 korganizer        Reinhold Kainhofer <reinhold@kainhofer.com>
                   Cornelius Schumacher <schumacher@kde.org>
-korn              Mart Kelder <mart.mailing@hccnet.nl>
 kpilot            Jason 'vanRijn' Kasper <vR@movingparts.net>
                   Adriaan de Groot <groot@kde.org>
 kresources        ?
Index: korganizer/koeditorgeneralevent.h
===================================================================
--- korganizer/koeditorgeneralevent.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneralevent.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -70,7 +70,7 @@
     */
     void readEvent( Event *event, bool tmpl = false );
     /** Write event settings to event object */
-    void writeEvent( Event * );
+    void fillEvent( Event * );
 
     /** Check if the input is valid. */
     bool validateInput();
Index: korganizer/koattendeeeditor.cpp
===================================================================
--- korganizer/koattendeeeditor.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koattendeeeditor.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -304,7 +304,7 @@
   }
 }
 
-void KOAttendeeEditor::writeIncidence( KCal::Incidence *incidence )
+void KOAttendeeEditor::fillIncidence( KCal::Incidence *incidence )
 {
   if ( mOrganizerCombo ) {
     // TODO: Don't take a string and split it up... Is there a better way?
Index: korganizer/koeditorgeneraltodo.cpp
===================================================================
--- korganizer/koeditorgeneraltodo.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneraltodo.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -304,6 +304,7 @@
     mDueDateEdit->setDate( QDate::currentDate() );
     mDueTimeEdit->setTime( QTime::currentTime() );
     mDueCheck->setChecked( false );
+    mTimeZoneComboDue->setEnabled( false );
   }
 
   if ( todo->hasStartDate() ) {
@@ -319,6 +320,7 @@
     mStartDateEdit->setDate( QDate::currentDate() );
     mStartTimeEdit->setTime( QTime::currentTime() );
     mStartCheck->setChecked( false );
+    mTimeZoneComboStart->setEnabled( false );
   }
 
   mTimeButton->setChecked( !todo->allDay() );
@@ -335,9 +337,9 @@
   mStartDateModified = false;
 }
 
-void KOEditorGeneralTodo::writeTodo( Todo *todo )
+void KOEditorGeneralTodo::fillTodo( Todo *todo )
 {
-  KOEditorGeneral::writeIncidence( todo );
+  KOEditorGeneral::fillIncidence( todo );
 
   // temp. until something better happens.
   QString tmpStr;
Index: korganizer/koeditorgeneral.cpp
===================================================================
--- korganizer/koeditorgeneral.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneral.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -493,7 +493,7 @@
   }
 }
 
-void KOEditorGeneral::writeIncidence( Incidence *incidence )
+void KOEditorGeneral::fillIncidence( Incidence *incidence )
 {
   incidence->setSummary( mSummaryEdit->text() );
   incidence->setLocation( mLocationEdit->text() );
@@ -532,7 +532,7 @@
       incidence->addAlarm( al );
     }
   }
-  mAttachments->writeIncidence( incidence );
+  mAttachments->fillIncidence( incidence );
 }
 
 void KOEditorGeneral::setSummary( const QString &text )
Index: korganizer/koeventeditor.cpp
===================================================================
--- korganizer/koeventeditor.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeventeditor.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -66,6 +66,16 @@
   }
 }
 
+bool KOEventEditor::incidenceModified() {
+  Event *newEvent = 0;
+
+  if ( mEvent ) {
+    newEvent = mEvent->clone();
+    fillEvent( newEvent );
+  }
+  return mEvent && !( *newEvent == *mEvent );
+}
+
 void KOEventEditor::init()
 {
   setupGeneral();
@@ -288,7 +298,7 @@
     Event *oldEvent = mEvent->clone();
     Event *event = mEvent->clone();
 
-    writeEvent( event );
+    fillEvent( event );
 
     if ( *event == *mEvent ) {
       // Don't do anything
@@ -301,7 +311,7 @@
       }
     } else {
       //IncidenceChanger::assignIncidence( mEvent, event );
-      writeEvent( mEvent );
+      fillEvent( mEvent );
       if ( mIsCounter ) {
         KOGroupware::instance()->sendCounterProposal( mCalendar, oldEvent, mEvent );
         // add dummy event at the position of the counter proposal
@@ -320,7 +330,7 @@
     mEvent = new Event;
     mEvent->setOrganizer( Person( KOPrefs::instance()->fullName(),
                           KOPrefs::instance()->email() ) );
-    writeEvent( mEvent );
+    fillEvent( mEvent );
     if ( !mChanger->addIncidence( mEvent, this ) ) {
       delete mEvent;
       mEvent = 0;
@@ -373,16 +383,16 @@
   }
 }
 
-void KOEventEditor::writeEvent( Event *event )
+void KOEventEditor::fillEvent( Event *event )
 {
-  mGeneral->writeEvent( event );
+  mGeneral->fillEvent( event );
   if ( mFreeBusy ) {
-    mFreeBusy->writeIncidence( event );
+    mFreeBusy->fillIncidence( event );
   }
 
   cancelRemovedAttendees( event );
 
-  mRecurrence->writeIncidence( event );
+  mRecurrence->fillIncidence( event );
 
   writeDesignerFields( event );
 }
@@ -429,7 +439,7 @@
 void KOEventEditor::slotSaveTemplate( const QString &templateName )
 {
   Event *event = new Event;
-  writeEvent( event );
+  fillEvent( event );
   saveAsTemplate( event, templateName );
 }
 
@@ -441,7 +451,7 @@
 void KOEventEditor::updateRecurrenceSummary()
 {
   Event *ev =  new Event();
-  writeEvent( ev );
+  fillEvent( ev );
   mGeneral->updateRecurrenceSummary( IncidenceFormatter::recurrenceString( ev ) );
   delete ev;
 }
Index: korganizer/koeditorfreebusy.cpp
===================================================================
--- korganizer/koeditorfreebusy.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorfreebusy.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -760,7 +760,7 @@
   dialog.exec();
 }
 
-void KOEditorFreeBusy::writeIncidence( Incidence *incidence )
+void KOEditorFreeBusy::fillIncidence( Incidence *incidence )
 {
   incidence->clearAttendees();
   QVector<FreeBusyItem*> toBeDeleted;
@@ -804,7 +804,7 @@
     }
   }
 
-  KOAttendeeEditor::writeIncidence( incidence );
+  KOAttendeeEditor::fillIncidence( incidence );
 
   // cleanup
   qDeleteAll( toBeDeleted );
Index: korganizer/koattendeeeditor.h
===================================================================
--- korganizer/koattendeeeditor.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koattendeeeditor.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -56,7 +56,7 @@
     virtual void insertAttendee( KCal::Attendee *attendee, bool fetchFB = true ) = 0;
 
     virtual void readIncidence( KCal::Incidence *incidence );
-    virtual void writeIncidence( KCal::Incidence *incidence );
+    virtual void fillIncidence( KCal::Incidence *incidence );
 
     /** return a clone of the incidence with attendees to be canceld*/
     void cancelAttendeeIncidence( KCal::Incidence *incidence );
Index: korganizer/dbuscalendar.desktop
===================================================================
--- korganizer/dbuscalendar.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/dbuscalendar.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -35,6 +35,7 @@
 Comment[kk]=D-BUS интерфейсті Ұйымдастырғыш
 Comment[km]=កម្មវិធី​រៀបចំ​ដែល​មាន​ចំណុច​ប្រទាក់ D-BUS
 Comment[ko]=D-BUS 인터페이스를 사용하는 오거나이저
+Comment[lt]=Planuoklis su D-Bus sąsaja
 Comment[lv]=Organizators ar D-BUS saskarni
 Comment[nb]=Organizer med D-BUS-grensesnitt
 Comment[nds]=En Daagpleger mit en D-Bus-Koppelsteed
Index: korganizer/koeditorgeneral.h
===================================================================
--- korganizer/koeditorgeneral.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneral.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -88,7 +88,7 @@
     void readIncidence( Incidence *incidence );
 
     /** Write event settings to event object */
-    void writeIncidence( Incidence * );
+    void fillIncidence( Incidence * );
 
     /** Check if the input is valid. */
     bool validateInput();
Index: korganizer/koeventeditor.h
===================================================================
--- korganizer/koeventeditor.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeventeditor.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -95,7 +95,7 @@
     /**
       Write event settings to event object
     */
-    void writeEvent( Event * );
+    void fillEvent( Event * );
 
     QObject *typeAheadReceiver() const;
 
@@ -126,6 +126,7 @@
     int msgItemDelete();
     void loadTemplate( CalendarLocal & );
     QStringList &templates() const;
+    bool incidenceModified();
 
   private:
     Event *mEvent;
Index: korganizer/koeditorgeneralevent.cpp
===================================================================
--- korganizer/koeditorgeneralevent.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneralevent.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -432,9 +432,9 @@
   readIncidence( event );
 }
 
-void KOEditorGeneralEvent::writeEvent( Event *event )
+void KOEditorGeneralEvent::fillEvent( Event *event )
 {
-  writeIncidence( event );
+  fillIncidence( event );
 
   QDate tmpDate;
   QTime tmpTime;
Index: korganizer/koeditorattachments.h
===================================================================
--- korganizer/koeditorattachments.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorattachments.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -104,7 +104,7 @@
     /** Read event object and setup widgets accordingly */
     void readIncidence( KCal::Incidence * );
     /** Write event settings to event object */
-    void writeIncidence( KCal::Incidence * );
+    void fillIncidence( KCal::Incidence * );
 
     bool hasAttachments();
 
Index: korganizer/korgac/alarmdialog.cpp
===================================================================
--- korganizer/korgac/alarmdialog.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/korgac/alarmdialog.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -413,8 +413,17 @@
       // FIXME: Check whether this should be done for all multiple alarms
       if ( alarm->type() == Alarm::Procedure ) {
         // FIXME: Add a message box asking whether the procedure should really be executed
-      kDebug() << "Starting program: '" << alarm->programFile() << "'";
-      QProcess::startDetached( alarm->programFile() );
+        kDebug() << "Starting program: '" << alarm->programFile() << "'";
+
+        QString program = alarm->programFile();
+
+        // if the program name contains spaces escape it
+        if ( program.contains( " " )   &&
+             !( program.startsWith( "\"" ) && program.endsWith( "\"" ) ) ) {
+          program = "\"" + program + "\"";
+        }
+
+        QProcess::startDetached( program + " " + alarm->programArguments() );
       } else if ( alarm->type() == Alarm::Audio ) {
         beeped = true;
         Phonon::MediaObject *player =
Index: korganizer/interfaces/calendar/calendarplugin.desktop
===================================================================
--- korganizer/interfaces/calendar/calendarplugin.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/interfaces/calendar/calendarplugin.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -14,6 +14,7 @@
 Name[ja]=カレンダープラグインインターフェース
 Name[km]=ប្រតិទិន​កម្មវិធី​ជំនួយ​ប្រតិទិន
 Name[ko]=달력 플러그인 인터페이스
+Name[lt]=Kalendoriaus priedo sąsaja
 Name[lv]=Kalendāra spraudņa saskarne
 Name[nb]=Calendar-grensesnitt for programtillegg
 Name[nds]=Kalennermoduul-Koppelsteed
Index: korganizer/interfaces/calendar/calendardecoration.desktop
===================================================================
--- korganizer/interfaces/calendar/calendardecoration.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/interfaces/calendar/calendardecoration.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -14,6 +14,7 @@
 Name[ja]=カレンダー装飾インターフェース
 Name[km]=ចំណុច​ប្រទាក់​តុបតែង​ប្រតិទិន
 Name[ko]=달력 장식 인터페이스
+Name[lt]=Kalendoriaus dekoracijų sąsaja
 Name[lv]=Kalendāra dekorāciju saskarne
 Name[nb]=Calendar dekorasjonsgrensesnitt
 Name[nds]=Kalennerdekoratschoon-Koppelsteed
Index: korganizer/kotodoeditor.cpp
===================================================================
--- korganizer/kotodoeditor.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/kotodoeditor.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -62,6 +62,16 @@
   emit dialogClose( mTodo );
 }
 
+bool KOTodoEditor::incidenceModified() {
+  Todo *newTodo = 0;
+
+  if ( mTodo ) {
+    newTodo = mTodo->clone();
+    fillTodo( newTodo );
+  }
+  return mTodo && !( *newTodo == *mTodo );
+}
+
 void KOTodoEditor::init()
 {
   setupGeneral();
@@ -220,13 +230,13 @@
     Todo *oldTodo = mTodo->clone();
     Todo *todo = mTodo->clone();
 
-    writeTodo( todo );
+    fillTodo( todo );
 
     if( *mTodo == *todo ) {
       // Don't do anything
     } else {
       //IncidenceChanger::assignIncidence( mTodo, todo );
-      writeTodo( mTodo );
+      fillTodo( mTodo );
       mChanger->changeIncidence( oldTodo, mTodo );
     }
     delete todo;
@@ -237,7 +247,7 @@
     mTodo = new Todo;
     mTodo->setOrganizer( Person( KOPrefs::instance()->fullName(), KOPrefs::instance()->email() ) );
 
-    writeTodo( mTodo );
+    fillTodo( mTodo );
 
     if ( !mChanger->addIncidence( mTodo, this ) ) {
       delete mTodo;
@@ -298,13 +308,13 @@
   readDesignerFields( todo );
 }
 
-void KOTodoEditor::writeTodo( Todo *todo )
+void KOTodoEditor::fillTodo( Todo *todo )
 {
   Incidence *oldIncidence = todo->clone();
 
-  mGeneral->writeTodo( todo );
-  mDetails->writeIncidence( todo );
-  mRecurrence->writeIncidence( todo );
+  mGeneral->fillTodo( todo );
+  mDetails->fillIncidence( todo );
+  mRecurrence->fillIncidence( todo );
 
   if ( *( oldIncidence->recurrence() ) != *( todo->recurrence() ) ) {
     todo->setDtDue( todo->dtDue(), true );
@@ -367,7 +377,7 @@
 void KOTodoEditor::slotSaveTemplate( const QString &templateName )
 {
   Todo *todo = new Todo;
-  writeTodo( todo );
+  fillTodo( todo );
   saveAsTemplate( todo, templateName );
 }
 
Index: korganizer/koeditorgeneraljournal.cpp
===================================================================
--- korganizer/koeditorgeneraljournal.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneraljournal.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -176,7 +176,7 @@
   setCategories( journal->categories() );
 }
 
-void KOEditorGeneralJournal::writeJournal( Journal *journal )
+void KOEditorGeneralJournal::fillJournal( Journal *journal )
 {
   journal->setSummary( mSummaryEdit->text() );
   if ( mRichDescription->isChecked() ) {
Index: korganizer/koeditordetails.cpp
===================================================================
--- korganizer/koeditordetails.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditordetails.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -281,7 +281,7 @@
   emit updateAttendeeSummary( mListView->childCount() );
 }
 
-void KOEditorDetails::writeIncidence( Incidence *incidence )
+void KOEditorDetails::fillIncidence( Incidence *incidence )
 {
   incidence->clearAttendees();
   QVector<Q3ListViewItem*> toBeDeleted;
@@ -326,7 +326,7 @@
     }
   }
 
-  KOAttendeeEditor::writeIncidence( incidence );
+  KOAttendeeEditor::fillIncidence( incidence );
 
   // cleanup
   qDeleteAll( toBeDeleted );
Index: korganizer/koeditorfreebusy.h
===================================================================
--- korganizer/koeditorfreebusy.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorfreebusy.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -60,7 +60,7 @@
     void clearAttendees();
 
     void readIncidence( KCal::Incidence *incidence );
-    void writeIncidence( KCal::Incidence *incidence );
+    void fillIncidence( KCal::Incidence *incidence );
 
     void triggerReload();
     void cancelReload();
Index: korganizer/koeditorrecurrence.cpp
===================================================================
--- korganizer/koeditorrecurrence.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorrecurrence.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1407,7 +1407,7 @@
   mExceptions->setDates( incidence->recurrence()->exDates() );
 }
 
-void KOEditorRecurrence::writeIncidence( Incidence *incidence )
+void KOEditorRecurrence::fillIncidence( Incidence *incidence )
 {
   if ( !mEnabledCheck->isChecked() || !isEnabled() ) {
     if ( incidence->recurs() ) {
Index: korganizer/kotodoeditor.h
===================================================================
--- korganizer/kotodoeditor.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/kotodoeditor.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -82,7 +82,7 @@
     void readTodo( Todo *todo, bool tmpl = false );
 
     /** Write To-do settings to todo object */
-    void writeTodo( Todo *todo );
+    void fillTodo( Todo *todo );
 
     /** Check if the input is valid. */
     bool validateInput();
@@ -108,6 +108,7 @@
     void setupGeneral();
     void setupRecurrence();
     int msgItemDelete();
+    bool incidenceModified();
 
   private:
     Todo *mTodo;
Index: korganizer/views/monthview/monthgraphicsitems.cpp
===================================================================
--- korganizer/views/monthview/monthgraphicsitems.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/views/monthview/monthgraphicsitems.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -164,7 +164,10 @@
 
 QPainterPath MonthGraphicsItem::shape() const
 {
-  return widgetPath( true );
+  // The returned shape must be a closed path,
+  // otherwise MonthScene:itemAt( pos ) can have
+  // problems detecting the item
+  return widgetPath( false );
 }
 
 // TODO: remove this method.
Index: korganizer/views/todoview/kotodomodel.cpp
===================================================================
--- korganizer/views/todoview/kotodomodel.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/views/todoview/kotodomodel.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -800,8 +800,10 @@
     if ( role == Qt::EditRole ) {
       switch ( index.column() ) {
         case SummaryColumn:
-          todo->setSummary( value.toString() );
-          modified = KOGlobals::SUMMARY_MODIFIED;
+          if ( !value.toString().isEmpty() ) {
+            todo->setSummary( value.toString() );
+            modified = KOGlobals::SUMMARY_MODIFIED;
+          }
           break;
         case PriorityColumn:
           todo->setPriority( value.toInt() );
Index: korganizer/views/todoview/kotodoview.cpp
===================================================================
--- korganizer/views/todoview/kotodoview.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/views/todoview/kotodoview.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -233,6 +233,8 @@
   }
   connect( mPercentageCompletedPopupMenu, SIGNAL(triggered(QAction *)),
            SLOT(setNewPercentage(QAction *)) );
+
+  setMinimumHeight( 50 );
 }
 
 KOTodoView::~KOTodoView()
Index: korganizer/views/whatsnextview/kowhatsnextview.cpp
===================================================================
--- korganizer/views/whatsnextview/kowhatsnextview.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/views/whatsnextview/kowhatsnextview.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -104,10 +104,10 @@
 
   Event::List events;
   KDateTime::Spec timeSpec = KOPrefs::instance()->timeSpec();
-  for ( QDate date = mStartDate; date <= mEndDate; date = date.addDays( 1 ) ) {
-    events += calendar()->events( date, timeSpec, EventSortStartDate, SortDirectionAscending );
-  }
 
+  events = calendar()->events( mStartDate, mEndDate, timeSpec, false );
+  calendar()->sortEvents( &events, EventSortStartDate, SortDirectionAscending );
+
   if ( events.count() > 0 ) {
     mText += "<p></p>";
     kil.loadIcon( "view-calendar-day", KIconLoader::NoGroup, 22,
Index: korganizer/views/agendaview/koagendaview.cpp
===================================================================
--- korganizer/views/agendaview/koagendaview.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/views/agendaview/koagendaview.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1244,8 +1244,15 @@
     }
     if ( todo ) {
       QTime t = todo->dtDue().toTimeSpec( KOPrefs::instance()->timeSpec() ).time();
-      endY = mAgenda->timeToY( t ) - 1;
-      startY = mAgenda->timeToY( t.addSecs( -1800 ) );
+
+      int halfHour = 1800;
+      if ( t.addSecs( -halfHour ) < t ) {
+        startY = mAgenda->timeToY( t.addSecs( -halfHour ) );
+        endY   = mAgenda->timeToY( t ) - 1;
+      } else {
+        startY = 0;
+        endY   = mAgenda->timeToY( t.addSecs( halfHour ) ) - 1;
+      }
     }
     if ( endY < startY ) {
       endY = startY;
@@ -1471,7 +1478,7 @@
 void KOAgendaView::updateEventIndicatorTop( int newY )
 {
   for ( int i = 0; i < mMinY.size(); ++i ) {
-    mEventIndicatorTop->enableColumn( i, newY >= mMinY[i] );
+    mEventIndicatorTop->enableColumn( i, newY > mMinY[i] );
   }
   mEventIndicatorTop->update();
 }
@@ -1544,8 +1551,12 @@
   KConfigGroup group = config->group( "Views" );
 
   QList<int> sizes = group.readEntry( "Separator AgendaView", QList<int>() );
-  mSplitterAgenda->setSizes( sizes );
 
+  // the size depends on the number of plugins used
+  if ( sizes.count() >= 2 ) {
+    mSplitterAgenda->setSizes( sizes );
+  }
+
   updateConfig();
 }
 
Index: korganizer/kojournaleditor.h
===================================================================
--- korganizer/kojournaleditor.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/kojournaleditor.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -97,7 +97,7 @@
     void readJournal( Journal *journal, bool tmpl = false );
 
     /** Write Journal settings to journal object */
-    void writeJournal( Journal * );
+    void fillJournal( Journal * );
 
     int msgItemDelete();
 
@@ -123,6 +123,7 @@
     void setupGeneral();
     void loadTemplate( CalendarLocal & );
     QStringList &templates() const;
+    bool incidenceModified();
 
   private:
     Journal *mJournal;
Index: korganizer/koeditordetails.h
===================================================================
--- korganizer/koeditordetails.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditordetails.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -95,7 +95,7 @@
     void readIncidence( Incidence * );
 
     /** Write settings to incidence */
-    void writeIncidence( Incidence * );
+    void fillIncidence( Incidence * );
 
     /** Check if the input is valid. */
     bool validateInput();
Index: korganizer/koeditorrecurrence.h
===================================================================
--- korganizer/koeditorrecurrence.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorrecurrence.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -314,7 +314,7 @@
     void readIncidence( Incidence * );
 
     /** Write event settings to event object */
-    void writeIncidence( Incidence * );
+    void fillIncidence( Incidence * );
 
     /** Check if the input is valid. */
     bool validateInput();
Index: korganizer/calendarview.cpp
===================================================================
--- korganizer/calendarview.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/calendarview.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -1256,6 +1256,7 @@
   if ( alarms.isEmpty() ) {
     // Add an alarm if it didn't have one
     Alarm *alm = incidence->newAlarm();
+    alm->setType( Alarm::Display );
     alm->setEnabled( true );
   }
   mChanger->changeIncidence( oldincidence, incidence, KOGlobals::ALARM_MODIFIED );
Index: korganizer/koeditorgeneraltodo.h
===================================================================
--- korganizer/koeditorgeneraltodo.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneraltodo.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -65,7 +65,7 @@
     void readTodo( Todo *todo, bool tmpl = false );
 
     /** Write todo settings to event object */
-    void writeTodo( Todo * );
+    void fillTodo( Todo * );
 
     /** Check if the input is valid. */
     bool validateInput();
Index: korganizer/kojournaleditor.cpp
===================================================================
--- korganizer/kojournaleditor.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/kojournaleditor.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -136,7 +136,7 @@
 
   if ( mJournal ) {
     Journal *oldJournal = mJournal->clone();
-    writeJournal( mJournal );
+    fillJournal( mJournal );
     mChanger->changeIncidence( oldJournal, mJournal );
     delete oldJournal;
   } else {
@@ -144,7 +144,7 @@
     mJournal->setOrganizer( Person( KOPrefs::instance()->fullName(),
                             KOPrefs::instance()->email() ) );
 
-    writeJournal( mJournal );
+    fillJournal( mJournal );
 
     if ( !mChanger->addIncidence( mJournal, this ) ) {
       delete mJournal;
@@ -176,6 +176,16 @@
   mGeneral->setTime( time );
 }
 
+bool KOJournalEditor::incidenceModified() {
+  Journal *newJournal = 0;
+
+  if ( mJournal ) {
+    newJournal = mJournal->clone();
+    fillJournal( newJournal );
+  }
+  return mJournal && !( *newJournal == *mJournal );
+}
+
 void KOJournalEditor::readJournal( Journal *journal, bool tmpl )
 {
   //TODO: just tmpl variable
@@ -189,10 +199,10 @@
   mDetails->readIncidence( journal );
 }
 
-void KOJournalEditor::writeJournal( Journal *journal )
+void KOJournalEditor::fillJournal( Journal *journal )
 {
-  mGeneral->writeJournal( journal );
-  mDetails->writeIncidence( journal );
+  mGeneral->fillJournal( journal );
+  mDetails->fillIncidence( journal );
 }
 
 bool KOJournalEditor::validateInput()
@@ -229,7 +239,7 @@
 void KOJournalEditor::slotSaveTemplate( const QString &templateName )
 {
   Journal *journal = new Journal;
-  writeJournal( journal );
+  fillJournal( journal );
   saveAsTemplate( journal, templateName );
 }
 
Index: korganizer/koeditorattachments.cpp
===================================================================
--- korganizer/koeditorattachments.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorattachments.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -775,7 +775,7 @@
   }
 }
 
-void KOEditorAttachments::writeIncidence( KCal::Incidence *i )
+void KOEditorAttachments::fillIncidence( KCal::Incidence *i )
 {
   i->clearAttachments();
 
Index: korganizer/koincidenceeditor.cpp
===================================================================
--- korganizer/koincidenceeditor.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koincidenceeditor.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -86,8 +86,13 @@
 {
 }
 
+bool KOIncidenceEditor::incidenceModified() {
+  return true;
+}
+
 void KOIncidenceEditor::slotButtonClicked( int button )
 {
+  bool dontQuit = false;
   switch( button ) {
   case KDialog::Ok:
   {
@@ -104,13 +109,16 @@
     processInput();
     break;
   case KDialog::Cancel:
-    if ( KMessageBox::questionYesNo(
-           this,
-           i18nc( "@info", "Do you really want to cancel?" ),
-           i18nc( "@title:window", "KOrganizer Confirmation" ) ) == KMessageBox::Yes ) {
-      processCancel();
-      KDialog::reject();
-    }
+    dontQuit = incidenceModified() &&
+               KMessageBox::questionYesNo(
+                 this,
+                 i18nc( "@info", "Do you really want to cancel?" ),
+                 i18nc( "@title:window", "KOrganizer Confirmation" ) ) == KMessageBox::No;
+
+    if ( !dontQuit ) {
+       processCancel();
+       KDialog::reject();
+     }
     break;
   default:
     KPageDialog::slotButtonClicked( button );
Index: korganizer/resourceview.cpp
===================================================================
--- korganizer/resourceview.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/resourceview.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -308,6 +308,8 @@
   connect( mDeleteButton, SIGNAL( clicked() ), SLOT( removeResource() ) );
   connect( mEditButton, SIGNAL( clicked() ), SLOT( editResource() ) );
 
+  setMinimumHeight( 50 );
+
   updateView();
 }
 
Index: korganizer/koeventpopupmenu.cpp
===================================================================
--- korganizer/koeventpopupmenu.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeventpopupmenu.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -103,6 +103,7 @@
     }
     for ( it = mRecurrenceItems.begin(); it != mRecurrenceItems.end(); ++it ) {
       (*it)->setVisible( mCurrentIncidence->recurs() );
+      (*it)->setEnabled( !mCurrentIncidence->isReadOnly() );
     }
     popup( QCursor::pos() );
   } else {
Index: korganizer/koeventviewer.cpp
===================================================================
--- korganizer/koeventviewer.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeventviewer.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -46,6 +46,7 @@
 {
   mIncidence = 0;
   setNotifyClick( true );
+  setMinimumHeight( 1 );
 }
 
 KOEventViewer::~KOEventViewer()
Index: korganizer/kcmconfigs/korganizer_configtime.desktop
===================================================================
--- korganizer/kcmconfigs/korganizer_configtime.desktop	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/kcmconfigs/korganizer_configtime.desktop	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -29,6 +29,7 @@
 Name[ja]=日付と時間
 Name[km]=កាលបរិច្ឆេទ និង​ពេលវេលា
 Name[ko]=시간과 날짜
+Name[lt]=Laikas ir data
 Name[lv]=Laiks un datums
 Name[nb]=Dato og klokkeslett
 Name[nds]=Tiet un Datum
Index: korganizer/koeditorgeneraljournal.h
===================================================================
--- korganizer/koeditorgeneraljournal.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditorgeneraljournal.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -69,7 +69,7 @@
     /** Read journal object and setup widgets accordingly */
     void readJournal( Journal *, bool tmpl = false );
     /** Write journal settings to event object */
-    void writeJournal( Journal * );
+    void fillJournal( Journal * );
 
     /** Check if the input is valid. */
     bool validateInput();
Index: korganizer/koeditoralarms.cpp
===================================================================
--- korganizer/koeditoralarms.cpp	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koeditoralarms.cpp	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -199,6 +199,7 @@
   connect( this, SIGNAL(okClicked()), SLOT(slotOk()) );
   connect( this, SIGNAL(applyClicked()), SLOT(slotApply()) );
   init();
+  mWidget.mTypeEmailRadio->hide();
 }
 
 KOEditorAlarms::~KOEditorAlarms()
@@ -282,7 +283,7 @@
   mWidget.mRepeats->setChecked( alarm->repeatCount() > 0 );
   if ( alarm->repeatCount() > 0 ) {
     mWidget.mRepeatCount->setValue( alarm->repeatCount() );
-    mWidget.mRepeatInterval->setValue( alarm->snoozeTime().asSeconds() / 60 );
+    mWidget.mRepeatInterval->setValue( alarm->snoozeTime().asSeconds() / 60 ); // show as minutes
   }
   int id = 0;
 
@@ -355,7 +356,7 @@
   // Repeating
   if ( mWidget.mRepeats->isChecked() ) {
     alarm->setRepeatCount( mWidget.mRepeatCount->value() );
-    alarm->setSnoozeTime( mWidget.mRepeatInterval->value() );
+    alarm->setSnoozeTime( mWidget.mRepeatInterval->value() * 60 ); // convert back to seconds
   } else {
     alarm->setRepeatCount( 0 );
   }
Index: korganizer/koincidenceeditor.h
===================================================================
--- korganizer/koincidenceeditor.h	(.../tags/KDE/4.2.1/kdepim)	(wersja 936453)
+++ korganizer/koincidenceeditor.h	(.../branches/KDE/4.2/kdepim)	(wersja 936453)
@@ -133,6 +133,11 @@
     void readDesignerFields( Incidence *i );
     void writeDesignerFields( Incidence *i );
 
+    /**
+      Returns true if the user made any alteration
+    */
+    virtual bool incidenceModified();
+
     // Returns the page widget. To remove the tab, just delete that one.
     QWidget *addDesignerTab( const QString &uifile );
 

Zmiany atrybutów dla: .
___________________________________________________________________
Dodane: svn:externals
   + 


