Index: akregator/plugins/onlinesync/akregator_onlinesync_plugin.desktop
===================================================================
--- akregator/plugins/onlinesync/akregator_onlinesync_plugin.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ akregator/plugins/onlinesync/akregator_onlinesync_plugin.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -9,7 +9,7 @@
 Name[de]=Akregator-Unterstützung für Online-Leser
 Name[el]=Υποστήριξη απομακρυσμένου αναγνώστη ροών
 Name[en_GB]=Akregator Online Feed Reader Support
-Name[es]=Soporte para el lector de fuentes en línea de Akregator
+Name[es]=Apoyo para el lector de fuentes en línea de Akregator
 Name[et]=Akregatori uudistevoogude võrgulugejate toetus
 Name[fi]=Akregatorin Online-syötelukijatuki
 Name[fr]=Prise en charge des lecteurs en ligne pour Akregator
Index: akregator/src/aboutdata.h
===================================================================
--- akregator/src/aboutdata.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ akregator/src/aboutdata.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -28,7 +28,7 @@
 #include "akregator_export.h"
 #include <kaboutdata.h>
 
-#define AKREGATOR_VERSION "1.6.1"
+#define AKREGATOR_VERSION "1.6.2"
 
 namespace Akregator {
 /**
Index: libkleo/libkleopatrarc.desktop
===================================================================
--- libkleo/libkleopatrarc.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ libkleo/libkleopatrarc.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -15,6 +15,7 @@
 Name[it]=TAR (compatibile PGP®)
 Name[ja]=TAR (PGP® 互換)
 Name[km]=TAR (PGP®-compatible)
+Name[lt]=TAR (suderinama su PGP®)
 Name[lv]=TAR (PGP®-saderīgs)
 Name[nb]=TAR (PGP®-kompatibel)
 Name[nds]=TAR (PGP®-kompatibel)
@@ -50,6 +51,7 @@
 Name[it]=ZIP
 Name[ja]=ZIP
 Name[km]=ZIP
+Name[lt]=ZIP
 Name[lv]=ZIP
 Name[nb]=ZIP
 Name[nds]=Zip
@@ -84,6 +86,7 @@
 Name[it]=TAR (con compressione bzip2)
 Name[ja]=TAR (bzip2 圧縮)
 Name[km]=TAR (with bzip2 compression)
+Name[lt]=TAR (su bzip2 suspaudimu)
 Name[lv]=TAR (ar bzip2 saspiešanu)
 Name[nb]=TAR (med bzip2-komprimering)
 Name[nds]=TAR (mit bzip2-Komprimeren)
@@ -476,7 +479,7 @@
 is-cardkey=true
 Name=Smartcard Key
 Name[ca]=Clau de targeta intel·ligent
-Name[ca@valencia]=Clau Smartcard
+Name[ca@valencia]=Clau de targeta intel·ligent
 Name[da]=Smartcard-nøgle
 Name[de]=Smartcard-Zertifikat
 Name[el]=Κλειδί έξυπνης κάρτας
@@ -488,6 +491,7 @@
 Name[it]=Chiave smartcard
 Name[ja]=スマートカードの鍵
 Name[km]=សោ Smartcard
+Name[lt]=Smartcard raktas
 Name[lv]=Viedkartes atslēga
 Name[nb]=Smartkort-nøkkel
 Name[nds]=Smartkoort-Slötel
Index: kleopatra/conf/kleopatra_config_gnupgsystem.desktop
===================================================================
--- kleopatra/conf/kleopatra_config_gnupgsystem.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kleopatra/conf/kleopatra_config_gnupgsystem.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -31,6 +31,7 @@
 Name[it]=Sistema GnuPG
 Name[ja]=GnuPG システム
 Name[km]=ប្រព័ន្ធ GnuPG
+Name[lt]=GnuPG sistema
 Name[lv]=GnuPG sistēma
 Name[nb]=GnuPG-system
 Name[nds]=GnuPG-Systeem
@@ -67,6 +68,7 @@
 Comment[it]=Configurazione delle opzioni del sistema GnuPG
 Comment[ja]=GnuPG システムのオプション設定
 Comment[km]=ការ​កំណត់​រចនា​សម្ព័ន្ធ​របស់​ជម្រើស​ប្រព័ន្ធ GnuPG
+Comment[lt]=GnuPG sistemos nustatymų konfigūravimas
 Comment[lv]=GnuPG sistēmas iestatījumu konfigurēšana
 Comment[nb]=Oppsett av GnuPG systemvalg
 Comment[nds]=De GnuPG-Systeemoptschonen instellen
Index: kleopatra/kleopatra.desktop
===================================================================
--- kleopatra/kleopatra.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kleopatra/kleopatra.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -80,6 +80,7 @@
 GenericName[it]=Gestore di certificati e interfaccia grafica unificata di crittografia
 GenericName[ja]=証明書マネージャと、統合された暗号 GUI
 GenericName[km]=កម្មវិធី​គ្រប់គ្រង​វិញ្ញាបនបត្រ និង​ Unified Crypto GUI
+GenericName[lt]=Sertifikatų tvarkyklė ir vieninga kriptografinė sąsaja
 GenericName[lv]=Sertifikātu pārvaldnieks un vienota kriptofgrāfijas saskarne
 GenericName[nb]=Sertifikatbehandler og forent krypto-GUI
 GenericName[nds]=Zertifikatenpleger un eenheitlich Verslötel-Böversiet
Index: runtime/plugins/akonadi_serializer_microblog.desktop
===================================================================
Index: runtime/agents/nepomuk_calendar_feeder/nepomukcalendarfeeder.desktop
===================================================================
Index: runtime/agents/maildispatcher/maildispatcheragent.desktop
===================================================================
Index: runtime/kresources/kabc/akonadi.desktop
===================================================================
Index: runtime/kresources/kcal/akonadi.desktop
===================================================================
Index: runtime/migration/kaddressbook/kaddressbookmigrator.desktop
===================================================================
Index: runtime/kcm/kcm_akonadi.desktop
===================================================================
Index: runtime/resources/pop3/pop3resource.desktop
===================================================================
Index: runtime/resources/imap/imapresource.desktop
===================================================================
Index: runtime/resources/birthdays/birthdaysresource.desktop
===================================================================
Index: runtime/resources/kolabproxy/kolabproxyresource.desktop
===================================================================
Index: runtime/resources/kabc/kabcresource.desktop
===================================================================
Index: runtime/resources/localbookmarks/localbookmarksresource.desktop
===================================================================
Index: runtime/resources/mbox/mboxresource.desktop
===================================================================
Index: runtime/resources/ical/notes/notesresource.desktop
===================================================================
Index: runtime/resources/mailtransport_dummy/mtdummyresource.desktop
===================================================================
Index: runtime/resources/microblog/microblog.desktop
===================================================================
Index: runtime/resources/contacts/contactsresource.desktop
===================================================================
Index: wizards/groupwisewizard.cpp
===================================================================
--- wizards/groupwisewizard.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ wizards/groupwisewizard.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -195,7 +195,7 @@
     {
       ChangeConfig *c = new ChangeConfig;
       c->file = "korganizerrc";
-      c->group = "FreeBusy";
+      c->group = "FreeBusy Retrieve";
       c->name = "FreeBusyRetrieveUrl";
       c->value = "groupwise://" + GroupwiseConfig::self()->host() + GroupwiseConfig::self()->path() +
         "/freebusy/";
Index: wizards/kolabwizard.cpp
===================================================================
--- wizards/kolabwizard.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ wizards/kolabwizard.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -176,7 +176,7 @@
 
       ChangeConfig *c = new ChangeConfig;
       c->file = "korganizerrc";
-      c->group = "FreeBusy";
+      c->group = "FreeBusy Publish";
 
       c->name = "FreeBusyPublishUrl";
 
@@ -201,7 +201,7 @@
 
       ChangeConfig *c = new ChangeConfig;
       c->file = "korganizerrc";
-      c->group = "FreeBusy";
+      c->group = "FreeBusy Retrieve";
       c->name = "FreeBusyRetrieveUrl";
       c->value = freeBusyBaseUrl.url();
       changes.append( c );
@@ -209,7 +209,7 @@
       // Use full email address for retrieval of free/busy lists
       c = new ChangeConfig;
       c->file = "korganizerrc";
-      c->group = "FreeBusy";
+      c->group = "FreeBusy Retrieve";
       c->name = "FreeBusyFullDomainRetrieval";
       c->value = "true";
       changes.append( c );
Index: wizards/groupwise.kcfg
===================================================================
--- wizards/groupwise.kcfg	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ wizards/groupwise.kcfg	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -72,14 +72,14 @@
   </group>
 
   <propagation source="groupwiserc/Constants/EnableFreeBusy"
-               target="korganizerrc/FreeBusy/FreeBusyRetrieveAuto" />
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrieveAuto" />
 
   <propagation source="groupwiserc/General/User"
-               target="korganizerrc/FreeBusy/FreeBusyRetrieveUser" />
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrieveUser" />
   <propagation source="groupwiserc/General/Password"
-               target="korganizerrc/FreeBusy/FreeBusyRetrievePassword"
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrievePassword"
                hidevalue="true" />
   <propagation source="groupwiserc/General/SavePassword"
-               target="korganizerrc/FreeBusy/FreeBusyRetrieveSavePassword" />
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrieveSavePassword" />
 
 </kcfg>
Index: wizards/slox.kcfg
===================================================================
--- wizards/slox.kcfg	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ wizards/slox.kcfg	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -47,6 +47,6 @@
   </group>
 
   <propagation source="sloxrc/Constants/EnableFreeBusy"
-               target="korganizerrc/FreeBusy/FreeBusyRetrieveAuto" />
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrieveAuto" />
 
 </kcfg>
Index: wizards/kolab.kcfg
===================================================================
--- wizards/kolab.kcfg	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ wizards/kolab.kcfg	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -45,24 +45,24 @@
   </group>
 
   <propagation source="kolabrc/General/Kolab1Legacy"
-               target="korganizerrc/FreeBusy/FreeBusyPublishAuto" />
+               target="korganizerrc/FreeBusy Publish/FreeBusyPublishAuto" />
   <propagation source="kolabrc/Constants/EnableFreeBusy"
-               target="korganizerrc/FreeBusy/FreeBusyRetrieveAuto" />
+               target="korganizerrc/FreeBusy Publish/FreeBusyRetrieveAuto" />
 
   <propagation source="kolabrc/General/User"
-               target="korganizerrc/FreeBusy/FreeBusyPublishUser" />
+               target="korganizerrc/FreeBusy Publish/FreeBusyPublishUser" />
   <propagation source="kolabrc/General/Password"
-               target="korganizerrc/FreeBusy/FreeBusyPublishPassword"
+               target="korganizerrc/FreeBusy Publish/FreeBusyPublishPassword"
                hidevalue="true" />
   <propagation source="kolabrc/General/SavePassword"
-               target="korganizerrc/FreeBusy/FreeBusyPublishSavePassword" />
+               target="korganizerrc/FreeBusy Publish/FreeBusyPublishSavePassword" />
 
   <propagation source="kolabrc/General/User"
-               target="korganizerrc/FreeBusy/FreeBusyRetrieveUser" />
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrieveUser" />
   <propagation source="kolabrc/General/Password"
-               target="korganizerrc/FreeBusy/FreeBusyRetrievePassword"
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrievePassword"
                hidevalue="true" />
   <propagation source="kolabrc/General/SavePassword"
-               target="korganizerrc/FreeBusy/FreeBusyRetrieveSavePassword" />
+               target="korganizerrc/FreeBusy Retrieve/FreeBusyRetrieveSavePassword" />
 
 </kcfg>
Index: wizards/groupwarewizard.desktop
===================================================================
--- wizards/groupwarewizard.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ wizards/groupwarewizard.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -11,7 +11,7 @@
 Name[de]=KDE Groupware-Assistent
 Name[el]=Μάγος Groupware του KDE
 Name[en_GB]=KDE Groupware Wizard
-Name[es]=Asistente de Groupware de KDE
+Name[es]=Asistente de trabajo en grupo de KDE
 Name[et]=KDE Groupware nõustaja
 Name[eu]=KDE groupware morroia
 Name[fa]=جادوگر KDE Groupware
Index: kresources/slox/kabc_ox.desktop
===================================================================
--- kresources/slox/kabc_ox.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kresources/slox/kabc_ox.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -58,7 +58,7 @@
 Comment[de]=Ermöglicht den Zugriff auf Kontakte, die auf einem Open-Xchange-Server gespeichert sind.
 Comment[el]=Προσφέρει πρόσβαση σε επαφές αποθηκευμένες σε έναν εξυπηρετητή Open-Xchange
 Comment[en_GB]=Provides access to contacts stored on an Open-Xchange server
-Comment[es]=Provee acceso a los contactos almacenados en un servidor Open-Xchange
+Comment[es]=Proporciona acceso a los contactos almacenados en un servidor Open-Xchange
 Comment[et]=Võimaldab kasutada kontakte, mis on salvestatud Open-Xchange'i serverisse
 Comment[fr]=Fourni l'accès aux contacts stockés dans un serveur Open-Xchange
 Comment[gl]=Permite acceder aos contactos almacenados nun servidor de Open-Xchange
Index: kresources/groupdav/kabc_groupdav.desktop
===================================================================
--- kresources/groupdav/kabc_groupdav.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kresources/groupdav/kabc_groupdav.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -57,7 +57,7 @@
 Comment[de]=Ermöglicht den Zugriff auf Kontakte, die in einem Adressbuch auf einem GroupDAV-Server wie z. B. OpenGroupware gespeichert sind.
 Comment[el]=Προσφέρει πρόσβαση σε επαφές αποθηκευμένες σε βιβλία διευθύνσεων εξυπηρετών GroupDAV, π.χ. το OpenGroupware
 Comment[en_GB]=Provides access to contacts stored in address books on GroupDAV enabled servers, e.g. OpenGroupware
-Comment[es]=Provee acceso a los contactos almacenados en la libreta de direcciones en un servidor GroupDAV habilitado, por ej. OpenGroupware
+Comment[es]=Provee acceso a los contactos almacenados en la libreta de direcciones en un servidor GroupDAV habilitado, p.e. OpenGroupware
 Comment[et]=Võimaldab kasutada kontakte, mis on salvestatud aadressiraamatutesse GroupDAVi toetusega serverites (nt OpenGroupware)
 Comment[fr]=Fourni l'accès aux contacts stockés dans les dossiers du carnet d'adresses d'un serveur prenant en charge GroupDAV comme par exemple OpenGroupware
 Comment[gl]=Permite acceder aos contactos almacenados nos cadernos de enderezos dos servidores con GroupDAV, p. ex. OpenGroupware
Index: kresources/kolab/kabc/kolab.desktop
===================================================================
--- kresources/kolab/kabc/kolab.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kresources/kolab/kabc/kolab.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -17,6 +17,7 @@
 Name[it]=Rubrica indirizzi su server IMAP via KMail
 Name[ja]=KMail 経由 IMAP サーバのアドレス帳
 Name[km]=សៀវភៅ​អាសយដ្ឋាន​លើ​ម៉ាស៊ីន​បម្រើ IMAP តាម​រយៈ KMail
+Name[lt]=Adresų knygelė IMAP serveryje per KMail
 Name[lv]=Adrešu grāmata uz IMAP servera, izmantojot KMail
 Name[nb]=Adressebok på IMAP-tjener via KMail
 Name[nds]=Adressbook op IMAP-Server över KMail
Index: kresources/groupware/kabc_groupware.desktop
===================================================================
--- kresources/groupware/kabc_groupware.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kresources/groupware/kabc_groupware.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -11,7 +11,7 @@
 Name[de]=Groupware-Server
 Name[el]=Εξυπηρετητής Groupware
 Name[en_GB]=Groupware Server
-Name[es]=Servidor de Groupware
+Name[es]=Servidor de trabajo en grupo
 Name[et]=Grupitöö server
 Name[eu]=Groupware zerbitzaria
 Name[fa]=کارساز Groupware
@@ -60,7 +60,7 @@
 Comment[de]=Ermöglicht den Zugriff auf Kontakte, die auf einem Groupware-Server gespeichert sind.
 Comment[el]=Προσφέρει πρόσβαση σε επαφές αποθηκευμένες σε έναν εξυπηρετητή Groupware.
 Comment[en_GB]=Provides access to contacts stored on a Groupware server.
-Comment[es]=Provee acceso a los contactos almacenados en un servidor Groupware.
+Comment[es]=Proporciona acceso a los contactos almacenados en un servidor de trabajo en grupo.
 Comment[et]=Võimaldab kasutada grupitöö serverisse salvestatud kontakte.
 Comment[fr]=Fourni l'accès aux contacts stockés un serveur collaboratif.
 Comment[gl]=Permite acceder aos contactos almacenados nun servidor de Groupware.
Index: kresources/groupware/kcal_groupware.desktop
===================================================================
--- kresources/groupware/kcal_groupware.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kresources/groupware/kcal_groupware.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -11,7 +11,7 @@
 Name[de]=Groupware-Server
 Name[el]=Εξυπηρετητής Groupware
 Name[en_GB]=Groupware Server
-Name[es]=Servidor de Groupware
+Name[es]=Servidor de trabajo en grupo
 Name[et]=Grupitöö server
 Name[eu]=Groupware zerbitzaria
 Name[fa]=کارساز Groupware
Index: kresources/remote/remote.desktop
===================================================================
--- kresources/remote/remote.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kresources/remote/remote.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -62,7 +62,7 @@
 Comment[de]=Ermöglicht den Zugriff auf einen entfernten Kalender über KDEs Netzwerk-Zugriffsystem KIO.
 Comment[el]=Προσφέρει πρόσβαση ημερολογίου σε ένα απομακρυσμένο αρχείο με τη χρήση του  συστήματος KIO του KDE
 Comment[en_GB]=Provides access to a calendar in a remote file using KDE's network framework KIO
-Comment[es]=Provee acceso a un calendario en un archivo remoto utilizando la infraestructura KIO de red de KDE 
+Comment[es]=Proporciona acceso a un calendario en un archivo remoto utilizando la infraestructura KIO de red de KDE 
 Comment[et]=Võimaldab kasutada võrgufaili salvestatud kalendrit KDE võrguraamistiku KIO abil
 Comment[fr]=Fourni l'accès à un calendrier dans un fichier distant en utilisant les composants réseaux KIO de KDE
 Comment[gl]=Permite acceder a un calendario nun ficheiro remoto empregando a infraestrutura de rede KIO do KDE
Index: blogilo/blogilo.desktop
===================================================================
--- blogilo/blogilo.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ blogilo/blogilo.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -15,6 +15,7 @@
 Name[it]=Blogilo
 Name[ja]=Blogilo
 Name[km]=Blogilo
+Name[lt]=Blogilo
 Name[lv]=Blogilo
 Name[nb]=Blogilo
 Name[nds]=Blogilo
@@ -41,7 +42,7 @@
 GenericName[de]=Ein Blogging-Programm für KDE
 GenericName[el]=Μία εφαρμογή του KDE για blogging
 GenericName[en_GB]=A KDE Blogging Client
-GenericName[es]=Un cliente de Blog par KDE
+GenericName[es]=Un cliente de Blog para KDE
 GenericName[et]=KDE ajaveebiklient
 GenericName[fi]=KDE:n blogausasiakas
 GenericName[fr]=Un client de Blog pour KDE
@@ -49,6 +50,7 @@
 GenericName[it]=Un client KDE per scrivere blog
 GenericName[ja]=KDE のブログ用クライアント
 GenericName[km]=ម៉ាស៊ីន​ប្លុក​របស់ KDE
+GenericName[lt]=KDE tinklaraštinimo klientas
 GenericName[lv]=KDE emuāru klients
 GenericName[nb]=En KDE-klient for blogging
 GenericName[nds]=En Nettdaagbook-Programm för KDE
Index: kmail/kmfilterdlg.cpp
===================================================================
--- kmail/kmfilterdlg.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmfilterdlg.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -268,6 +268,7 @@
       mKeySeqWidget->setObjectName( "FilterShortcutSelector" );
       gl->addWidget( mKeySeqWidget, 8, 3, 1, 1);
       mKeySeqWidget->setEnabled( false );
+      mKeySeqWidget->setModifierlessAllowed( true );
       mKeySeqWidget->setCheckActionCollections(
                              kmkernel->getKMMainWidget()->actionCollections() );
       mConfigureToolbar = new QCheckBox( i18n("Additionally add this filter to the toolbar"), mAdvOptsGroup );
Index: kmail/dbusmail.desktop
===================================================================
--- kmail/dbusmail.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/dbusmail.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -20,6 +20,7 @@
 Comment[it]=Programma di posta con un'interfaccia D-Bus
 Comment[ja]=D-Bus インターフェースを持つメールクライアント
 Comment[km]=កម្មវិធី​សំបុត្រ​ដែល​មាន​ចំណុច​ប្រទាក់ D-BUS
+Comment[lt]=Pašto programa su D-Bus sąsaja
 Comment[lv]=Pasta programma ar D-BUS saskarni
 Comment[nb]=E-postprogram med D-Bus-grensesnitt
 Comment[nds]=Nettpostprogramm mit en D-Bus-Koppelsteed
Index: kmail/application_octetstream.desktop
===================================================================
--- kmail/application_octetstream.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/application_octetstream.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -18,6 +18,7 @@
 Name[it]=Application Octetstream
 Name[ja]=アプリケーション/オクテットストリーム
 Name[km]=ស្ទ្រីម Octet របស់​កម្មវិធី
+Name[lt]=Formatas application/octet-stream
 Name[lv]=Programma oktetu virkne
 Name[nb]=Program oktettstrøm
 Name[nds]=Application/Octet-Stream
Index: kmail/kmail.notifyrc
===================================================================
--- kmail/kmail.notifyrc	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmail.notifyrc	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -77,6 +77,7 @@
 Name[it]=Errore durante il controllo della posta
 Name[ja]=メールのチェック中にエラー
 Name[km]=កំហុស​ខណៈពេល​ពិនិត្យ​មើល​សំបុត្រ​ថ្មី
+Name[lt]=Klaida tikrinant paštą
 Name[lv]=Kļūda pārbaudot pastu
 Name[nb]=Feil under sjekking av e-post
 Name[nds]=Fehler bi't Kieken na Nettpost
@@ -109,6 +110,7 @@
 Comment[it]=Si è verificato un errore durante il controllo dei nuovi messaggi di posta
 Comment[ja]=新着メールのチェック中にエラーが発生しました
 Comment[km]=មាន​កំហុស​ខណៈពេល​ពិនិត្យ​មើល​សំបុត្រ​ថ្មី
+Comment[lt]=Tikrinant paštą įvyko klaida
 Comment[lv]=Gadījās kļūda pārbaudot ienākošo pastu
 Comment[nb]=Det oppsto en feil mens det ble sett etter ny e-post
 Comment[nds]=Bi't Kieken na nieg Nettpost hett dat en Fehler geven
Index: kmail/kmreadermainwin.cpp
===================================================================
--- kmail/kmreadermainwin.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmreadermainwin.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -186,6 +186,7 @@
       mReaderWin->htmlOverride(), mReaderWin->htmlLoadExtOverride(),
       mReaderWin->isFixedFont(), mReaderWin->overrideEncoding() );
   command->setOverrideFont( mReaderWin->cssHelper()->bodyFont( mReaderWin->isFixedFont(), true /*printing*/ ) );
+  command->setAttachmentStrategy( mReaderWin->attachmentStrategy() );
   command->start();
 }
 
Index: kmail/kmreaderwin.cpp
===================================================================
--- kmail/kmreaderwin.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmreaderwin.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -3087,7 +3087,9 @@
   }
 
   if ( Observable * o = memento ? memento->asObservable() : 0 ) {
-    o->attach( this );
+    if ( o ) {
+      o->attach( this );
+    }
   }
 }
 
Index: kmail/dbusimap.desktop
===================================================================
--- kmail/dbusimap.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/dbusimap.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -20,6 +20,7 @@
 Comment[it]=Programma di posta con un'interfaccia D-Bus
 Comment[ja]=D-Bus インターフェースを持つメールクライアント
 Comment[km]=កម្មវិធី​សំបុត្រ​ដែល​មាន​ចំណុច​ប្រទាក់ D-BUS
+Comment[lt]=Pašto programa su D-Bus sąsaja
 Comment[lv]=Pasta programma ar D-BUS saskarni
 Comment[nb]=E-postprogram med D-Bus-grensesnitt
 Comment[nds]=Nettpostprogramm mit en D-Bus-Koppelsteed
Index: kmail/kmcommands.cpp
===================================================================
--- kmail/kmcommands.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmcommands.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -1616,6 +1616,7 @@
                                 bool useFixedFont, const QString & encoding )
   : KMCommand( parent, msg ),
     mHeaderStyle( headerStyle ), mHeaderStrategy( headerStrategy ),
+    mAttachmentStrategy( 0 ),
     mHtmlOverride( htmlOverride ),
     mHtmlLoadExtOverride( htmlLoadExtOverride ),
     mUseFixedFont( useFixedFont ), mEncoding( encoding )
@@ -1634,6 +1635,10 @@
   mOverrideFont = font;
 }
 
+void KMPrintCommand::setAttachmentStrategy( const KMail::AttachmentStrategy *strategy )
+{
+  mAttachmentStrategy = strategy;
+}
 
 KMCommand::Result KMPrintCommand::execute()
 {
@@ -1649,6 +1654,8 @@
   printerWin->setOverrideEncoding( mEncoding );
   printerWin->cssHelper()->setPrintFont( mOverrideFont );
   printerWin->setDecryptMessageOverwrite( true );
+  if ( mAttachmentStrategy != 0 )
+    printerWin->setAttachmentStrategy( mAttachmentStrategy );
   printerWin->printMsg( retrievedMessage() );
 
   return OK;
Index: kmail/headerstyle.cpp
===================================================================
--- kmail/headerstyle.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/headerstyle.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -485,11 +485,11 @@
 
     QString userHTML;
 
-    Akonadi::ContactSearchJob *job = new Akonadi::ContactSearchJob();
+    /*Akonadi::ContactSearchJob *job = new Akonadi::ContactSearchJob();
     job->setLimit( 1 );
     job->setQuery( Akonadi::ContactSearchJob::Email, KPIMUtils::firstEmailAddress( message->from() ) );
-    job->exec();
-    const KABC::Addressee::List addresses = job->contacts();
+    job->exec();*/
+    const KABC::Addressee::List addresses; /*= job->contacts();*/
 
     QString photoURL;
     int photoWidth = 60;
Index: kmail/kmmainwidget.cpp
===================================================================
--- kmail/kmmainwidget.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmmainwidget.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -142,6 +142,7 @@
 #include "accountwizard.h"
 #include "archivefolderdialog.h"
 #include "folderutil.h"
+#include "csshelper.h"
 
 #if !defined(NDEBUG)
     #include "sievedebugdialog.h"
@@ -2653,12 +2654,16 @@
   KConfigGroup reader( KMKernel::config(), "Reader" );
   bool useFixedFont = mMsgView ? mMsgView->isFixedFont() : GlobalSettings::self()->useFixedFont();
 
-  KMCommand *command =
+  // FIXME: Remove code duplication with KMReaderMainWin::slotPrintMsg. Maybe move to MessageActions?
+  KMPrintCommand *command =
     new KMPrintCommand( this, msg,
                         mMsgView ? mMsgView->headerStyle() : 0,
                         mMsgView ? mMsgView->headerStrategy() : 0,
                         htmlOverride, htmlLoadExtOverride,
                         useFixedFont, overrideEncoding() );
+  command->setAttachmentStrategy( mMsgView ? mMsgView->attachmentStrategy() : 0 );
+  if ( mMsgView )
+    command->setOverrideFont( mMsgView->cssHelper()->bodyFont( mMsgView->isFixedFont(), true /*printing*/ ) );
   command->start();
 }
 
Index: kmail/kmcommands.h
===================================================================
--- kmail/kmcommands.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmcommands.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -36,6 +36,7 @@
   class EditorWatcher;
   class HeaderStyle;
   class HeaderStrategy;
+  class AttachmentStrategy;
 }
 namespace GpgME { class Error; }
 namespace Kleo { class SpecialJob; }
@@ -591,12 +592,14 @@
                   const QString & encoding = QString() );
 
   void setOverrideFont( const QFont& );
+  void setAttachmentStrategy( const KMail::AttachmentStrategy *strategy );
 
 private:
   virtual Result execute();
 
   const KMail::HeaderStyle *mHeaderStyle;
   const KMail::HeaderStrategy *mHeaderStrategy;
+  const KMail::AttachmentStrategy *mAttachmentStrategy;
   bool mHtmlOverride;
   bool mHtmlLoadExtOverride;
   bool mUseFixedFont;
Index: kmail/kmfolder.cpp
===================================================================
--- kmail/kmfolder.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kmail/kmfolder.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -1009,6 +1009,7 @@
   int count = mStorage->countUnread();
   if ( count > 0 ) {
     mIndicator->setCountProperty( count );
+    mIndicator->setDrawAttentionProperty( true );
     mIndicator->show();
   } else {
     mIndicator->hide();
Index: doc/kontact-admin/kontact-admin.desktop
===================================================================
--- doc/kontact-admin/kontact-admin.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ doc/kontact-admin/kontact-admin.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -15,6 +15,7 @@
 Name[it]=Gestione di Kontact
 Name[ja]=Kontact の管理
 Name[km]=ការ​គ្រប់គ្រង Kontact
+Name[lt]=Kontact administravimas
 Name[lv]=Kontact administrācija
 Name[nb]=Kontact-administrasjon
 Name[nds]=Kontaktenpleeg
Index: doc/kmail/configure.docbook
===================================================================
--- doc/kmail/configure.docbook	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ doc/kmail/configure.docbook	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -708,7 +708,7 @@
 
 <para>With <guilabel>Date Display</guilabel> you can choose between several
 date formats. The <guilabel>Localized Format</guilabel> is the one you can
-specify under <guilabel>Country &amp; Language</guilabel> in &kcontrol;.
+specify under <guilabel>Country/Region &amp; Language</guilabel> in the &systemsettings;.
 For the <guilabel>Custom</guilabel> format you can get
 a description of the possible values by pressing
 <keycombo action="simul">&Shift;<keycap>F1</keycap></keycombo>
Index: doc/kmail/faq.docbook
===================================================================
--- doc/kmail/faq.docbook	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ doc/kmail/faq.docbook	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -182,7 +182,7 @@
 open links in messages?</para></question>
 <answer>
 <para>Change the <guilabel>File Associations</guilabel> for HTML files
-using &kcontrol;.</para>
+using &systemsettings;.</para>
 </answer>
 </qandaentry>
 
Index: kalarm/alarmcalendar.cpp
===================================================================
--- kalarm/alarmcalendar.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/alarmcalendar.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -31,8 +31,6 @@
 #include "preferences.h"
 
 #include <kcal/calendarlocal.h>
-#include <kcal/vcaldrag.h>
-#include <kcal/vcalformat.h>
 #include <kcal/icalformat.h>
 
 #include <kglobal.h>
@@ -44,16 +42,8 @@
 #include <kio/netaccess.h>
 #include <kfileitem.h>
 #include <ktemporaryfile.h>
-#include <kfiledialog.h>
 #include <kdebug.h>
 
-#include <QFile>
-#include <QTextStream>
-#include <QRegExp>
-
-#include <unistd.h>
-#include <time.h>
-
 using namespace KCal;
 
 static const QString displayCalendarName = QLatin1String("displaying.ics");
Index: kalarm/messagewin.cpp
===================================================================
--- kalarm/messagewin.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/messagewin.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -1,7 +1,7 @@
 /*
  *  messagewin.cpp  -  displays an alarm message
  *  Program:  kalarm
- *  Copyright © 2001-2009 by David Jarvie <djarvie@kde.org>
+ *  Copyright © 2001-2010 by David Jarvie <djarvie@kde.org>
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -1361,7 +1361,7 @@
 {
 	kDebug();
 	if (mAudioThread)
-		mAudioThread->stop();
+		mAudioThread->stop(wait);
 }
 
 /******************************************************************************
Index: kalarm/akonadi/kalarmresource_active.desktop
===================================================================
--- kalarm/akonadi/kalarmresource_active.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/akonadi/kalarmresource_active.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -14,6 +14,7 @@
 Name[it]=Avvisi attivi di KAlarm
 Name[ja]=KAlarm アクティブなアラーム
 Name[km]=ការ​ជូនដំណឹង​សកម្ម​របស់ KAlarm
+Name[lt]=KAlarm aktyvūs priminimai
 Name[lv]=KAlarm aktīvie atgādinājumi
 Name[nb]=KAlarm aktive varslinger
 Name[nds]=Aktiev KAlarm-Alarms
@@ -45,6 +46,7 @@
 Comment[it]=Carica dati da un calendario per avvisi attivi di KAlarm
 Comment[ja]=KAlarm のアクティブなアラームのカレンダーファイルからデータを読み込みます
 Comment[km]=ផ្ទុក​ទិន្នន័យ​ពី​ឯកសារ​ប្រតិទិន​ជូនដំណឹង​សកម្ម​របស់ KAlarm
+Comment[lt]=Įkelia duomenis iš KAlarm aktyvių priminimų kalendoriaus failo
 Comment[lv]=Ielādē datus no KAlarm aktīvo atgādinājumu kalendāra faila
 Comment[nb]=Laster data fra en KAlarm kalenderfil med aktive varslinger
 Comment[nds]=Laadt Daten ut en KAlarm-Kalennerdatei för aktiev Alarms
Index: kalarm/akonadi/kalarmresource_template.desktop
===================================================================
--- kalarm/akonadi/kalarmresource_template.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/akonadi/kalarmresource_template.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -14,6 +14,7 @@
 Name[it]=Modelli di KAlarm
 Name[ja]=KAlarm テンプレート
 Name[km]=ពុម្ព​របស់ KAlarm
+Name[lt]=KAlarm šablonai
 Name[lv]=KAlarm sagataves
 Name[nb]=KAlarm-maler
 Name[nds]=KAlarm-Vörlagen
@@ -45,6 +46,7 @@
 Comment[it]=Carica dati da un modello di avviso di KAlarm
 Comment[ja]=KAlarm のアラームテンプレートファイルからデータを読み込みます
 Comment[km]=ផ្ទុក​ទិន្នន័យ​ពី​ឯកសារ​ពុម្ព​ជូន​ដំណឹង​របស់ KAlarm
+Comment[lt]=Įkelia duomenis iš KAlarm šablono failo
 Comment[lv]=Ielādē datus no KAlarm atgādinājumu sagatavju faila
 Comment[nb]=Laster data fra en KAlarm kalenderfil med varslingsmaler
 Comment[nds]=Laadt Daten ut en KAlarm-Alarmvörlagendatei
Index: kalarm/akonadi/kalarmresource.desktop
===================================================================
--- kalarm/akonadi/kalarmresource.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/akonadi/kalarmresource.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -14,6 +14,7 @@
 Name[it]=Calendario di KAlarm
 Name[ja]=KAlarm カレンダーファイル
 Name[km]=ឯកសារ​ប្រតិទិន​របស់ KAlarm
+Name[lt]=KAlarm kalendoriaus failas
 Name[lv]=KAlarm kalendāra fails
 Name[nb]=KAlarm kalenderfil
 Name[nds]=KAlarm-Kalennerdatei
@@ -45,6 +46,7 @@
 Comment[it]=Carica dati da un calendario di KAlarm
 Comment[ja]=KAlarm のカレンダーファイルからデータを読み込みます
 Comment[km]=ផ្ទុក​ទិន្នន័យ​ពី​ឯកសារ​ប្រតិទិន​របស់ KAlarm
+Comment[lt]=Įkelia duomenis iš KAlarm kalendoriaus failo
 Comment[lv]=Ielādē datus no KAlarm kalendāra faila
 Comment[nb]=Laster data fra en KAlarm kalenderfil
 Comment[nds]=Laadt Daten ut en KAlarm-Kalennerdatei
Index: kalarm/akonadi/kalarmresource_archived.desktop
===================================================================
--- kalarm/akonadi/kalarmresource_archived.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/akonadi/kalarmresource_archived.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -14,6 +14,7 @@
 Name[it]=Avvisi archiviati di KAlarm
 Name[ja]=KAlarm アーカイブのアラーム
 Name[km]=ការ​ជូន​ដំណឹង​ប័ណ្ណសារ​របស់ KAlarm
+Name[lt]=KAlarm archyvuoti priminimai
 Name[lv]=KAlarm arhivētie atgādinājumi
 Name[nb]=KAlarm arkiverte varslinger
 Name[nds]=Aflopen KAlarm-Alarms
@@ -44,6 +45,7 @@
 Comment[it]=Carica dati da un calendario per avvisi archiviati di KAlarm
 Comment[ja]=KAlarm のアーカイブされたアラームのカレンダーファイルからデータを読み込みます
 Comment[km]=ផ្ទុក​ទិន្នន័យ​ពី​ឯកសារ​ប្រតិទិន​ជូន​ដំណឹង​ប័ណ្ណសារ​របស់ KAlarm
+Comment[lt]=Įkelia duomenis iš KAlarm aktyvių archyvuotų priminimų kalendoriaus failo
 Comment[lv]=Ielādē datus no KAlarm arhivēto atgādinājumu kalendāra faila
 Comment[nb]=Laster data fra en KAlarm kalenderfil med arkiverte varslinger
 Comment[nds]=Laadt Daten ut en KAlarm-Kalennerdatei för aflopen Alarms
Index: kalarm/akonadi/akonadi_serializer_kalarm.desktop
===================================================================
--- kalarm/akonadi/akonadi_serializer_kalarm.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/akonadi/akonadi_serializer_kalarm.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -14,6 +14,7 @@
 Name[it]=Serializzatore degli eventi di KAlarm
 Name[ja]=KAlarm イベント用シリアライザ
 Name[km]=កម្មវិធី​បោះពុម្ព​ព្រឹត្តិការណ៍ KAlarm
+Name[lt]=KAlarm įvykių serializatorius
 Name[lv]=KAlarm notikumu serializētājs
 Name[nb]=KAlarm hendelsesserialiserer
 Name[nds]=KAlarm-Begeefnis-Reegmoduul
@@ -44,6 +45,7 @@
 Comment[it]=Un'estensione di Akonadi per la serializzazione di eventi di KAlarm
 Comment[ja]=KAlarm のイベントのための Akonadi シリアライザプラグイン
 Comment[km]=កម្មវិធី​ជំនួយ​​របស់​កម្មវិធី​បោះពុម្ព Akonadi សម្រាប់​ព្រឹត្តិការណ៍ KAlarm
+Comment[lt]=Akonadi serializatoriaus priedas KAlarm įvykiams
 Comment[lv]=Akonadi KAlarm notikumu serializēšanas spraudnis
 Comment[nb]=Et Akonadi serialiserer-programtillegg for KAlarm-hendelser
 Comment[nds]=Akonadi-Inreegmoduul för KAlarm-Begeefnissen
Index: kalarm/kaevent.cpp
===================================================================
--- kalarm/kaevent.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kalarm/kaevent.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -82,7 +82,8 @@
 }
 
 KAEvent::Private::Private(const KAEvent::Private& e)
-    : QSharedData(e)
+    : KAEventData::Observer(e),
+      QSharedData(e)
 {
     copy(e, false);
     mEventData = new KAEventData(this, *e.mEventData);
Index: kaddressbook/xxport/ldap/kcmldap.desktop
===================================================================
--- kaddressbook/xxport/ldap/kcmldap.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kaddressbook/xxport/ldap/kcmldap.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -25,6 +25,7 @@
 Name[it]=Impostazioni server LDAP
 Name[ja]=LDAP サーバの設定
 Name[km]=ការ​កំណត់​ម៉ាស៊ីន​បម្រើ LDAP
+Name[lt]=LDAP serverio nustatymai
 Name[lv]=LDAP serveru iestatījumi
 Name[nb]=LDAP tjenerinnstillinger
 Name[nds]=LDAP-Server instellen
@@ -57,6 +58,7 @@
 Comment[it]=Configura i server LDAP disponibili
 Comment[ja]=利用可能な LDAP サーバの設定
 Comment[km]=កំណត់​រចនាសម្ព័ន្ធ​ម៉ាស៊ីន​បម្រើ LDAP ដែល​អាច​ប្រើបាន
+Comment[lt]=Konfigūruoti prieinamus LDAP serverius
 Comment[lv]=Konfigurē pieejamos LDAP serverus
 Comment[nb]=Sett opp de tilgjengelige LDAP-tjenerne
 Comment[nds]=De verföögboren LDAP-Servers instellen
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ CMakeLists.txt	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -210,7 +210,7 @@
 
     include( SopranoAddOntology )
     add_subdirectory( ontologies )
-    add_subdirectory( nepomuk_email_feeder )
+#    add_subdirectory( nepomuk_email_feeder )
   endif(QGPGME_FOUND)
 
   if(KDEPIM_BUILD_EVERYTHING)
Index: ktimetracker/ktimetracker_config_storage.desktop
===================================================================
--- ktimetracker/ktimetracker_config_storage.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ ktimetracker/ktimetracker_config_storage.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -31,6 +31,7 @@
 Name[it]=Memorizzazione
 Name[ja]=保存
 Name[km]=ឧបករណ៍​ផ្ទុក
+Name[lt]=Saugojimas
 Name[lv]=Glabāšana
 Name[nb]=Lagring
 Name[nds]=Sekern
@@ -66,6 +67,7 @@
 Comment[it]=Configura la memorizzazione
 Comment[ja]=保存の設定
 Comment[km]=កំណត់​រចនា​សម្ព័ន្ធ​ឧបករណ៍​ផ្ទុក
+Comment[lt]=Konfigūruoti saugojimą
 Comment[lv]=Konfigurēt glabāšanu
 Comment[nb]=Sett opp lagring
 Comment[nds]=Sekern instellen
Index: ktimetracker/ktimetracker_config_behavior.desktop
===================================================================
--- ktimetracker/ktimetracker_config_behavior.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ ktimetracker/ktimetracker_config_behavior.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -32,6 +32,7 @@
 Name[it]=Comportamento
 Name[ja]=挙動
 Name[km]=ឥរិយាបទ
+Name[lt]=Elgsena
 Name[lv]=Izturēšanās
 Name[nb]=Oppførsel
 Name[nds]=Bedregen
Index: kjots/kjots_config_misc.desktop
===================================================================
--- kjots/kjots_config_misc.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kjots/kjots_config_misc.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -90,6 +90,7 @@
 Comment[it]=Impostazioni varie per KJots
 Comment[ja]=KJots のオプション設定
 Comment[km]=ការ​ដំឡើង​ផ្សេងៗ​សម្រាប់ KJots
+Comment[lt]=Papildomi KJots nustatymai
 Comment[lv]=Dažādi KJots iestatījumi
 Comment[nb]=Sett opp diverse for KJots
 Comment[nds]=Anner Saken binnen KJots instellen
Index: nepomuk_email_feeder/shared/nepomukfeederagentbase.cpp
===================================================================
--- nepomuk_email_feeder/shared/nepomukfeederagentbase.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ nepomuk_email_feeder/shared/nepomukfeederagentbase.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -320,7 +320,7 @@
   emit status( Broken, i18n( "Nepomuk not operational" ) );
   if ( !QDBusConnection::sessionBus().registerService( QLatin1String( "org.kde.pim.nepomukfeeder.selftestreport" ) ) )
     return;
-  KMessageBox::error( 0, message, i18n( "Nepomuk Indexing Disabled" ), KMessageBox::Notify | KMessageBox::AllowLink );
+  //KMessageBox::error( 0, message, i18n( "Nepomuk Indexing Disabled" ), KMessageBox::Notify | KMessageBox::AllowLink );
   QDBusConnection::sessionBus().unregisterService( QLatin1String( "org.kde.pim.nepomukfeeder.selftestreport" ) );
 }
 
Index: nepomuk_email_feeder/nepomukemailfeeder.desktop
===================================================================
--- nepomuk_email_feeder/nepomukemailfeeder.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ nepomuk_email_feeder/nepomukemailfeeder.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -7,7 +7,7 @@
 Name[de]=E-Mail-Modul für Nepomuk
 Name[el]=Τροφοδότης EMail του Nepomuk
 Name[en_GB]=Nepomuk EMail Feeder
-Name[es]=Proveedor de correos electrónicos de Nepomuk
+Name[es]=Proveedor de correo de Nepomuk
 Name[et]=Nepomuki e-kirjade söötur
 Name[fr]=Connecteur Nepomuk pour les courriels
 Name[ga]=Fothaire Ríomhphoist Nepomuk
@@ -44,7 +44,7 @@
 Comment[de]=Erweiterung zur Übergabe von E-Mails an Nepomuk
 Comment[el]=Επέκταση για την τροφοδοσία email στο Nepomuk
 Comment[en_GB]=Extension to push emails into Nepomuk
-Comment[es]=Extensión para introducir correos electrónicos en Nepomuk
+Comment[es]=Extensión para introducir correos en Nepomuk
 Comment[et]=Laiendus e-kirjade edastamiseks Nepomukile
 Comment[fr]=Extension pour fournir les courriels à Nepomuk
 Comment[gl]=Engadido para lle pasar mensaxes de correo a Nepomuk
Index: nepomuk_email_feeder/CMakeLists.txt
===================================================================
--- nepomuk_email_feeder/CMakeLists.txt	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ nepomuk_email_feeder/CMakeLists.txt	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -4,6 +4,7 @@
   ${Boost_INCLUDE_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/shared
   ${SOPRANO_INCLUDE_DIR}
+  ${STRIGI_INCLUDE_DIR}
 )
 
 set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}" )
Index: console/kabcclient/doc/examples/letters/work_address.desktop
===================================================================
--- console/kabcclient/doc/examples/letters/work_address.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ console/kabcclient/doc/examples/letters/work_address.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -11,7 +11,7 @@
 Name[de]=Postanschrift (geschäftlich)
 Name[el]=Διευθύνσεις αλληλογραφίας επιχείρησης/εργασίας
 Name[en_GB]=Letter business/work address
-Name[es]=Dirección postal de negocios/del trabajo
+Name[es]=Dirección postal de negocios/trabajo
 Name[et]=Kirja äri/tööaadress
 Name[fi]=Työpostiosoite
 Name[fr]=Adresse postale professionnelle
Index: knotes/knote_config_style.desktop
===================================================================
--- knotes/knote_config_style.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ knotes/knote_config_style.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -69,6 +69,7 @@
 Comment[it]=Impostazioni dello stile
 Comment[ja]=スタイルの設定
 Comment[km]=ការ​កំណត់​រចនា​ប័ទ្ម
+Comment[lt]=Stiliaus nustatymai
 Comment[lv]=Stila iestatījumi
 Comment[nb]=Stilinnstillinger
 Comment[nds]=Stil instellen
Index: knotes/knote_config_display.desktop
===================================================================
--- knotes/knote_config_display.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ knotes/knote_config_display.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -32,6 +32,7 @@
 Name[it]=Visualizzazione
 Name[ja]=表示
 Name[km]=បង្ហាញ
+Name[lt]=Rodymas
 Name[lv]=Attēlošana
 Name[nb]=Visning
 Name[nds]=Dorstellen
@@ -69,6 +70,7 @@
 Comment[it]=Imposta la visualizzazione per le note
 Comment[ja]=ポップアップメモの表示設定
 Comment[km]=រៀបចំ​ការ​បង្ហាញ​សម្រាប់​ការ​កត់​ចំណាំ
+Comment[lt]=Nustatyti lapelių rodymą
 Comment[lv]=Piezīmju attēlošanas konfigurēšana
 Comment[nb]=Sett opp visning av notater
 Comment[nds]=Dorstellen för Notizen inrichten
Index: knotes/knote_config_editor.desktop
===================================================================
--- knotes/knote_config_editor.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ knotes/knote_config_editor.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -32,6 +32,7 @@
 Name[it]=Editor
 Name[ja]=エディタ
 Name[km]=កម្មវិធី​និពន្ធ
+Name[lt]=Rengyklė
 Name[lv]=Redaktors
 Name[nb]=Redigerer
 Name[nds]=Editor
@@ -69,6 +70,7 @@
 Comment[it]=Imposta l'editor
 Comment[ja]=エディタの設定
 Comment[km]=រៀបចំ​កម្មវិធី​និពន្ធ
+Comment[lt]=Nustatyti rengyklę
 Comment[lv]=Redaktora konfigurēšana
 Comment[nb]=Sett opp redigering
 Comment[nds]=Editor inrichten
Index: knotes/knote_config_network.desktop
===================================================================
--- knotes/knote_config_network.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ knotes/knote_config_network.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -32,6 +32,7 @@
 Name[it]=Rete
 Name[ja]=ネットワーク
 Name[km]=បណ្ដាញ
+Name[lt]=Tinklas
 Name[lv]=Tīkls
 Name[nb]=Nettverk
 Name[nds]=Nettwark
@@ -69,6 +70,7 @@
 Comment[it]=Impostazioni di rete
 Comment[ja]=ネットワークの設定
 Comment[km]=ការ​កំណត់​បណ្ដាញ
+Comment[lt]=Tinklo nustatymai
 Comment[lv]=Tīlka iestatījumi
 Comment[nb]=Nettverksinnstillinger
 Comment[nds]=Nettwark instellen
Index: knotes/knote_config_action.desktop
===================================================================
--- knotes/knote_config_action.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ knotes/knote_config_action.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -68,6 +68,7 @@
 Comment[it]=Imposta le azioni per le note
 Comment[ja]=ポップアップメモのアクション設定
 Comment[km]=រៀបចំ​សកម្មភាព​សម្រាប់​ការ​កត់​ចំណាំ
+Comment[lt]=Nustatyti veiksmus pastaboms
 Comment[lv]=Piezīmju darbību konfigurēšana
 Comment[nb]=Sett opp handlinger for notater
 Comment[nds]=Akschonen för Notizen inrichten
Index: kontact/plugins/summary/summaryplugin.desktop
===================================================================
--- kontact/plugins/summary/summaryplugin.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kontact/plugins/summary/summaryplugin.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -23,7 +23,7 @@
 Comment[de]=Übersichts-Modul für Kontact
 Comment[el]=Πρόσθετο προβολής σύνοψης του Kontact
 Comment[en_GB]=Kontact SummaryView Plugin
-Comment[es]=Complemento de Vista resumen para Kontact
+Comment[es]=Complemento de vista de resumen para Kontact
 Comment[et]=Kontakti kokkuvõttevaate plugin
 Comment[eu]=Kontact-en SummaryView plugin-a
 Comment[fa]=وصلۀ نمای خلاصۀ Kontact
Index: kontact/src/kontactconfig.desktop
===================================================================
--- kontact/src/kontactconfig.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ kontact/src/kontactconfig.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -84,6 +84,7 @@
 Comment[gl]=Compoñente por omisión de Kontact de KDE
 Comment[it]=Componente predefinito di KDE Kontact
 Comment[km]=សមាសធាតុ​ Kontact របស់ KDE លំនាំដើម
+Comment[lt]=Numatytasis KDE Kontact komponentas
 Comment[lv]=Noklusētā KDE kontaktu komponente
 Comment[nb]=Standard KDE Kontact-komponent
 Comment[nds]=Standard-Kontaktkomponent för KDE
Index: korganizer/printing/calprinter.cpp
===================================================================
--- korganizer/printing/calprinter.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/printing/calprinter.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -43,8 +43,8 @@
 
 #ifndef KORG_NOPRINTER
 
-CalPrinter::CalPrinter( QWidget *parent, Calendar *calendar, KOrg::CoreHelper *helper )
-  : QObject( parent )
+CalPrinter::CalPrinter( QWidget *parent, Calendar *calendar, KOrg::CoreHelper *helper, bool uniqItem )
+  : QObject( parent ), mUniqItem( uniqItem )
 {
   mParent = parent;
   mConfig = new KConfig( "korganizer_printing.rc", KConfig::SimpleConfig );
@@ -98,7 +98,7 @@
   for ( it = mPrintPlugins.begin(); it != mPrintPlugins.end(); ++it ) {
     (*it)->setSelectedIncidences( selectedIncidences );
   }
-  QPointer<CalPrintDialog> printDialog = new CalPrintDialog( mPrintPlugins, mParent );
+  QPointer<CalPrintDialog> printDialog = new CalPrintDialog( mPrintPlugins, mParent, mUniqItem );
   KConfigGroup grp( mConfig, "" ); //orientation setting isn't in a group
   printDialog->setOrientation( CalPrinter::ePrintOrientation( grp.readEntry( "Orientation", 1 ) ) );
   printDialog->setPreview( preview );
@@ -165,7 +165,7 @@
 {
 }
 
-CalPrintDialog::CalPrintDialog( KOrg::PrintPlugin::List plugins, QWidget *parent )
+CalPrintDialog::CalPrintDialog( KOrg::PrintPlugin::List plugins, QWidget *parent, bool uniqItem )
   : KDialog( parent )
 {
   setCaption( i18n( "Print" ) );
@@ -228,6 +228,8 @@
     typeLayout->addWidget( radioButton );
     id++;
   }
+  if ( uniqItem )
+    typeBox->hide();
   typeLayout->insertStretch( -1, 100 );
   connect( this, SIGNAL(okClicked()), SLOT(slotOk()) );
   setMinimumSize( minimumSizeHint() );
Index: korganizer/printing/calprinter.h
===================================================================
--- korganizer/printing/calprinter.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/printing/calprinter.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -63,7 +63,7 @@
       \param cal calendar to be printed
       \param helper is a pointer to the KOrg::CoreHelper object
     */
-    CalPrinter( QWidget *par, Calendar *cal, KOrg::CoreHelper *helper );
+    CalPrinter( QWidget *par, Calendar *cal, KOrg::CoreHelper *helper, bool uniqItem = false );
     virtual ~CalPrinter();
 
     void init( Calendar *calendar );
@@ -98,13 +98,14 @@
     QWidget *mParent;
     KConfig *mConfig;
     KOrg::CoreHelper *mCoreHelper;
+    bool mUniqItem;
 };
 
 class CalPrintDialog : public KDialog
 {
   Q_OBJECT
   public:
-    explicit CalPrintDialog( KOrg::PrintPlugin::List plugins, QWidget *parent = 0 );
+    explicit CalPrintDialog( KOrg::PrintPlugin::List plugins, QWidget *parent = 0, bool mUniqItem = false );
     virtual ~CalPrintDialog();
     KOrg::PrintPlugin *selectedPlugin();
     void setOrientation( CalPrinter::ePrintOrientation orientation );
Index: korganizer/koeditorgeneraltodo.cpp
===================================================================
--- korganizer/koeditorgeneraltodo.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koeditorgeneraltodo.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -277,11 +277,14 @@
 
   KOEditorGeneral::readIncidence( todo );
 
+  KDateTime dueDT = todo->dtDue();
+
   if ( todo->hasDueDate() ) {
-    enableAlarm( true );
-    KDateTime dueDT = todo->dtDue();
+    enableAlarm( true );    
     if ( todo->recurs() && date.isValid() ) {
-      dueDT.addDays( todo->dtDue().date().daysTo( date ) );
+      KDateTime dt( date, QTime( 0, 0, 0 ) );
+      dt = dt.addSecs( -1 );
+      dueDT.setDate( todo->recurrence()->getNextDateTime( dt ).date() );
     }
     if ( dueDT.isUtc() ) {
       dueDT = dueDT.toLocalZone();
@@ -303,8 +306,9 @@
 
   if ( todo->hasStartDate() ) {
     KDateTime startDT = todo->dtStart();
-    if ( todo->recurs() && date.isValid() ) {
-      startDT.setDate( date );
+    if ( todo->recurs() && date.isValid() && todo->hasDueDate() ) {
+      int days = todo->dtStart( true ).daysTo( todo->dtDue( true ) );
+      startDT.setDate( dueDT.date().addDays( -days ) );
     }
     if ( startDT.isUtc() ) {
       startDT = startDT.toLocalZone();
Index: korganizer/koeventview.h
===================================================================
--- korganizer/koeventview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koeventview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -113,6 +113,8 @@
      */
     static bool usesCompletedTodoPixmap( Todo *todo, const QDate &date );
 
+    bool supportsDateNavigation() const { return true; }
+
   public slots:
 
     /*
Index: korganizer/dbuscalendar.desktop
===================================================================
--- korganizer/dbuscalendar.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/dbuscalendar.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -60,6 +60,7 @@
 Comment[it]=Agenda con un'interfaccia D-Bus
 Comment[ja]=D-Bus インターフェースを持つカレンダー
 Comment[km]=កម្មវិធី​រៀបចំ​ជាមួយ​ចំណុច​ប្រទាក់ D-Bus
+Comment[lt]=Tvarkyklė su D-Bus sąsaja
 Comment[lv]=Organizators ar D-BUS saskarni
 Comment[nb]=Organizer med D-Bus-grensesnitt
 Comment[nds]=En Daagpleger mit en D-Bus-Koppelsteed
Index: korganizer/calendarview.h
===================================================================
--- korganizer/calendarview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/calendarview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -607,7 +607,18 @@
     void adaptNavigationUnits();
 
     /**
+      Returns the date of the selected incidence.
+
+      If the selected incidence is recurring, it will return
+      the date of the selected occurrence
+    **/
+    QDate activeIncidenceDate();
+
+    /**
       Returns the best guess at the current active date in the view.
+      This has nothing to do with selected incidences, use activeIncidenceDate()
+      for that, for example, agenda supports time selection and incidence selection
+      and they can have diferent dates.
 
       @param fallbackToToday If guessing doesn't work, some views will prefer
       today to be returned instead of the first select date in the day matrix,
Index: korganizer/koeditorgeneralevent.cpp
===================================================================
--- korganizer/koeditorgeneralevent.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koeditorgeneralevent.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -434,15 +434,11 @@
     if ( event->recurs() && date.isValid() ) {
       // Consider the active date when editing recurring Events.
       KDateTime kdt( date, QTime( 0, 0, 0 ), KSystemTimeZones::local() );
-      int diffDays = startDT.daysTo( kdt );
+      const int eventLength = startDT.daysTo( endDT );
       kdt = kdt.addSecs( -1 );
       startDT.setDate( event->recurrence()->getNextDateTime( kdt ).date() );
       if ( event->hasEndDate() ) {
-        endDT = endDT.addDays( diffDays );
-        if ( startDT > endDT ) {
-          startDT.setDate( event->recurrence()->getPreviousDateTime( kdt ).date() );
-          endDT = startDT.addDays( event->dtStart().daysTo( event->dtEnd() ) );
-        }
+        endDT.setDate( startDT.addDays( eventLength ).date() );
       } else {
         if ( event->hasDuration() ) {
           endDT = startDT.addSecs( event->duration().asSeconds() );
Index: korganizer/interfaces/korganizer/baseview.h
===================================================================
--- korganizer/interfaces/korganizer/baseview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/interfaces/korganizer/baseview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -87,7 +87,7 @@
       probably only select a single event at a time, but some may be able
       to select more than one.
     */
-    virtual DateList selectedDates() = 0;
+    virtual DateList selectedIncidenceDates() = 0;
 
     /**
        Returns the start of the selection, or an invalid QDateTime if there is no selection
@@ -138,6 +138,11 @@
      */
     virtual bool usesFullWindow();
 
+    /** Returns true if the view supports navigation through the date navigator
+        ( selecting a date range, changing month, changing year, etc. )
+     */
+    virtual bool supportsDateNavigation() const { return false; }
+
   public Q_SLOTS:
     /**
       Show incidences for the given date range. The date range actually shown
Index: korganizer/plugins/picoftheday/picoftheday.cpp
===================================================================
--- korganizer/plugins/picoftheday/picoftheday.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/plugins/picoftheday/picoftheday.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -149,6 +149,7 @@
   KIO::StoredTransferJob *const transferJob = static_cast<KIO::StoredTransferJob*>( job );
   mFileName = QString::fromUtf8( transferJob->data().data(), transferJob->data().size() );
   mFileName = mFileName.left( mFileName.indexOf( "<noinclude>" ) );
+  mFileName = mFileName.section( '|', 1, 1 );
   kDebug() << "POTD:" << mDate << ": got POTD file name:" << mFileName;
 
   if ( !mFileName.isEmpty() ) {
Index: korganizer/views/monthview/monthview.cpp
===================================================================
--- korganizer/views/monthview/monthview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/monthview/monthview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -125,10 +125,9 @@
   return 6 * 7;
 }
 
-DateList MonthView::selectedDates()
+DateList MonthView::selectedIncidenceDates()
 {
   DateList list;
-
   if ( mScene->selectedItem() ) {
     IncidenceMonthItem *tmp = qobject_cast<IncidenceMonthItem *>( mScene->selectedItem() );
     if ( tmp ) {
@@ -137,6 +136,8 @@
         list << selectedItemDate;
       }
     }
+  } else if( mScene->selectedCell()) {
+    list << mScene->selectedCell()->date();
   }
 
   return list;
Index: korganizer/views/monthview/monthscene.cpp
===================================================================
--- korganizer/views/monthview/monthscene.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/monthview/monthscene.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -676,10 +676,10 @@
   mSelectedItem = item;
   Q_ASSERT( tmp->incidence() );
 
-  if ( mMonthView && mMonthView->selectedDates().isEmpty() ) {
+  if ( mMonthView && mMonthView->selectedIncidenceDates().isEmpty() ) {
     emit incidenceSelected( tmp->incidence(), QDate() );
   } else {
-    emit incidenceSelected( tmp->incidence(), mMonthView->selectedDates().first() );
+    emit incidenceSelected( tmp->incidence(), mMonthView->selectedIncidenceDates().first() );
   }
 }
 
Index: korganizer/views/monthview/monthitem.cpp
===================================================================
--- korganizer/views/monthview/monthitem.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/monthview/monthitem.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -558,8 +558,8 @@
 QString IncidenceMonthItem::toolTipText() const
 {
   QDate date;
-  if ( monthScene()->mMonthView && !monthScene()->mMonthView->selectedDates().isEmpty() ) {
-    date = monthScene()->mMonthView->selectedDates().first();
+  if ( monthScene()->mMonthView && !monthScene()->mMonthView->selectedIncidenceDates().isEmpty() ) {
+    date = monthScene()->mMonthView->selectedIncidenceDates().first();
   }
   return IncidenceFormatter::toolTipStr(
     monthScene()->calendar(), mIncidence, date, true, KOPrefs::instance()->timeSpec() );
Index: korganizer/views/monthview/monthview.h
===================================================================
--- korganizer/views/monthview/monthview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/monthview/monthview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -51,7 +51,7 @@
     Incidence::List selectedIncidences();
 
     /** Returns dates of the currently selected events */
-    virtual DateList selectedDates();
+    virtual DateList selectedIncidenceDates();
 
     virtual QDateTime selectionStart();
 
Index: korganizer/views/todoview/kotodoview.h
===================================================================
--- korganizer/views/todoview/kotodoview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/todoview/kotodoview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -62,7 +62,7 @@
     virtual void setCalendar( Calendar *cal );
 
     virtual Incidence::List selectedIncidences();
-    virtual DateList selectedDates();
+    virtual DateList selectedIncidenceDates();
     virtual int currentDateCount() { return 0; }
 
     void setDocumentId( const QString & ) {}
@@ -120,10 +120,11 @@
     void purgeCompletedSignal();
     void unSubTodoSignal();
     void unAllSubTodoSignal();
+    void configChanged();
 
   private:
     QMenu *createCategoryPopupMenu();
-
+    void printTodo( bool preview );
     KOTodoViewView *mView;
     KOTodoModel *mModel;
     KOTodoViewSortFilterProxyModel *mProxyModel;
Index: korganizer/views/todoview/kotodoview.cpp
===================================================================
--- korganizer/views/todoview/kotodoview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/todoview/kotodoview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -276,7 +276,7 @@
   return ret;
 }
 
-DateList KOTodoView::selectedDates()
+DateList KOTodoView::selectedIncidenceDates()
 {
   // The todo view only lists todo's. It's probably not a good idea to
   // return something about the selected todo here, because it has got
@@ -486,10 +486,10 @@
 
   Todo *todo = static_cast<Todo *>( selection[0].data( KOTodoModel::TodoRole ).value<void *>() );
 
-  if ( selectedDates().isEmpty() ) {
+  if ( selectedIncidenceDates().isEmpty() ) {
     emit incidenceSelected( todo, QDate() );
   } else {
-    emit incidenceSelected( todo, selectedDates().first() );
+    emit incidenceSelected( todo, selectedIncidenceDates().first() );
   }
 }
 
@@ -517,7 +517,7 @@
   emit editIncidenceSignal( todo );
 }
 
-void KOTodoView::printTodo()
+void KOTodoView::printTodo( bool preview )
 {
 #ifndef KORG_NOPRINTER
   QModelIndexList selection = mView->selectionModel()->selectedRows();
@@ -528,7 +528,7 @@
   Todo *todo = static_cast<Todo *>( selection[0].data( KOTodoModel::TodoRole ).value<void *>() );
 
   KOCoreHelper helper;
-  CalPrinter printer( this, BaseView::calendar(), &helper );
+  CalPrinter printer( this, BaseView::calendar(), &helper, true );
   connect( this, SIGNAL(configChanged()), &printer, SLOT(updateConfig()) );
 
   Incidence::List selectedIncidences;
@@ -542,39 +542,19 @@
   }
 
   printer.print( KOrg::CalPrinterBase::Incidence,
-                 todoDate.date(), todoDate.date(), selectedIncidences, false );
+                 todoDate.date(), todoDate.date(), selectedIncidences, preview );
 
 #endif
 }
 
-void KOTodoView::printPreviewTodo()
+void KOTodoView::printTodo()
 {
-#ifndef KORG_NOPRINTER
-  QModelIndexList selection = mView->selectionModel()->selectedRows();
-  if ( selection.size() != 1 ) {
-    return;
-  }
+  printTodo( false );
+}
 
-  Todo *todo = static_cast<Todo *>( selection[0].data( KOTodoModel::TodoRole ).value<void *>() );
-
-  KOCoreHelper helper;
-  CalPrinter printer( this, BaseView::calendar(), &helper );
-  connect( this, SIGNAL(configChanged()), &printer, SLOT(updateConfig()) );
-
-  Incidence::List selectedIncidences;
-  selectedIncidences.append( todo );
-
-  KDateTime todoDate;
-  if ( todo->hasStartDate() ) {
-    todoDate = todo->dtStart();
-  } else {
-    todoDate = todo->dtDue();
-  }
-
-  printer.print( KOrg::CalPrinterBase::Incidence,
-                 todoDate.date(), todoDate.date(), selectedIncidences, true );
-
-#endif
+void KOTodoView::printPreviewTodo()
+{
+  printTodo( true );
 }
 
 void KOTodoView::deleteTodo()
Index: korganizer/views/whatsnextview/kowhatsnextview.h
===================================================================
--- korganizer/views/whatsnextview/kowhatsnextview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/whatsnextview/kowhatsnextview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -55,8 +55,10 @@
 
     virtual int currentDateCount();
     virtual Incidence::List selectedIncidences() { return Incidence::List(); }
-    DateList selectedDates() { return DateList(); }
+    DateList selectedIncidenceDates() { return DateList(); }
 
+    bool supportsDateNavigation() const { return true; }
+
   public slots:
     virtual void updateView();
     virtual void showDates( const QDate &start, const QDate &end );
Index: korganizer/views/timelineview/kotimelineview.h
===================================================================
--- korganizer/views/timelineview/kotimelineview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/timelineview/kotimelineview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -51,7 +51,7 @@
     ~KOTimelineView();
 
     virtual KCal::Incidence::List selectedIncidences();
-    virtual KCal::DateList selectedDates();
+    virtual KCal::DateList selectedIncidenceDates();
     virtual int currentDateCount();
     virtual void showDates( const QDate &, const QDate & );
     virtual void showIncidences( const KCal::Incidence::List &incidenceList, const QDate &date );
Index: korganizer/views/timelineview/kotimelineview.cpp
===================================================================
--- korganizer/views/timelineview/kotimelineview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/timelineview/kotimelineview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -84,7 +84,7 @@
 }
 
 /*virtual*/
-KCal::DateList KOTimelineView::selectedDates()
+KCal::DateList KOTimelineView::selectedIncidenceDates()
 {
   return KCal::DateList();
 }
Index: korganizer/views/timespentview/kotimespentview.h
===================================================================
--- korganizer/views/timespentview/kotimespentview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/timespentview/kotimespentview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -47,7 +47,7 @@
     {
       return Incidence::List();
     }
-    DateList selectedDates()
+    DateList selectedIncidenceDates()
     {
       return DateList();
     }
Index: korganizer/views/journalview/kojournalview.h
===================================================================
--- korganizer/views/journalview/kojournalview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/journalview/kojournalview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -47,7 +47,7 @@
 
     virtual int currentDateCount();
     virtual Incidence::List selectedIncidences();
-    DateList selectedDates() { return DateList(); }
+    DateList selectedIncidenceDates() { return DateList(); }
     void appendJournal( Journal *journal, const QDate &dt );
 
     /** documentation in baseview.h */
Index: korganizer/views/journalview/journalview.cpp
===================================================================
--- korganizer/views/journalview/journalview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/journalview/journalview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -231,7 +231,7 @@
 #ifndef KORG_NOPRINTER
   if ( mJournal ) {
     KOCoreHelper helper;
-    CalPrinter printer( this, mCalendar, &helper );
+    CalPrinter printer( this, mCalendar, &helper, true );
     connect( this, SIGNAL(configChanged()), &printer, SLOT(updateConfig()) );
 
     Incidence::List selectedIncidences;
@@ -310,12 +310,11 @@
   cursor.insertBlock();
   cursor.insertBlock();
   cursor.setBlockCharFormat( bodyFormat );
+  const QString description = mJournal->description();
   if ( mJournal->descriptionIsRich() ) {
-    QString description = mJournal->description();
     mBrowser->insertHtml( description );
   } else {
-    mBrowser->insertPlainText( mJournal->description() );
+    mBrowser->insertPlainText( description );
   }
-  cursor.insertBlock();
   setReadOnly( mJournal->isReadOnly() );
 }
Index: korganizer/views/multiagendaview/multiagendaview.cpp
===================================================================
--- korganizer/views/multiagendaview/multiagendaview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/multiagendaview/multiagendaview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -276,11 +276,11 @@
   return list;
 }
 
-DateList MultiAgendaView::selectedDates()
+DateList MultiAgendaView::selectedIncidenceDates()
 {
   DateList list;
   foreach ( KOAgendaView *agendaView, mAgendaViews ) {
-    list += agendaView->selectedDates();
+    list += agendaView->selectedIncidenceDates();
   }
   return list;
 }
Index: korganizer/views/multiagendaview/multiagendaview.h
===================================================================
--- korganizer/views/multiagendaview/multiagendaview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/multiagendaview/multiagendaview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -48,7 +48,7 @@
     ~MultiAgendaView();
 
     Incidence::List selectedIncidences();
-    DateList selectedDates();
+    DateList selectedIncidenceDates();
     int currentDateCount();
     int maxDatesHint();
 
Index: korganizer/views/listview/kolistview.cpp
===================================================================
--- korganizer/views/listview/kolistview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/listview/kolistview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -304,7 +304,7 @@
   return eventList;
 }
 
-DateList KOListView::selectedDates()
+DateList KOListView::selectedIncidenceDates()
 {
   return mSelectedDates;
 }
Index: korganizer/views/listview/kolistview.h
===================================================================
--- korganizer/views/listview/kolistview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/listview/kolistview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -74,7 +74,7 @@
     virtual int maxDatesHint();
     virtual int currentDateCount();
     virtual Incidence::List selectedIncidences();
-    virtual DateList selectedDates();
+    virtual DateList selectedIncidenceDates();
 
     void showDates( bool show );
 
Index: korganizer/views/agendaview/koagendaview.cpp
===================================================================
--- korganizer/views/agendaview/koagendaview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/agendaview/koagendaview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -611,7 +611,7 @@
   return selected;
 }
 
-DateList KOAgendaView::selectedDates()
+DateList KOAgendaView::selectedIncidenceDates()
 {
   DateList selected;
   QDate qd;
@@ -1195,6 +1195,10 @@
     if ( todo ) {
       QTime t = todo->dtDue().toTimeSpec( KOPrefs::instance()->timeSpec() ).time();
 
+      if ( t == QTime( 0, 0 ) ) {
+        t = QTime( 23, 59 );
+      }
+
       int halfHour = 1800;
       if ( t.addSecs( -halfHour ) < t ) {
         startY = mAgenda->timeToY( t.addSecs( -halfHour ) );
@@ -1350,7 +1354,7 @@
   }
 
   if ( incidence->recurs() ) {
-    int eventDuration = incDtStart.daysTo( incDtEnd );
+    int eventDuration = event ? incDtStart.daysTo( incDtEnd ) : 0;
 
     // if there's a multiday event that starts before firstVisibleDateTime but ends after
     // lets include it. timesInInterval() ignores incidences that aren't totaly inside
@@ -1367,6 +1371,13 @@
     if ( todo && todo->hasDueDate() && !todo->isOverdue() ) {
       // If it's not overdue it will be shown at the original date (not today)
       dateToAdd = todo->dtDue().toTimeSpec( KOPrefs::instance()->timeSpec() );
+
+      // To-dos are drawn with the bottom of the rectangle at dtDue
+      // if dtDue is at 00:00, then it should be displayed in the previous day, at 23:59
+      if ( dateToAdd.time() == QTime( 0, 0 ) ) {
+        dateToAdd = dateToAdd.addSecs( -1 );
+      }
+
       incidenceEnd = dateToAdd;
     } else if ( event ) {
       dateToAdd = incDtStart;
@@ -1375,10 +1386,11 @@
       if ( dateToAdd.isDateOnly() ) {
         // so comparisons with < > actually work
         dateToAdd.setTime( QTime( 0, 0 ) );
+        incidenceEnd.setTime( QTime( 23, 59 ) );
       }
     }
 
-    if  ( dateToAdd <= lastVisibleDateTime && incidenceEnd >= firstVisibleDateTime ) {
+    if  ( dateToAdd <= lastVisibleDateTime && incidenceEnd > firstVisibleDateTime ) {
       dateTimeList += dateToAdd;
     }
   }
Index: korganizer/views/agendaview/koagendaview.h
===================================================================
--- korganizer/views/agendaview/koagendaview.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/views/agendaview/koagendaview.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -113,7 +113,7 @@
     virtual Incidence::List selectedIncidences();
 
     /** returns the currently selected incidence's dates */
-    virtual DateList selectedDates();
+    virtual DateList selectedIncidenceDates();
 
     /** return the default start/end date/time for new events   */
     virtual bool eventDurationHint( QDateTime &startDt, QDateTime &endDt, bool &allDay );
Index: korganizer/timelabelszone.cpp
===================================================================
--- korganizer/timelabelszone.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/timelabelszone.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -108,7 +108,7 @@
   foreach ( TimeLabels *timeLabel, mTimeLabelsList ) {
     timeLabel->updateConfig();
     timeLabel->positionChanged();
-    timeLabel->repaint();
+    timeLabel->repaintContents();
   }
 }
 
Index: korganizer/korganizer.upd
===================================================================
--- korganizer/korganizer.upd	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/korganizer.upd	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -53,3 +53,26 @@
 Key=ShowAllDayTodo,ShowTodosAgendaView
 Key=ShowAllDayTodo,ShowTodosMonthView
 RemoveKey=ShowAllDayTodo
+#
+#
+#
+Id=korganizer_4.4_MoveFreeBusy
+File=korganizerrc
+Group=FreeBusy,FreeBusy Publish
+Key=FreeBusyPublishAuto,FreeBusyPublishAuto
+Key=FreeBusyPublishPassword,FreeBusyPublishPassword
+Key=FreeBusyPublishSavePassword,FreeBusyPublishSavePassword
+Key=FreeBusyPublishUser,FreeBusyPublishUser
+Key=FreeBusyPublishDelay,FreeBusyPublishDelay
+Key=FreeBusyPublishDays,FreeBusyPublishDays
+Key=FreeBusyPublishUrl,FreeBusyPublishUrl
+
+Group=FreeBusy,FreeBusy Retrieve
+Key=FreeBusyFullDomainRetrieval,FreeBusyFullDomainRetrieval
+Key=FreeBusyRetrieveSavePassword,FreeBusyRetrieveSavePassword
+Key=FreeBusyRetrieveUrl,FreeBusyRetrieveUrl
+Key=FreeBusyRetrieveUser,FreeBusyRetrieveUser
+Key=FreeBusyRetrieveAuto,FreeBusyRetrieveAuto
+Key=FreeBusyRetrievePassword,FreeBusyRetrievePassword
+
+RemoveGroup=FreeBusy
Index: korganizer/calendarview.cpp
===================================================================
--- korganizer/calendarview.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/calendarview.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -317,10 +317,10 @@
       return curView->selectionStart().date();
     }
 
-    // Try the view's selectedDates()
-    if ( !curView->selectedDates().isEmpty() ) {
-      if ( curView->selectedDates().first().isValid() ) {
-        return curView->selectedDates().first();
+    // Try the view's selectedIncidenceDates()
+    if ( !curView->selectedIncidenceDates().isEmpty() ) {
+      if ( curView->selectedIncidenceDates().first().isValid() ) {
+        return curView->selectedIncidenceDates().first();
       }
     }
   }
@@ -333,6 +333,19 @@
   }
 }
 
+QDate CalendarView::activeIncidenceDate()
+{
+  KOrg::BaseView *curView = mViewManager->currentView();
+  if ( curView ) {
+    DateList dates = curView->selectedIncidenceDates();
+    if ( !dates.isEmpty() ) {
+      return dates.first();
+    }
+  }
+
+  return QDate();
+}
+
 QDate CalendarView::startDate()
 {
   DateList dates = mDateNavigator->selectedDates();
@@ -870,16 +883,15 @@
     if ( !aView->selectedIsAllDay() ) {
       time = aView->selectionStart().time();
     }
-  } else if ( curView == mView && !mView->selectedDates().isEmpty() ) {
-    date = mView->selectedDates().first();
-  } else {
-    // default to the selected date from the navigator
-    if ( !mDateNavigator->selectedDates().isEmpty() ) {
+  } else if ( curView == mView && mView->selectionStart().isValid() ) {
+    date = mView->selectionStart().date();
+  } else if ( !mDateNavigator->selectedDates().isEmpty() &&
+              curView->supportsDateNavigation() ) {
+      // default to the selected date from the navigator
       date = mDateNavigator->selectedDates().first();
-    }
   }
 
-  if ( !date.isValid() ) {
+  if ( !date.isValid() && curView->supportsDateNavigation() ) {
     KMessageBox::sorry(
       this,
       i18n( "Paste failed: unable to determine a valid target date." ) );
@@ -931,17 +943,16 @@
 void CalendarView::dateTimesForNewEvent( QDateTime &startDt, QDateTime &endDt,
                                          bool &allDay )
 {
-  if ( !startDt.isValid() ) {
-    startDt.setDate( activeDate() );
+  mViewManager->currentView()->eventDurationHint( startDt, endDt, allDay );
+
+  if ( !startDt.isValid() || !endDt.isValid() ) {
+    startDt.setDate( activeDate( true ) );
     startDt.setTime( KOPrefs::instance()->mStartTime.time() );
-  }
-  if ( !endDt.isValid() ) {
     int addSecs =
       ( KOPrefs::instance()->mDefaultDuration.time().hour() * 3600 ) +
       ( KOPrefs::instance()->mDefaultDuration.time().minute() * 60 );
     endDt = startDt.addSecs( addSecs );
   }
-  mViewManager->currentView()->eventDurationHint( startDt, endDt, allDay );
 }
 
 KOEventEditor *CalendarView::newEventEditor( const QDateTime &startDtParam,
@@ -2322,7 +2333,7 @@
   connectIncidenceEditor( incidenceEditor );
 
   mDialogList.insert( incidence, incidenceEditor );
-  incidenceEditor->editIncidence( incidence, activeDate(), mCalendar );
+  incidenceEditor->editIncidence( incidence, activeIncidenceDate(), mCalendar );
   incidenceEditor->show();
   return true;
 }
Index: korganizer/koviewmanager.h
===================================================================
--- korganizer/koviewmanager.h	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koviewmanager.h	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -60,6 +60,16 @@
 {
   Q_OBJECT
   public:
+
+    enum AgendaMode {
+      AGENDA_NONE,
+      AGENDA_DAY,
+      AGENDA_WORK_WEEK,
+      AGENDA_WEEK,
+      AGENDA_NEXTX,
+      AGENDA_OTHER // for example, showing 8 days
+    };
+
     explicit KOViewManager( CalendarView * );
     virtual ~KOViewManager();
 
@@ -130,6 +140,12 @@
 
     void setUpdateNeeded();
 
+    /**
+      If the agenda view is selected it returns the current range mode:
+      week, work week, day or nextX days
+     */
+    AgendaMode agendaMode() const { return mAgendaMode; }
+
   private slots:
     void currentAgendaViewTabChanged( QWidget * );
 
@@ -150,6 +166,8 @@
 
     KOrg::BaseView *mLastEventView;
     KTabWidget *mAgendaViewTabs;
+
+    AgendaMode mAgendaMode;
 };
 
 #endif
Index: korganizer/koeditoralarms_base.ui
===================================================================
--- korganizer/koeditoralarms_base.ui	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koeditoralarms_base.ui	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -61,7 +61,7 @@
            <number>1</number>
           </property>
           <property name="maximum">
-           <number>999999999</number>
+           <number>99999</number>
           </property>
          </widget>
         </item>
Index: korganizer/koeditorattachments.cpp
===================================================================
--- korganizer/koeditorattachments.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koeditorattachments.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -569,6 +569,18 @@
   menu.addSeparator();
   cancelAction = menu.addAction( i18nc( "@action:inmenu", "C&ancel" ) );
 
+  QByteArray data;
+  QString mimeType;
+  QString label;
+
+  if(!mimeData->formats().isEmpty() && !probablyWeHaveUris) {
+    data=mimeData->data( mimeData->formats().first() );
+    mimeType = mimeData->formats().first();
+    if( KMimeType::mimeType( mimeData->formats().first() ) )
+      label = KMimeType::mimeType( mimeData->formats().first() )->name();
+  }
+
+
   QAction *ret = menu.exec( QCursor::pos() );
   if ( linkAction == ret ) {
     QStringList::ConstIterator jt = labels.constBegin();
@@ -585,9 +597,9 @@
         connect( job, SIGNAL(result(KJob *)), SLOT(downloadComplete(KJob *)) );
       }
     } else { // we take anything
-      addDataAttachment( mimeData->data( mimeData->formats().first() ),
-                         mimeData->formats().first(),
-                         KMimeType::mimeType( mimeData->formats().first() )->name() );
+      addDataAttachment( data,
+                         mimeType,
+                         label );
     }
   }
 }
Index: korganizer/koeventpopupmenu.cpp
===================================================================
--- korganizer/koeventpopupmenu.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koeventpopupmenu.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -114,9 +114,8 @@
     delete mMoveToCalendarMenu;
     mMoveToCalendarMenu = 0;
   }
+  if ( mCurrentIncidence /*&& qd.isValid()*/ ) {
 
-  if ( mCurrentIncidence && qd.isValid() ) {
-
     if ( mCurrentIncidence->recurs() ) {
       KDateTime thisDateTime( qd, KOPrefs::instance()->timeSpec() );
       bool isLastOccurrence =
Index: korganizer/koviewmanager.cpp
===================================================================
--- korganizer/koviewmanager.cpp	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ korganizer/koviewmanager.cpp	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -66,6 +66,7 @@
   mAgendaViewTabs = 0;
   mTimeSpentView = 0;
   mMonthView = 0;
+  mAgendaMode = AGENDA_NONE;
 }
 
 KOViewManager::~KOViewManager()
@@ -100,7 +101,27 @@
   } else if ( view == QLatin1String( "Month" ) ) {
     showMonthView();
   } else {
-    showAgendaView();
+    mAgendaMode = AgendaMode( generalConfig.readEntry( "Agenda Mode", int( AGENDA_OTHER ) ) );
+
+    switch ( mAgendaMode ) {
+      case AGENDA_WORK_WEEK:
+        showWorkWeekView();
+        break;
+      case AGENDA_WEEK:
+        showWeekView();
+        break;
+      case AGENDA_NEXTX:
+        showNextXView();
+        break;
+      case AGENDA_DAY:
+        showDayView();
+        break;
+      case AGENDA_NONE:
+        // Someone has been playing with the config file.
+      default:
+        mAgendaMode = AGENDA_OTHER;
+        showAgendaView();
+    } 
   }
 }
 
@@ -125,6 +146,7 @@
     view = QLatin1String( "Month" );
   } else {
     view = QLatin1String( "Agenda" );
+    generalConfig.writeEntry( "Agenda Mode", int( mAgendaMode ) );
   }
 
   generalConfig.writeEntry( "Current View", view );
@@ -452,6 +474,7 @@
 
 void KOViewManager::showDayView()
 {
+  mAgendaMode = AGENDA_DAY;
   QDate date = mMainView->activeDate();
   showAgendaView();
   mMainView->dateNavigator()->selectDate( date );
@@ -460,6 +483,7 @@
 void KOViewManager::showWorkWeekView()
 {
   if ( KOGlobals::self()->getWorkWeekMask() != 0 ) {
+    mAgendaMode = AGENDA_WORK_WEEK;
     QDate date = mMainView->activeDate();
     showAgendaView();
     mMainView->dateNavigator()->selectWorkWeek( date );
@@ -473,6 +497,7 @@
 
 void KOViewManager::showWeekView()
 {
+  mAgendaMode = AGENDA_WEEK;
   QDate date = mMainView->activeDate();
   showAgendaView();
   mMainView->dateNavigator()->selectWeek( date );
@@ -480,6 +505,7 @@
 
 void KOViewManager::showNextXView()
 {
+  mAgendaMode = AGENDA_NEXTX;
   showAgendaView();
   mMainView->dateNavigator()->selectDates( QDate::currentDate(),
                                            KOPrefs::instance()->mNextXDays );
@@ -550,7 +576,7 @@
 {
   QDate qd;
   if ( mCurrentView ) {
-    DateList qvl = mCurrentView->selectedDates();
+    DateList qvl = mCurrentView->selectedIncidenceDates();
     if ( !qvl.isEmpty() ) {
       qd = qvl.first();
     }
Index: examples/mailreader/Messages.sh
===================================================================
--- examples/mailreader/Messages.sh	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ examples/mailreader/Messages.sh	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -1,4 +1,4 @@
 #! /usr/bin/env bash
 $EXTRACTRC `find . -name \*.rc -o -name \*.ui -o -name \*.kcfg` >> rc.cpp
 $XGETTEXT *.cpp -o $podir/mailreader.pot
-rm -f *.cpp
+rm -f rc.cpp
Index: examples/mailreader/kmail-plasma/plasma-applet-kpapplet.desktop
===================================================================
--- examples/mailreader/kmail-plasma/plasma-applet-kpapplet.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ examples/mailreader/kmail-plasma/plasma-applet-kpapplet.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -80,6 +80,7 @@
 Comment[it]=La tua posta elettronica
 Comment[ja]=あなたのメール
 Comment[km]=អ៊ីមែល​របស់​អ្នក
+Comment[lt]=Jūsų e. pašto adresai
 Comment[lv]=viesturs.zarins@mii.lu.lv
 Comment[nb]=Dine e-poster
 Comment[nds]=Dien Nettpost
Index: examples/mailreader/mailreader.desktop
===================================================================
--- examples/mailreader/mailreader.desktop	(.../tags/KDE/4.4.1/kdepim)	(wersja 1104128)
+++ examples/mailreader/mailreader.desktop	(.../branches/KDE/4.4/kdepim)	(wersja 1104128)
@@ -14,6 +14,7 @@
 Name[it]=mailreader
 Name[ja]=メールリーダー
 Name[km]=កម្មវិធីអាន​សំបុត្រ
+Name[lt]=pašto skaityklė
 Name[lv]=pasta lasītājs
 Name[nb]=e-postleser
 Name[nds]=Nettpostkieker
@@ -49,6 +50,7 @@
 GenericName[it]=Un'applicazione KDE4
 GenericName[ja]=KDE4 アプリケーション
 GenericName[km]=កម្មវិធី​របស់ KDE4
+GenericName[lt]=KDE4 programa
 GenericName[lv]=KDE4 lietotne
 GenericName[nb]=Et KDE4-program
 GenericName[nds]=En Programm för KDE 4

Zmiany atrybutów dla: .
___________________________________________________________________
Dodane: svn:externals
   + 


