--- CMakeLists.txt.old	2008-07-25 13:28:52.000000000 +0200
+++ CMakeLists.txt	2008-07-25 13:29:01.000000000 +0200
@@ -242,7 +242,7 @@
   endif(NOT WIN32)
 
 #enable korn for 4.2
-#  macro_optional_add_subdirectory(korn)
+  macro_optional_add_subdirectory(korn)
   add_subdirectory(kresources)
   macro_optional_add_subdirectory(ktnef)
   macro_optional_add_subdirectory(kontact)
--- doc/CMakeLists.txt.old	2008-07-25 13:29:23.000000000 +0200
+++ doc/CMakeLists.txt	2008-07-25 13:29:30.000000000 +0200
@@ -29,7 +29,7 @@
   add_subdirectory(kleopatra)
 endif(QGPGME_FOUND)
 # disable korn for KDE 4.1
- #add_subdirectory(korn)
+add_subdirectory(korn)
 add_subdirectory(ktnef)
 add_subdirectory(kwatchgnupg)
 
