Index: akregator/src/aboutdata.h
===================================================================
--- akregator/src/aboutdata.h	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ akregator/src/aboutdata.h	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -28,7 +28,7 @@
 #include "akregator_export.h"
 #include <kaboutdata.h>
 
-#define AKREGATOR_VERSION "1.5.3"
+#define AKREGATOR_VERSION "1.5.4"
 
 namespace Akregator {
 /**
Index: kresources/slox/sloxbase.h
===================================================================
--- kresources/slox/sloxbase.h	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kresources/slox/sloxbase.h	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -34,6 +34,7 @@
       ClientId,
       FolderId,
       LastSync,
+      LastModified,
       ObjectType,
       ObjectStatus,
       CreatedBy,
Index: kresources/slox/kcalresourceslox.cpp
===================================================================
--- kresources/slox/kcalresourceslox.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kresources/slox/kcalresourceslox.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -334,6 +334,11 @@
   WebdavHandler::addSloxElement( this, doc, prop, fieldName( ClientId ),
                                  mUploadedIncidence->uid() );
 
+  if ( type() == "ox" ) {
+    const QString lastModified = mUploadedIncidence->customProperty( "SLOX", "LastModified" );
+    WebdavHandler::addSloxElement( this, doc, prop, fieldName( LastModified ), lastModified );
+  }
+
   if ( mUploadIsDelete ) {
     if ( mUploadedIncidence->type() == "Event" ) {
       url.setPath( "/servlet/webdav.calendar/" + sloxId );
@@ -347,6 +352,10 @@
 
     if ( type() == "ox" ) {
       WebdavHandler::addSloxElement( this, doc, prop, "method", "DELETE" );
+      if ( mUploadedIncidence->type() == "Event" )
+        WebdavHandler::addSloxElement( this, doc, prop, fieldName( FolderId ), mPrefs->calendarFolderId() );
+      else if ( mUploadedIncidence->type() == "Todo" )
+        WebdavHandler::addSloxElement( this, doc, prop, fieldName( FolderId ), mPrefs->taskFolderId() );
     } else {
       QDomElement remove = WebdavHandler::addElement( doc, pu, "D:remove" );
       QDomElement prop = WebdavHandler::addElement( doc, remove, "D:prop" );
@@ -694,6 +703,8 @@
     parseReadRightsAttribute( e, incidence );
   } else if ( tag == fieldName( Categories ) ) {
     incidence->setCategories( text.split( QRegExp(",\\s*") ) );
+  } else if ( tag == fieldName( LastModified ) && type() == "ox" ) {
+    incidence->setCustomProperty( "SLOX", "LastModified", text );
   }
 }
 
@@ -1162,6 +1173,16 @@
           i->setUid( uid );
           i->setCustomProperty( "SLOX", "ID", sloxId );
 
+          if ( type() == "ox" ) {
+            // Update the last_modified property
+            const QDomNode lastModifiedNode = prop.namedItem( fieldName( LastModified ) );
+            if ( !lastModifiedNode.isNull() ) {
+              const QDomElement lastModifiedElement = lastModifiedNode.toElement();
+              const QString lastModified = lastModifiedElement.text();
+              i->setCustomProperty( "SLOX", "LastModified", lastModified );
+            }
+          }
+
           disableChangeNotification();
           calendar()->deleteIncidence( mUploadedIncidence );
           calendar()->addIncidence( i );
Index: kresources/slox/sloxbase.cpp
===================================================================
--- kresources/slox/sloxbase.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kresources/slox/sloxbase.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -29,6 +29,7 @@
   {"clientid", "client_id"},
   {"folderid", "folder_id"},
   {"lastsync", "lastsync"},
+  {"n/a", "last_modified"},
   {"objecttype", "objectmode"},
   {"sloxstatus", "object_status"},
   {"createfrom", "created_by"},
Index: kalarm/lib/lineedit.cpp
===================================================================
--- kalarm/lib/lineedit.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kalarm/lib/lineedit.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -88,8 +88,8 @@
 QString LineEdit::text() const
 {
 	if (mType == Url)
-		return KShell::tildeExpand(QLineEdit::text());
-	return QLineEdit::text();
+		return KShell::tildeExpand(KLineEdit::text());
+	return KLineEdit::text();
 }
 
 void LineEdit::setText(const QString& text)
Index: kaddressbook/xxport/gmx_xxport.desktop
===================================================================
--- kaddressbook/xxport/gmx_xxport.desktop	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kaddressbook/xxport/gmx_xxport.desktop	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -68,6 +68,7 @@
 Comment[pl]=Wtyczka do importu i eksportu wizytówek w formacie książki adresowej GMX
 Comment[pt]=Um 'plugin' para importar e exportar os contactos no formato GMX
 Comment[pt_BR]=Plug-in para importar e exportar contatos no formato de livro de endereços do GMX
+Comment[ru]=Импорт и экспорт контактов в формате GMX
 Comment[sv]=Insticksprogram för import och export av kontakter med GMX-adressboksformat
 Comment[tr]=GMX'in adres defteri biçimini içeriye ya da dışarıya aktarma eklentisi
 Comment[uk]=Втулок для імпорту та експорту контактів у форматі адресних книг GMX
Index: kaddressbook/xxport/vcard_xxport.desktop
===================================================================
--- kaddressbook/xxport/vcard_xxport.desktop	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kaddressbook/xxport/vcard_xxport.desktop	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -93,7 +93,7 @@
 Comment[pl]=Wtyczka do importu i eksportu wizytówek w formacie vCard
 Comment[pt]=Um 'plugin' para importar e exportar os contactos no formato vCard
 Comment[pt_BR]=Plug-in para importar e exportar contatos no formato vCard
-Comment[ru]=Импорта и экспорт контактов в формате vCard
+Comment[ru]=Импорт и экспорт контактов в формате vCard
 Comment[sk]=Plugin pre import a export kontaktov v vCard formáte
 Comment[sl]=Vstavek za uvoz in izvoz stikov v obliki vCard
 Comment[sv]=Insticksprogram för import och export av kontakter med vCard-format
Index: kaddressbook/xxport/pab_xxport.desktop
===================================================================
--- kaddressbook/xxport/pab_xxport.desktop	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kaddressbook/xxport/pab_xxport.desktop	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -24,6 +24,7 @@
 Name[pl]=Wtyczka KAB do importu Osobistej książki adresowej MS Exchange - XXPort Plugin
 Name[pt]='Plugin' XXPort do Livro de Endereços Pessoal do MS Exchange para o KAB
 Name[pt_BR]=Plug-in do KAB para importação/exportação de livros de endereços pessoais do MS Exchange
+Name[ru]=Импорт контактов из персональной адресной книги MS Exchange
 Name[sv]=Adressbokens överföringsinsticksprogram för MS Exchange personlig adressbok
 Name[tr]=KAB MS Exchange Kişisel Adres Defteri XXPort Eklentisi
 Name[uk]=Втулок KAB для обміну з персональною адресною книгою MS Exchange
Index: kaddressbook/xxport/gnokii_xxport.desktop
===================================================================
--- kaddressbook/xxport/gnokii_xxport.desktop	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ kaddressbook/xxport/gnokii_xxport.desktop	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -78,6 +78,7 @@
 Comment[pl]=Wtyczka do importowania i eksportowania wpisów z książki adresowej telefonu komórkowego
 Comment[pt]=Um 'plugin' para importar e exportar contactos da agenda do telemóvel
 Comment[pt_BR]=Plug-in para importar e exportar entradas do livro de endereços do celular
+Comment[ru]=Импорт и экспорт контактов адресной книги
 Comment[sv]=Insticksprogram för import och export av adressboksposter till mobiltelefon
 Comment[tr]=Adres defteri girdilerini cep telefonuna aktarma ve cep telefonundan alma eklentisi
 Comment[uk]=Втулок для імпорту та експорту записів у адресній книзі мобільних телефонів
Index: libkdepim/categoryeditdialog.cpp
===================================================================
--- libkdepim/categoryeditdialog.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ libkdepim/categoryeditdialog.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -69,6 +69,8 @@
 
   fillList();
 
+  mWidgets->mCategories->setFocus();
+
   connect( mWidgets->mCategories, SIGNAL(currentItemChanged(QTreeWidgetItem *,QTreeWidgetItem *)),
            SLOT(editItem(QTreeWidgetItem *)) );
   connect( mWidgets->mCategories, SIGNAL(itemSelectionChanged()),
Index: libkdepim/categoryselectdialog.cpp
===================================================================
--- libkdepim/categoryselectdialog.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ libkdepim/categoryselectdialog.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -188,7 +188,7 @@
   lay->addWidget( mWidgets );
 
   mWidgets->setCategories();
-  mWidgets->setFocus();
+  mWidgets->listView()->setFocus();
 
   connect( mWidgets, SIGNAL(editCategories()), SIGNAL(editCategories()) );
 
@@ -243,10 +243,4 @@
   mWidgets->setSelected( selList );
 }
 
-void CategorySelectDialog::enterEvent( QEvent *event )
-{
-  Q_UNUSED( event );
-  mWidgets->listView()->setFocus();
-}
-
 #include "categoryselectdialog.moc"
Index: libkdepim/categoryselectdialog.h
===================================================================
--- libkdepim/categoryselectdialog.h	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ libkdepim/categoryselectdialog.h	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -88,9 +88,6 @@
     void categoriesSelected( const QStringList & );
     void editCategories();
 
-  protected:
-    /*reimp*/void enterEvent( QEvent * );
-
   private:
     CategorySelectWidget *mWidgets;
 
Index: akonadi/resources/kolabproxy/kolabproxyresource.desktop
===================================================================
Index: akonadi/resources/kabc/kabcresource.desktop
===================================================================
Index: korganizer/views/monthview/monthscene.cpp
===================================================================
--- korganizer/views/monthview/monthscene.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ korganizer/views/monthview/monthscene.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -61,6 +61,8 @@
   mReadonlyPixmap  = KOGlobals::self()->smallIcon( "object-locked" );
   mReplyPixmap     = KOGlobals::self()->smallIcon( "mail-reply-sender" );
   mHolidayPixmap   = KOGlobals::self()->smallIcon( "favorites" );
+
+  setSceneRect( 0, 0, parent->width(), parent->height() );
 }
 
 MonthScene::~MonthScene()
Index: korganizer/views/todoview/kotodoviewquickaddline.cpp
===================================================================
--- korganizer/views/todoview/kotodoviewquickaddline.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ korganizer/views/todoview/kotodoviewquickaddline.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -25,6 +25,7 @@
 
 #include "kotodoviewquickaddline.h"
 
+#include <KLocale>
 #include <QWidget>
 #include <QKeyEvent>
 
@@ -33,6 +34,9 @@
 {
   connect( this, SIGNAL(returnPressed()),
            this, SLOT(returnPressed()) );
+
+  mClickMessage = i18n( "Click to add a new to-do" );
+  setToolTip( mClickMessage );
 }
 
 void KOTodoViewQuickAddLine::keyPressEvent( QKeyEvent *event )
@@ -49,4 +53,14 @@
   emit returnPressed( mModifiers );
 }
 
+void KOTodoViewQuickAddLine::resizeEvent ( QResizeEvent * event )
+{
+  KLineEdit::resizeEvent( event );
+
+  setClickMessage( fontMetrics().elidedText(
+                     mClickMessage,
+                     Qt::ElideRight,
+                     width() - clearButtonUsedSize().width() ) );
+}
+
 #include "kotodoviewquickaddline.moc"
Index: korganizer/views/todoview/kotodoviewquickaddline.h
===================================================================
--- korganizer/views/todoview/kotodoviewquickaddline.h	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ korganizer/views/todoview/kotodoviewquickaddline.h	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -42,6 +42,7 @@
 
   protected:
     void keyPressEvent( QKeyEvent *event );
+    void resizeEvent ( QResizeEvent * event );
 
   Q_SIGNALS:
     void returnPressed( Qt::KeyboardModifiers modifiers );
@@ -51,6 +52,7 @@
 
   private:
     Qt::KeyboardModifiers mModifiers;
+    QString mClickMessage;
 };
 
 #endif
Index: korganizer/views/todoview/kotodoview.cpp
===================================================================
--- korganizer/views/todoview/kotodoview.cpp	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ korganizer/views/todoview/kotodoview.cpp	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -112,8 +112,6 @@
 
   mQuickAdd = new KOTodoViewQuickAddLine( this );
   mQuickAdd->setClearButtonShown( true );
-  QString clickText = i18n( "Click to add a new to-do" );
-  mQuickAdd->setToolTip( clickText );
   mQuickAdd->setVisible( KOPrefs::instance()->enableQuickTodo() );
   connect( mQuickAdd, SIGNAL(returnPressed(Qt::KeyboardModifiers)),
            this, SLOT(addQuickTodo(Qt::KeyboardModifiers)) );
@@ -140,12 +138,6 @@
 
   setLayout( layout );
 
-  // Do elided text after the layout is set so the width is properly computed
-  mQuickAdd->setClickMessage(
-    fontMetrics().elidedText(
-      clickText, Qt::ElideRight,
-      mQuickAdd->width() + mQuickAdd->clearButtonUsedSize().width() ) );
-
   // ---------------- POPUP-MENUS -----------------------
   mItemPopupMenu = new QMenu( this );
 
Index: korganizer/kcmconfigs/korganizer_configcolorsandfonts.desktop
===================================================================
--- korganizer/kcmconfigs/korganizer_configcolorsandfonts.desktop	(.../tags/KDE/4.3.4/kdepim)	(wersja 1059841)
+++ korganizer/kcmconfigs/korganizer_configcolorsandfonts.desktop	(.../branches/KDE/4.3/kdepim)	(wersja 1059841)
@@ -42,7 +42,7 @@
 Name[pt]=Cores e Tipos de Letra
 Name[pt_BR]=Fontes e cores
 Name[ro]=Culori și fonturi
-Name[ru]=Цветы и шрифты
+Name[ru]=Цвета и шрифты
 Name[sl]=Barve in pisave
 Name[sv]=Färger och teckensnitt
 Name[th]=สีและแบบอักษร

Zmiany atrybutów dla: .
___________________________________________________________________
Dodane: svn:externals
   + 


